<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>dcAlgo @ dcGOR 1.0.5</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dcGOR 1.0.5</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
       
        <!-- HTML for GOOLE SEARCH BAR 
        <ul class="nav">
            <div id="search-wrapper">
                <form id="search" method="get" action="http://www.google.com/custom">
                    <div class="search-box">
                        <i class="icon-search icon-white"></i>
                        <input type="text" id="tfq" class="textinput" name="q" size="21" maxlength="120" value="Google Search"/>
                    </div>
                </form>
            </div>
        </ul>
        -->
        
      </div>
    </div>
  </div>
</div>

<!-- CSS styles for standard search box with placeholder text-->
<style type="text/css">
    #search-wrapper {
        width:300px;
        height:40px;
        background-color:transparent;
        margin:0px auto 0;
        border:0px solid #ff0000;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

    #search {
        background-color:transparent;
        padding:0px;
        height:40px;
        overflow:hidden;
    }

    .search-box {
        width:250px;
        height:32px;
        background-color:transparent;
        margin:2px 7px;
        border:0px solid #cfcfcf;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

	.textinput{
		margin: 0;
		padding: 5px 15px;
		font-family: Arial, Helvetica, sans-serif;
		font-size:14px;
		color:#999999;
		border:1px solid #f00; border-right:0px;
		border-top-left-radius: 5px 5px;
		border-bottom-left-radius: 5px 5px;
	}
</style>

<!-- JAVASCRIPT to clear search text when the field is clicked -->
<script type="text/javascript">
window.onload = function(){
	//Get submit button
	var submitbutton = document.getElementById("tfq");
	//Add listener to submit button
	if(submitbutton.addEventListener){
		submitbutton.addEventListener("click", function() {
			if (submitbutton.value == 'Google Search'){//Customize this text string to whatever you want
				submitbutton.value = 'dcGOR R ';
			}
		});
	}
}
</script>

<script>
  (function() {
    var cx = '012094530736424854714:k8ieyz4ag4a';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') +
        '//www.google.com/cse/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>

<div><gcse:search></gcse:search></div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Function to apply dcGO algorithm to infer domain-centric ontology</h2>

<div class="row">
  <div class="span8">
  
    <h3>Description</h3>
    <p><code>dcAlgo</code> is supposed to apply dcGO algorithm to infer
domain-centric ontology from input files. It requires two input files:
1) an annotation file containing annotations between proteins/genes and
ontology terms; 2) an architecture file containing domain architectures
for proteins/genes.</p>
  
    <h3>Usage</h3>
    <pre>dcAlgo(anno.file, architecture.file, output.file = NULL, ontology = c(NA, "GOBP", 
  "GOMF", "GOCC", "DO", "HPPA", "HPMI", "HPON", "MP", "EC", "KW", "UP"), feature.mode = c("supra", 
      "individual", "comb"), min.overlap = 3, fdr.cutoff = 0.001, hscore.type = c("zscore", 
      "fdr"), parallel = TRUE, multicores = NULL, verbose = T, RData.ontology.customised = NULL, 
      RData.location = "http://dcgor.r-forge.r-project.org/data")</pre>
    
    <h3>Arguments</h3>
    <dl>
      <dt>anno.file</dt>
      <dd>an annotation file containing annotations between
proteins/genes and ontology terms. For example, a file containing
annotations between human genes and HP terms can be found in
<a href = 'http://dcgor.r-forge.r-project.org/data/Algo/HP_anno.txt'>http://dcgor.r-forge.r-project.org/data/Algo/HP_anno.txt</a>. As seen
in this example, the input file must contain the header (in the first
row) and two columns: 1st column for 'SeqID' (actually these IDs can be
anything), 2nd column for 'termID' (HP terms). Note: the file should
use the tab delimiter as the field separator between columns</dd>
      <dt>architecture.file</dt>
      <dd>an architecture file containing domain
architectures (including individual domains) for proteins/genes. For
example, a file containing human genes and domain architectures can be
found in
<a href = 'http://dcgor.r-forge.r-project.org/data/Algo/SCOP_architecture.txt'>http://dcgor.r-forge.r-project.org/data/Algo/SCOP_architecture.txt</a>.
As seen in this example, the input file must contain the header (in the
first row) and two columns: 1st column for 'SeqID' (actually these IDs
can be anything), 2nd column for 'Architecture' (SCOP domain
architectures, each represented as comma-separated domains). Note: the
file should use the tab delimiter as the field separator between
columns</dd>
      <dt>output.file</dt>
      <dd>an output file containing results. If not NULL, a
tab-delimited text file will be also written out, with 1st column
'Feature_id' for features/domains, 2nd column 'Term_id' for ontology
terms, 3rd column 'Score' for hypergeometric scores (indicative of
strength for feature-term associations). Otherwise, there is no output
file (by default)</dd>
      <dt>ontology</dt>
      <dd>the ontology identity. It can be "GOBP" for Gene
Ontology Biological Process, "GOMF" for Gene Ontology Molecular
Function, "GOCC" for Gene Ontology Cellular Component, "DO" for Disease
Ontology, "HPPA" for Human Phenotype Phenotypic Abnormality, "HPMI" for
Human Phenotype Mode of Inheritance, "HPON" for Human Phenotype ONset
and clinical course, "MP" for Mammalian Phenotype, "EC" for Enzyme
Commission, "KW" for UniProtKB KeyWords, "UP" for UniProtKB UniPathway.
For details on the eligibility for pairs of input domain and ontology,
please refer to the online Documentations at
<a href = 'http://supfam.org/dcGOR/docs.html'>http://supfam.org/dcGOR/docs.html</a>. If NA, then the user has to
input a customised RData-formatted file (see
<code>RData.ontology.customised</code> below)</dd>
      <dt>feature.mode</dt>
      <dd>the mode of how to define the features thereof. It
can be: "supra" for combinations of one or two successive domains
(including individual domains; considering the order), "individual" for
individual domains only, and "comb" for all possible combinations
(including individual domains; ignoring the order)</dd>
      <dt>min.overlap</dt>
      <dd>the minimum number of overlaps with each term in
consideration. By default, it sets to a minimum of 3</dd>
      <dt>fdr.cutoff</dt>
      <dd>the fdr cutoff to call the significant associations
between features and terms. By default, it sets to 1e-3</dd>
      <dt>hscore.type</dt>
      <dd>the type of defining the hypergeometric score. It
can be: "zscore" for z-score (by default), "fdr" for fdr (after being
transformed via <code class = 'eq'>-1*log_2(fdr)</code>)</dd>
      <dt>parallel</dt>
      <dd>logical to indicate whether parallel computation with
multicores is used. By default, it sets to true, but not necessarily
does so. Partly because parallel backends available will be
system-specific (now only Linux or Mac OS). Also, it will depend on
whether these two packages "foreach" and "doMC" have been installed. It
can be installed via:
<code>source("http://bioconductor.org/biocLite.R");
biocLite(c("foreach","doMC"))</code>. If not yet installed, this option will
be disabled</dd>
      <dt>multicores</dt>
      <dd>an integer to specify how many cores will be
registered as the multicore parallel backend to the 'foreach' package.
If NULL, it will use a half of cores available in a user's computer.
This option only works when parallel computation is enabled</dd>
      <dt>verbose</dt>
      <dd>logical to indicate whether the messages will be
displayed in the screen. By default, it sets to TRUE for display</dd>
      <dt>RData.ontology.customised</dt>
      <dd>a file name for RData-formatted file
containing an object of S4 class 'Onto' (i.g. ontology). By default, it
is NULL. It is only needed when the user wants to perform customised
analysis using their own ontology. See <code><a href='dcBuildOnto.html'>dcBuildOnto</a></code> for
how to creat this object</dd>
      <dt>RData.location</dt>
      <dd>the characters to tell the location of built-in
RData files. By default, it remotely locates at
"http://supfam.org/dcGOR/data" or
"http://dcgor.r-forge.r-project.org/data". For the user equipped with
fast internet connection, this option can be just left as default. But
it is always advisable to download these files locally. Especially when
the user needs to run this function many times, there is no need to ask
the function to remotely download every time (also it will
unnecessarily increase the runtime). For examples, these files (as a
whole or part of them) can be first downloaded into your current
working directory, and then set this option as:
<code class = 'eq'>RData.location="."</code>. If RData to load is already part of package
itself, this parameter can be ignored (since this function will try to
load it via function <code>data</code> first)</dd>
    </dl>
    <!-- 
    -->
    
    <!-- 
    <div class="Value">
    -->
      <h3>Value</h3>
      
      <p>a data frame containing three columns: 1st column 'Feature_id' for
features, 2nd 'Term_id' for terms, and 3rd 'Score' for the
hypergeometric score indicative of strength of associations beteen
features and terms</p>
    <!-- 
    </div>
    -->
    <!-- 
    <div class="Note">
    -->
      <h3>Note</h3>
      
      <p>When 'output.file' is specified, a tab-delimited text file is output,
with the column names: 1st column 'Feature_id' for features, 2nd
'Term_id' for terms, and 3rd 'Score' for the hypergeometric score
indicative of strength of associations beteen features and terms</p>
    <!-- 
    </div>
    -->
    
    <h3 id="examples">Examples</h3>
    <pre class="examples"><div class='input'># 1) Prepare input file: anno.file and architecture.file
anno.file <- "http://dcgor.r-forge.r-project.org/data/Algo/HP_anno.txt"
architecture.file <-
"http://dcgor.r-forge.r-project.org/data/Algo/SCOP_architecture.txt"

# 2) Do inference using built-in data
res <- dcAlgo(anno.file, architecture.file, ontology="HPPA",
feature.mode="supra", parallel=FALSE)
</div>
<strong class='message'>Start at 2014-12-15 15:47:41</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'HPPA' (2014-12-15 15:47:41) ...</strong>
<strong class='message'>'onto.HPPA' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, import files for annotations 'http://dcgor.r-forge.r-project.org/data/Algo/HP_anno.txt' and architectures 'http://dcgor.r-forge.r-project.org/data/Algo/SCOP_architecture.txt' (2014-12-15 15:47:41) ...</strong>
<strong class='message'>Third, propagate annotations (2014-12-15 15:47:43) ...</strong>
<strong class='message'>	At level 16, there are 2 nodes, and 5 incoming neighbors.</strong>
<strong class='message'>	At level 15, there are 7 nodes, and 9 incoming neighbors.</strong>
<strong class='message'>	At level 14, there are 21 nodes, and 42 incoming neighbors.</strong>
<strong class='message'>	At level 13, there are 52 nodes, and 80 incoming neighbors.</strong>
<strong class='message'>	At level 12, there are 101 nodes, and 103 incoming neighbors.</strong>
<strong class='message'>	At level 11, there are 267 nodes, and 184 incoming neighbors.</strong>
<strong class='message'>	At level 10, there are 440 nodes, and 289 incoming neighbors.</strong>
<strong class='message'>	At level 9, there are 751 nodes, and 428 incoming neighbors.</strong>
<strong class='message'>	At level 8, there are 956 nodes, and 517 incoming neighbors.</strong>
<strong class='message'>	At level 7, there are 1141 nodes, and 571 incoming neighbors.</strong>
<strong class='message'>	At level 6, there are 1253 nodes, and 517 incoming neighbors.</strong>
<strong class='message'>	At level 5, there are 915 nodes, and 280 incoming neighbors.</strong>
<strong class='message'>	At level 4, there are 397 nodes, and 99 incoming neighbors.</strong>
<strong class='message'>	At level 3, there are 113 nodes, and 21 incoming neighbors.</strong>
<strong class='message'>	At level 2, there are 21 nodes, and 1 incoming neighbors.</strong>
<strong class='message'>	At level 1, there are 1 nodes, and 0 incoming neighbors.</strong>
<strong class='message'>	There are 6438 terms used (2014-12-15 15:48:04).</strong>
<strong class='message'>Fourth, define groups using feature mode 'supra' (2014-12-15 15:48:04) ...</strong>
<strong class='message'>	i) split into features (2014-12-15 15:48:04) ...</strong>
<strong class='message'>	1 out of 4351 (2014-12-15 15:48:04)</strong>
<strong class='message'>	436 out of 4351 (2014-12-15 15:48:05)</strong>
<strong class='message'>	872 out of 4351 (2014-12-15 15:48:05)</strong>
<strong class='message'>	1308 out of 4351 (2014-12-15 15:48:05)</strong>
<strong class='message'>	1744 out of 4351 (2014-12-15 15:48:06)</strong>
<strong class='message'>	2180 out of 4351 (2014-12-15 15:48:06)</strong>
<strong class='message'>	2616 out of 4351 (2014-12-15 15:48:06)</strong>
<strong class='message'>	3052 out of 4351 (2014-12-15 15:48:07)</strong>
<strong class='message'>	3488 out of 4351 (2014-12-15 15:48:07)</strong>
<strong class='message'>	3924 out of 4351 (2014-12-15 15:48:08)</strong>
<strong class='message'>	4351 out of 4351 (2014-12-15 15:48:08)</strong>
<strong class='message'>	ii) obtain feature-based groups (2014-12-15 15:48:08) ...</strong>
<strong class='message'>	1 out of 4351 (2014-12-15 15:48:08)</strong>
<strong class='message'>	436 out of 4351 (2014-12-15 15:48:08)</strong>
<strong class='message'>	872 out of 4351 (2014-12-15 15:48:08)</strong>
<strong class='message'>	1308 out of 4351 (2014-12-15 15:48:08)</strong>
<strong class='message'>	1744 out of 4351 (2014-12-15 15:48:08)</strong>
<strong class='message'>	2180 out of 4351 (2014-12-15 15:48:08)</strong>
<strong class='message'>	2616 out of 4351 (2014-12-15 15:48:08)</strong>
<strong class='message'>	3052 out of 4351 (2014-12-15 15:48:08)</strong>
<strong class='message'>	3488 out of 4351 (2014-12-15 15:48:08)</strong>
<strong class='message'>	3924 out of 4351 (2014-12-15 15:48:08)</strong>
<strong class='message'>	4351 out of 4351 (2014-12-15 15:48:08)</strong>
<strong class='message'>	There are 2194 features used (2014-12-15 15:48:08).</strong>
<strong class='message'>Finally, estimate associations between 2194 features and 6438 terms, with 3 min overlaps and 1.0e-03 fdr cutoff (2014-12-15 15:48:08) ...</strong>
<strong class='message'>	1 out of 2194 (2014-12-15 15:48:12)</strong>
<strong class='message'>	220 out of 2194 (2014-12-15 15:48:14)</strong>
<strong class='message'>	440 out of 2194 (2014-12-15 15:48:15)</strong>
<strong class='message'>	660 out of 2194 (2014-12-15 15:48:15)</strong>
<strong class='message'>	880 out of 2194 (2014-12-15 15:48:16)</strong>
<strong class='message'>	1100 out of 2194 (2014-12-15 15:48:17)</strong>
<strong class='message'>	1320 out of 2194 (2014-12-15 15:48:18)</strong>
<strong class='message'>	1540 out of 2194 (2014-12-15 15:48:19)</strong>
<strong class='message'>	1760 out of 2194 (2014-12-15 15:48:20)</strong>
<strong class='message'>	1980 out of 2194 (2014-12-15 15:48:21)</strong>
<strong class='message'>	2194 out of 2194 (2014-12-15 15:48:22)</strong>
<strong class='message'>
End at 2014-12-15 15:48:22</strong>
<strong class='message'>Runtime in total is: 41 secs
</strong>
<div class='input'>res[1:5,]
</div>
<div class='output'>  Feature_id    Term_id Score
1     100895 HP:0004295   4.8
2     103473 HP:0010932   5.0
3     103473 HP:0004352   1.3
4     103473 HP:0002908   3.2
5     103473 HP:0001289   4.3
</div>
<div class='input'>
# 3) Advanced usage: using customised data
x <-
base::load(base::url("http://dcgor.r-forge.r-project.org/data/onto.HPPA.RData"))
RData.ontology.customised <- 'onto.HPPA.RData'
base::save(list=x, file=RData.ontology.customised)
#list.files(pattern='*.RData')
## you will see an RData file 'onto.HPPA.RData' in local directory
res <- dcAlgo(anno.file, architecture.file, feature.mode="supra",
parallel=FALSE, RData.ontology.customised=RData.ontology.customised)
</div>
<strong class='message'>Start at 2014-12-15 15:48:23</strong>
<strong class='message'></strong>
<strong class='message'>First, load customised ontology 'onto.HPPA.RData' (2014-12-15 15:48:23)...</strong>
<strong class='message'>Second, import files for annotations 'http://dcgor.r-forge.r-project.org/data/Algo/HP_anno.txt' and architectures 'http://dcgor.r-forge.r-project.org/data/Algo/SCOP_architecture.txt' (2014-12-15 15:48:23) ...</strong>
<strong class='message'>Third, propagate annotations (2014-12-15 15:48:25) ...</strong>
<strong class='message'>	At level 16, there are 2 nodes, and 5 incoming neighbors.</strong>
<strong class='message'>	At level 15, there are 7 nodes, and 9 incoming neighbors.</strong>
<strong class='message'>	At level 14, there are 21 nodes, and 42 incoming neighbors.</strong>
<strong class='message'>	At level 13, there are 52 nodes, and 80 incoming neighbors.</strong>
<strong class='message'>	At level 12, there are 101 nodes, and 103 incoming neighbors.</strong>
<strong class='message'>	At level 11, there are 267 nodes, and 184 incoming neighbors.</strong>
<strong class='message'>	At level 10, there are 440 nodes, and 289 incoming neighbors.</strong>
<strong class='message'>	At level 9, there are 751 nodes, and 428 incoming neighbors.</strong>
<strong class='message'>	At level 8, there are 956 nodes, and 517 incoming neighbors.</strong>
<strong class='message'>	At level 7, there are 1141 nodes, and 571 incoming neighbors.</strong>
<strong class='message'>	At level 6, there are 1253 nodes, and 517 incoming neighbors.</strong>
<strong class='message'>	At level 5, there are 915 nodes, and 280 incoming neighbors.</strong>
<strong class='message'>	At level 4, there are 397 nodes, and 99 incoming neighbors.</strong>
<strong class='message'>	At level 3, there are 113 nodes, and 21 incoming neighbors.</strong>
<strong class='message'>	At level 2, there are 21 nodes, and 1 incoming neighbors.</strong>
<strong class='message'>	At level 1, there are 1 nodes, and 0 incoming neighbors.</strong>
<strong class='message'>	There are 6438 terms used (2014-12-15 15:48:44).</strong>
<strong class='message'>Fourth, define groups using feature mode 'supra' (2014-12-15 15:48:44) ...</strong>
<strong class='message'>	i) split into features (2014-12-15 15:48:44) ...</strong>
<strong class='message'>	1 out of 4351 (2014-12-15 15:48:44)</strong>
<strong class='message'>	436 out of 4351 (2014-12-15 15:48:44)</strong>
<strong class='message'>	872 out of 4351 (2014-12-15 15:48:45)</strong>
<strong class='message'>	1308 out of 4351 (2014-12-15 15:48:45)</strong>
<strong class='message'>	1744 out of 4351 (2014-12-15 15:48:45)</strong>
<strong class='message'>	2180 out of 4351 (2014-12-15 15:48:46)</strong>
<strong class='message'>	2616 out of 4351 (2014-12-15 15:48:46)</strong>
<strong class='message'>	3052 out of 4351 (2014-12-15 15:48:46)</strong>
<strong class='message'>	3488 out of 4351 (2014-12-15 15:48:47)</strong>
<strong class='message'>	3924 out of 4351 (2014-12-15 15:48:47)</strong>
<strong class='message'>	4351 out of 4351 (2014-12-15 15:48:48)</strong>
<strong class='message'>	ii) obtain feature-based groups (2014-12-15 15:48:48) ...</strong>
<strong class='message'>	1 out of 4351 (2014-12-15 15:48:48)</strong>
<strong class='message'>	436 out of 4351 (2014-12-15 15:48:48)</strong>
<strong class='message'>	872 out of 4351 (2014-12-15 15:48:48)</strong>
<strong class='message'>	1308 out of 4351 (2014-12-15 15:48:48)</strong>
<strong class='message'>	1744 out of 4351 (2014-12-15 15:48:48)</strong>
<strong class='message'>	2180 out of 4351 (2014-12-15 15:48:48)</strong>
<strong class='message'>	2616 out of 4351 (2014-12-15 15:48:48)</strong>
<strong class='message'>	3052 out of 4351 (2014-12-15 15:48:48)</strong>
<strong class='message'>	3488 out of 4351 (2014-12-15 15:48:48)</strong>
<strong class='message'>	3924 out of 4351 (2014-12-15 15:48:48)</strong>
<strong class='message'>	4351 out of 4351 (2014-12-15 15:48:48)</strong>
<strong class='message'>	There are 2194 features used (2014-12-15 15:48:48).</strong>
<strong class='message'>Finally, estimate associations between 2194 features and 6438 terms, with 3 min overlaps and 1.0e-03 fdr cutoff (2014-12-15 15:48:48) ...</strong>
<strong class='message'>	1 out of 2194 (2014-12-15 15:48:50)</strong>
<strong class='message'>	220 out of 2194 (2014-12-15 15:48:51)</strong>
<strong class='message'>	440 out of 2194 (2014-12-15 15:48:52)</strong>
<strong class='message'>	660 out of 2194 (2014-12-15 15:48:53)</strong>
<strong class='message'>	880 out of 2194 (2014-12-15 15:48:54)</strong>
<strong class='message'>	1100 out of 2194 (2014-12-15 15:48:55)</strong>
<strong class='message'>	1320 out of 2194 (2014-12-15 15:48:56)</strong>
<strong class='message'>	1540 out of 2194 (2014-12-15 15:48:57)</strong>
<strong class='message'>	1760 out of 2194 (2014-12-15 15:48:58)</strong>
<strong class='message'>	1980 out of 2194 (2014-12-15 15:48:59)</strong>
<strong class='message'>	2194 out of 2194 (2014-12-15 15:49:00)</strong>
<strong class='message'>
End at 2014-12-15 15:49:00</strong>
<strong class='message'>Runtime in total is: 37 secs
</strong>
<div class='input'>res[1:5,]
</div>
<div class='output'>  Feature_id    Term_id Score
1     100895 HP:0004295   4.8
2     103473 HP:0010932   5.0
3     103473 HP:0004352   1.3
4     103473 HP:0002908   3.2
5     103473 HP:0001289   4.3
</div></pre>
  </div>
  
  <div class="span4">
    
    <!-- 
    <ul>
      <li>dcAlgo</li>
    </ul>
    <ul>
      
    </ul> 
    -->
    
    <h2>Source code</h2>
    <a href="javascript:newWin('R/dcAlgo.r', 'R/dcAlgo.r', '1200', '1200')" title="Click to view"><B><code>dcAlgo.r</code></B></a>
    
    <h2>Source man</h2>
    <a href="javascript:newWin('man/dcAlgo.txt', 'man/dcAlgo.txt', '1200', '1200')" title="Click to view"><B><code>dcAlgo.Rd</code></B></a>
    
    <a href="javascript:newWin('man/dcAlgo.pdf', 'man/dcAlgo.pdf', '1200', '1200')" title="Click to view"><B><code>dcAlgo.pdf</code></B></a>
    
    <h2>See also</h2>
    
<code><a href='dcRDataLoader.html'>dcRDataLoader</a></code>, <code><a href='dcSplitArch.html'>dcSplitArch</a></code>,
<code><a href='dcConverter.html'>dcConverter</a></code>, <code><a href='dcDuplicated.html'>dcDuplicated</a></code>,
<code><a href='dcAlgoPropagate.html'>dcAlgoPropagate</a></code>

    <!--

    <h2>Citation</h2>
    <p>Fang H. (2014) <B>dcGOR: an R package for analysing ontologies and protein domain annotations</B>. <I>PLoS Computational Biology</I> 10(10):e1003929. <a href="http://dx.doi.org/10.1371/journal.pcbi.1003929">doi:10.1371/journal.pcbi.1003929</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25356683">25356683</a></p>

<p>You may also cite the package as:</p>

<ul>
<li><code>database</code></li>
</ul>

<p>Fang H, Gough J. (2013) <B>dcGO: database of domain-centric ontologies on functions, phenotypes, diseases and more</B>. <I>Nucleic Acids Research</I> 41(Database issue):D536-44. <a href="http://dx.doi.org/10.1093/nar/gks1080">doi:10.1093/nar/gks1080</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23161684">23161684</a></p>

<ul>
<li><code>web link/URL</code></li>
</ul>

<p>Package &ldquo;dcGOR&rdquo; [<a href="http://cran.r-project.org/package=dcGOR">http://cran.r-project.org/package=dcGOR</a>]</p>

    
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://supfam.org/dcGOR">http://supfam.org/dcGOR</a></li>
      <li><a href=" http://dcgor.r-forge.r-project.org"> http://dcgor.r-forge.r-project.org</a></li>
      <li><a href=" http://cran.r-project.org/package=dcGOR"> http://cran.r-project.org/package=dcGOR</a></li>
      <li><a href=" https://github.com/hfang-bristol/dcGOR"> https://github.com/hfang-bristol/dcGOR</a></li>
    </ul>
    
    
    -->
    
  </div>
  
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>