<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>dcEnrichment @ dcGOR 1.0.5</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dcGOR 1.0.5</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
       
        <!-- HTML for GOOLE SEARCH BAR 
        <ul class="nav">
            <div id="search-wrapper">
                <form id="search" method="get" action="http://www.google.com/custom">
                    <div class="search-box">
                        <i class="icon-search icon-white"></i>
                        <input type="text" id="tfq" class="textinput" name="q" size="21" maxlength="120" value="Google Search"/>
                    </div>
                </form>
            </div>
        </ul>
        -->
        
      </div>
    </div>
  </div>
</div>

<!-- CSS styles for standard search box with placeholder text-->
<style type="text/css">
    #search-wrapper {
        width:300px;
        height:40px;
        background-color:transparent;
        margin:0px auto 0;
        border:0px solid #ff0000;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

    #search {
        background-color:transparent;
        padding:0px;
        height:40px;
        overflow:hidden;
    }

    .search-box {
        width:250px;
        height:32px;
        background-color:transparent;
        margin:2px 7px;
        border:0px solid #cfcfcf;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

	.textinput{
		margin: 0;
		padding: 5px 15px;
		font-family: Arial, Helvetica, sans-serif;
		font-size:14px;
		color:#999999;
		border:1px solid #f00; border-right:0px;
		border-top-left-radius: 5px 5px;
		border-bottom-left-radius: 5px 5px;
	}
</style>

<!-- JAVASCRIPT to clear search text when the field is clicked -->
<script type="text/javascript">
window.onload = function(){
	//Get submit button
	var submitbutton = document.getElementById("tfq");
	//Add listener to submit button
	if(submitbutton.addEventListener){
		submitbutton.addEventListener("click", function() {
			if (submitbutton.value == 'Google Search'){//Customize this text string to whatever you want
				submitbutton.value = 'dcGOR R ';
			}
		});
	}
}
</script>

<script>
  (function() {
    var cx = '012094530736424854714:k8ieyz4ag4a';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') +
        '//www.google.com/cse/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>

<div><gcse:search></gcse:search></div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Function to conduct ontology enrichment analysis given a group of domains</h2>

<div class="row">
  <div class="span8">
  
    <h3>Description</h3>
    <p><code>dcEnrichment</code> is supposed to conduct enrichment analysis for an
input group of domains using a specified ontology. It returns an object
of S4 class "Eoutput". Enrichment analysis is based on either Fisher's
exact test or Hypergeometric test. The test can respect the hierarchy
of the ontology. The user can customise the background domains;
otherwise, the function will use all annotatable domains as the test
background</p>
  
    <h3>Usage</h3>
    <pre>dcEnrichment(data, background = NULL, domain = c(NA, "SCOP.sf", "SCOP.fa", "Pfam", 
  "InterPro", "Rfam"), ontology = c(NA, "GOBP", "GOMF", "GOCC", "DO", "HPPA", "HPMI", 
      "HPON", "MP", "EC", "KW", "UP"), sizeRange = c(10, 1000), min.overlap = 3, which_distance = NULL, 
      test = c("HypergeoTest", "FisherTest", "BinomialTest"), p.adjust.method = c("BH", 
          "BY", "bonferroni", "holm", "hochberg", "hommel"), ontology.algorithm = c("none", 
          "pc", "elim", "lea"), elim.pvalue = 0.01, lea.depth = 2, verbose = T, domain.RData = NULL, 
      ontology.RData = NULL, annotations.RData = NULL, RData.location = "http://dcgor.r-forge.r-project.org/data")</pre>
    
    <h3>Arguments</h3>
    <dl>
      <dt>data</dt>
      <dd>an input vector. It contains id for a list of domains, for
example, sunids for SCOP domains</dd>
      <dt>background</dt>
      <dd>a background vector. It contains id for a list of
background domains, for example, sunids for SCOP domains. If NULL, by
default all annotatable domains are used as background</dd>
      <dt>domain</dt>
      <dd>the domain identity. It can be one of 'SCOP.sf' for SCOP
superfamilies, 'SCOP.fa' for SCOP families, 'Pfam' for Pfam domains,
'InterPro' for InterPro domains, 'Rfam' for Rfam RNA families</dd>
      <dt>ontology</dt>
      <dd>the ontology identity. It can be "GOBP" for Gene
Ontology Biological Process, "GOMF" for Gene Ontology Molecular
Function, "GOCC" for Gene Ontology Cellular Component, "DO" for Disease
Ontology, "HPPA" for Human Phenotype Phenotypic Abnormality, "HPMI" for
Human Phenotype Mode of Inheritance, "HPON" for Human Phenotype ONset
and clinical course, "MP" for Mammalian Phenotype, "EC" for Enzyme
Commission, "KW" for UniProtKB KeyWords, "UP" for UniProtKB UniPathway.
For details on the eligibility for pairs of input domain and ontology,
please refer to the online Documentations at
<a href = 'http://supfam.org/dcGOR/docs.html'>http://supfam.org/dcGOR/docs.html</a></dd>
      <dt>sizeRange</dt>
      <dd>the minimum and maximum size of members of each term
in consideration. By default, it sets to a minimum of 10 but no more
than 1000</dd>
      <dt>min.overlap</dt>
      <dd>the minimum number of overlaps. Only those terms
that overlap with input data at least min.overlap (3 domains by
default) will be processed</dd>
      <dt>which_distance</dt>
      <dd>which distance of terms in the ontology is used
to restrict terms in consideration. By default, it sets to 'NULL' to
consider all distances</dd>
      <dt>test</dt>
      <dd>the statistic test used. It can be "FisherTest" for using
fisher's exact test, "HypergeoTest" for using hypergeometric test, or
"BinomialTest" for using binomial test. Fisher's exact test is to test
the independence between domain group (domains belonging to a group or
not) and domain annotation (domains annotated by a term or not), and
thus compare sampling to the left part of background (after sampling
without replacement). Hypergeometric test is to sample at random
(without replacement) from the background containing annotated and
non-annotated domains, and thus compare sampling to background. Unlike
hypergeometric test, binomial test is to sample at random (with
replacement) from the background with the constant probability. In
terms of the ease of finding the significance, they are in order:
hypergeometric test > binomial test > fisher's exact test. In other
words, in terms of the calculated p-value, hypergeometric test <
binomial test < fisher's exact test</dd>
      <dt>p.adjust.method</dt>
      <dd>the method used to adjust p-values. It can be
one of "BH", "BY", "bonferroni", "holm", "hochberg" and "hommel". The
first two methods "BH" (widely used) and "BY" control the false
discovery rate (FDR: the expected proportion of false discoveries
amongst the rejected hypotheses); the last four methods "bonferroni",
"holm", "hochberg" and "hommel" are designed to give strong control of
the family-wise error rate (FWER). Notes: FDR is a less stringent
condition than FWER</dd>
      <dt>ontology.algorithm</dt>
      <dd>the algorithm used to account for the
hierarchy of the ontology. It can be one of "none", "pc", "elim" and
"lea". For details, please see 'Note'</dd>
      <dt>elim.pvalue</dt>
      <dd>the parameter only used when "ontology.algorithm" is
"elim". It is used to control how to declare a signficantly enriched
term (and subsequently all domains in this term are eliminated from all
its ancestors)</dd>
      <dt>lea.depth</dt>
      <dd>the parameter only used when "ontology.algorithm" is
"lea". It is used to control how many maximum depth is uded to consider
the children of a term (and subsequently all domains in these children
term are eliminated from the use for the recalculation of the
signifance at this term)</dd>
      <dt>verbose</dt>
      <dd>logical to indicate whether the messages will be
displayed in the screen. By default, it sets to TRUE for display</dd>
      <dt>domain.RData</dt>
      <dd>a file name for RData-formatted file containing an
object of S4 class 'InfoDataFrame' (i.g. domain). By default, it is
NULL. It is only needed when the user wants to customise enrichment
analysis using their own data. See <code><a href='dcBuildInfoDataFrame.html'>dcBuildInfoDataFrame</a></code>
for how to creat this object</dd>
      <dt>ontology.RData</dt>
      <dd>a file name for RData-formatted file containing
an object of S4 class 'Onto' (i.g. ontology). By default, it is NULL.
It is only needed when the user wants to customise enrichment analysis
using their own data. See <code><a href='dcBuildOnto.html'>dcBuildOnto</a></code> for how to creat
this object</dd>
      <dt>annotations.RData</dt>
      <dd>a file name for RData-formatted file
containing an object of S4 class 'Anno' (i.g. annotations). By default,
it is NULL. It is only needed when the user wants to customise
enrichment analysis using their own data. See <code><a href='dcBuildAnno.html'>dcBuildAnno</a></code>
for how to creat this object</dd>
      <dt>RData.location</dt>
      <dd>the characters to tell the location of built-in
RData files. By default, it remotely locates at
"http://supfam.org/dcGOR/data" or
"http://dcgor.r-forge.r-project.org/data". For the user equipped with
fast internet connection, this option can be just left as default. But
it is always advisable to download these files locally. Especially when
the user needs to run this function many times, there is no need to ask
the function to remotely download every time (also it will
unnecessarily increase the runtime). For examples, these files (as a
whole or part of them) can be first downloaded into your current
working directory, and then set this option as:
<code class = 'eq'>RData.location="."</code>. If RData to load is already part of package
itself, this parameter can be ignored (since this function will try to
load it via function <code>data</code> first)</dd>
    </dl>
    <!-- 
    -->
    
    <!-- 
    <div class="Value">
    -->
      <h3>Value</h3>
      
      <p>an object of S4 class <code><a href='Eoutput-class.html'>Eoutput</a></code>, with following slots:
<ul>
<li><code>domain</code>: a character specifying the domain identity
</li>
<li><code>ontology</code>: a character specifying the ontology used
</li>
<li><code>term_info</code>: a matrix of nTerm X 5 containing term
information, where nTerm is the number of terms in consideration, and
the 5 columns are "term_id" (i.e. "Term ID"), "term_name" (i.e. "Term
Name"), "namespace" (i.e. "Term Namespace"), "distance" (i.e. "Term
Distance") and "IC" (i.e. "Information Content for the term based on
annotation frequency by it")
</li>
<li><code>anno</code>: a list of terms, each storing annotated domain
members (also within the background domains). Always, terms are
identified by "term_id" and domain members identified by their ids
(e.g. sunids for SCOP domains)
</li>
<li><code>data</code>: a vector containing input data in consideration. It
is not always the same as the input data as only those mappable and
annotatable are retained
</li>
<li><code>background</code>: a vector containing background in
consideration. It is not always the same as the input background as
only those mappable/annotatable are retained
</li>
<li><code>overlap</code>: a list of terms, each storing domains overlapped
between domains annotated by a term and domains in the input data (i.e.
the domains of interest). Always, terms are identified by "term_id" and
domain members identified by their IDs (e.g. sunids for SCOP domains)
</li>
<li><code>zscore</code>: a vector containing z-scores
</li>
<li><code>pvalue</code>: a vector containing p-values
</li>
<li><code>adjp</code>: a vector containing adjusted p-values. It is the p
value but after being adjusted for multiple comparisons
</li>
</ul></p>
      <p></p>
    <!-- 
    </div>
    -->
    <!-- 
    <div class="Note">
    -->
      <h3>Note</h3>
      
      <p>The interpretation of the algorithms used to account for the hierarchy
of the ontology is:
<ul>
<li>"none": does not consider the ontology hierarchy at all.
</li>
<li>"lea": computers the significance of a term in terms of the
significance of its children at the maximum depth (e.g. 2). Precisely,
once domains are already annotated to any children terms with a more
signficance than itself, then all these domains are eliminated from the
use for the recalculation of the signifance at that term. The final
p-values takes the maximum of the original p-value and the recalculated
p-value.
</li>
<li>"elim": computers the significance of a term in terms of the
significance of its all children. Precisely, once domains are already
annotated to a signficantly enriched term under the cutoff of e.g.
pvalue<1e-2, all these domains are eliminated from the ancestors of
that term).
</li>
<li>"pc": requires the significance of a term not only using the
whole domains as background but also using domains annotated to all its
direct parents/ancestors as background. The final p-value takes the
maximum of both p-values in these two calculations.
</li>
<li>"Notes": the order of the number of significant terms is: "none"
> "lea" > "elim" > "pc".
</li>
</ul></p>
      <p></p>
    <!-- 
    </div>
    -->
    
    <h3 id="examples">Examples</h3>
    <pre class="examples"><div class='input'># 1) Enrichment analysis for SCOP domain superfamilies (sf)
## 1a) load SCOP.sf (as 'InfoDataFrame' object)
SCOP.sf <- dcRDataLoader('SCOP.sf')
</div>
<strong class='message'>'SCOP.sf' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<div class='input'>### randomly select 50 domains as a list of domains of interest
data <- sample(rowNames(SCOP.sf), 50)
## 1b) perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, domain="SCOP.sf", ontology="GOMF")
</div>
<strong class='message'>Start at 2015-06-04 16:17:23</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'SCOP.sf', and their associations (2015-06-04 16:17:23) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf2GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2015-06-04 16:17:29) ...</strong>
<strong class='message'>	There are 811 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2015-06-04 16:17:30) ...</strong>
<strong class='message'>
End at 2015-06-04 16:17:30</strong>
<strong class='message'>Runtime in total is: 7 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'SCOP.sf'
  @ontology: 'GOMF'
  @term_info: a data.frame of 152 terms X 5 information
  @anno: a list of 152 terms, each storing annotated domains
  @data: a vector containing a group of 25 input domains (annotatable)
  @background: a vector containing a group of 1083 background domains (annotatable)
  @overlap: a list of 152 terms, each containing domains overlapped with input domains
  @zscore: a vector of 152 terms, containing z-scores
  @pvalue: a vector of 152 terms, containing p-values
  @adjp: a vector of 152 terms, containing adjusted p-values
In summary, a total of 152 terms ('GOMF') are analysed for a group of 25 input domains ('SCOP.sf')
</div>
<div class='input'>## 1c) view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore  pvalue  adjp
GO:0032559 GO:0032559    18     25        3   4.09 0.00055 0.039
GO:0030554 GO:0030554    19     25        3   3.95 0.00068 0.039
GO:0005487 GO:0005487    34     25        4   3.73 0.00076 0.039
GO:0043021 GO:0043021    28     25        3   3.00 0.00310 0.120
GO:0008409 GO:0008409    34     25        3   2.57 0.00640 0.200
GO:0051540 GO:0051540    69     25        4   1.99 0.01800 0.450
GO:0016298 GO:0016298   101     25        5   1.86 0.02300 0.480
GO:0051538 GO:0051538    55     25        3   1.59 0.03400 0.480
GO:0043565 GO:0043565   111     25        5   1.63 0.03500 0.480
GO:0019237 GO:0019237    84     25        4   1.56 0.03800 0.480
                                        term_name     term_namespace
GO:0032559          adenyl ribonucleotide binding molecular_function
GO:0030554              adenyl nucleotide binding molecular_function
GO:0005487 nucleocytoplasmic transporter activity molecular_function
GO:0043021      ribonucleoprotein complex binding molecular_function
GO:0008409             5'-3' exonuclease activity molecular_function
GO:0051540                  metal cluster binding molecular_function
GO:0016298                        lipase activity molecular_function
GO:0051538       3 iron, 4 sulfur cluster binding molecular_function
GO:0043565          sequence-specific DNA binding molecular_function
GO:0019237                centromeric DNA binding molecular_function
           term_distance                          members
GO:0032559             7                46955,53092,47719
GO:0030554             6                46955,53092,47719
GO:0005487             2       109885,161070,90096,143575
GO:0043021             2               46955,161070,50715
GO:0008409             6               161070,90096,56219
GO:0051540             2        158457,46955,47719,143575
GO:0016298             4   46955,161070,47719,56219,50715
GO:0051538             4               158457,46955,47719
GO:0043565             5 52029,109885,161070,90096,143575
GO:0019237             6       109885,161070,90096,143575
</div>
<div class='input'>## 1d) visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 18 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<div class='input'>## 1e) the same as above but using a customised background
### randomly select 500 domains as background
background <- sample(rowNames(SCOP.sf), 500)
</div>
<a href="javascript:newWin('dcEnrichment-22.png', 'dcEnrichment-22.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-22.png', width="400" /></a>
<div class='input'>### perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, background=background, domain="SCOP.sf",
ontology="GOMF")
</div>
<strong class='message'>Start at 2015-06-04 16:17:32</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'SCOP.sf', and their associations (2015-06-04 16:17:32) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf2GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2015-06-04 16:17:37) ...</strong>
<strong class='message'>	There are 432 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2015-06-04 16:17:37) ...</strong>
<strong class='message'>
End at 2015-06-04 16:17:37</strong>
<strong class='message'>Runtime in total is: 5 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'SCOP.sf'
  @ontology: 'GOMF'
  @term_info: a data.frame of 147 terms X 5 information
  @anno: a list of 147 terms, each storing annotated domains
  @data: a vector containing a group of 25 input domains (annotatable)
  @background: a vector containing a group of 268 background domains (annotatable)
  @overlap: a list of 147 terms, each containing domains overlapped with input domains
  @zscore: a vector of 147 terms, containing z-scores
  @pvalue: a vector of 147 terms, containing p-values
  @adjp: a vector of 147 terms, containing adjusted p-values
In summary, a total of 147 terms ('GOMF') are analysed for a group of 25 input domains ('SCOP.sf')
</div>
<div class='input'>### view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue adjp
GO:0019237 GO:0019237    19     25        4   1.82  0.022 0.79
GO:0032555 GO:0032555    13     25        3   1.74  0.024 0.79
GO:0016298 GO:0016298    27     25        5   1.73  0.027 0.79
GO:0017076 GO:0017076    14     25        3   1.60  0.031 0.79
GO:0043565 GO:0043565    29     25        5   1.55  0.038 0.79
GO:0032553 GO:0032553    15     25        3   1.46  0.040 0.79
GO:0004620 GO:0004620    22     25        4   1.49  0.041 0.79
GO:0051540 GO:0051540    23     25        4   1.39  0.049 0.79
GO:0003677 GO:0003677    59     25        8   1.26  0.069 0.79
GO:0051538 GO:0051538    18     25        3   1.11  0.073 0.79
                                  term_name     term_namespace term_distance
GO:0019237          centromeric DNA binding molecular_function             6
GO:0032555    purine ribonucleotide binding molecular_function             6
GO:0016298                  lipase activity molecular_function             4
GO:0017076        purine nucleotide binding molecular_function             5
GO:0043565    sequence-specific DNA binding molecular_function             5
GO:0032553           ribonucleotide binding molecular_function             5
GO:0004620           phospholipase activity molecular_function             5
GO:0051540            metal cluster binding molecular_function             2
GO:0003677                      DNA binding molecular_function             4
GO:0051538 3 iron, 4 sulfur cluster binding molecular_function             4
                                                      members
GO:0019237                         90096,161070,143575,109885
GO:0032555                                  47719,46955,53092
GO:0016298                     47719,46955,161070,56219,50715
GO:0017076                                  47719,46955,53092
GO:0043565                   90096,161070,143575,109885,52029
GO:0032553                                  47719,46955,53092
GO:0004620                           47719,46955,161070,50715
GO:0051540                          47719,46955,143575,158457
GO:0003677 54782,90096,53092,47027,161070,143575,109885,52029
GO:0051538                                 47719,46955,158457
</div>
<div class='input'>### visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 17 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<div class='input'>
###########################################################
# 2) Enrichment analysis for Pfam domains (Pfam)
## 2a) load Pfam (as 'InfoDataFrame' object)
Pfam <- dcRDataLoader('Pfam')
</div>
<strong class='message'>'Pfam' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<a href="javascript:newWin('dcEnrichment-43.png', 'dcEnrichment-43.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-43.png', width="400" /></a>
<div class='input'>### randomly select 100 domains as a list of domains of interest
data <- sample(rowNames(Pfam), 100)
## 2b) perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, domain="Pfam", ontology="GOMF")
</div>
<strong class='message'>Start at 2015-06-04 16:17:40</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'Pfam', and their associations (2015-06-04 16:17:40) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Pfam' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Pfam2GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2015-06-04 16:17:43) ...</strong>
<strong class='message'>	There are 334 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2015-06-04 16:17:43) ...</strong>
<strong class='message'>
End at 2015-06-04 16:17:43</strong>
<strong class='message'>Runtime in total is: 3 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'Pfam'
  @ontology: 'GOMF'
  @term_info: a data.frame of 16 terms X 5 information
  @anno: a list of 16 terms, each storing annotated domains
  @data: a vector containing a group of 22 input domains (annotatable)
  @background: a vector containing a group of 3359 background domains (annotatable)
  @overlap: a list of 16 terms, each containing domains overlapped with input domains
  @zscore: a vector of 16 terms, containing z-scores
  @pvalue: a vector of 16 terms, containing p-values
  @adjp: a vector of 16 terms, containing adjusted p-values
In summary, a total of 16 terms ('GOMF') are analysed for a group of 22 input domains ('Pfam')
</div>
<div class='input'>## 2c) view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue   adjp
GO:0019210 GO:0019210    82     22        3  3.410 0.0017 0.0082
GO:0004860 GO:0004860    82     22        3  3.410 0.0017 0.0082
GO:0019887 GO:0019887    85     22        3  3.330 0.0020 0.0082
GO:0019207 GO:0019207    86     22        3  3.300 0.0021 0.0082
GO:0004857 GO:0004857   106     22        3  2.820 0.0044 0.0140
GO:0030234 GO:0030234   198     22        3  1.550 0.0370 0.0990
GO:0022891 GO:0022891   273     22        3  0.949 0.0980 0.2000
GO:0016740 GO:0016740   767     22        7  1.010 0.1100 0.2000
GO:0022857 GO:0022857   290     22        3  0.838 0.1200 0.2000
GO:0022892 GO:0022892   296     22        3  0.801 0.1200 0.2000
                                                       term_name
GO:0019210                             kinase inhibitor activity
GO:0004860                     protein kinase inhibitor activity
GO:0019887                     protein kinase regulator activity
GO:0019207                             kinase regulator activity
GO:0004857                             enzyme inhibitor activity
GO:0030234                             enzyme regulator activity
GO:0022891 substrate-specific transmembrane transporter activity
GO:0016740                                  transferase activity
GO:0022857                    transmembrane transporter activity
GO:0022892               substrate-specific transporter activity
               term_namespace term_distance
GO:0019210 molecular_function             3
GO:0004860 molecular_function             4
GO:0019887 molecular_function             3
GO:0019207 molecular_function             2
GO:0004857 molecular_function             2
GO:0030234 molecular_function             1
GO:0022891 molecular_function             3
GO:0016740 molecular_function             2
GO:0022857 molecular_function             2
GO:0022892 molecular_function             2
                                                           members
GO:0019210                                 PF06452,PF05148,PF04616
GO:0004860                                 PF06452,PF05148,PF04616
GO:0019887                                 PF06452,PF05148,PF04616
GO:0019207                                 PF06452,PF05148,PF04616
GO:0004857                                 PF06452,PF05148,PF04616
GO:0030234                                 PF06452,PF05148,PF04616
GO:0022891                                 PF03418,PF15291,PF05388
GO:0016740 PF00328,PF00444,PF03418,PF06452,PF15291,PF09478,PF02172
GO:0022857                                 PF03418,PF15291,PF05388
GO:0022892                                 PF03418,PF15291,PF05388
</div>
<div class='input'>## 2d) visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 7 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<div class='input'>## 2e) the same as above but using a customised background
### randomly select 1000 domains as background
background <- sample(rowNames(Pfam), 1000)
</div>
<a href="javascript:newWin('dcEnrichment-63.png', 'dcEnrichment-63.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-63.png', width="400" /></a>
<div class='input'>### perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, background=background, domain="Pfam",
ontology="GOMF")
</div>
<strong class='message'>Start at 2015-06-04 16:17:45</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'Pfam', and their associations (2015-06-04 16:17:45) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Pfam' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Pfam2GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2015-06-04 16:17:48) ...</strong>
<strong class='message'>	There are 44 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2015-06-04 16:17:48) ...</strong>
<strong class='message'>
End at 2015-06-04 16:17:48</strong>
<strong class='message'>Runtime in total is: 3 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'Pfam'
  @ontology: 'GOMF'
  @term_info: a data.frame of 16 terms X 5 information
  @anno: a list of 16 terms, each storing annotated domains
  @data: a vector containing a group of 22 input domains (annotatable)
  @background: a vector containing a group of 252 background domains (annotatable)
  @overlap: a list of 16 terms, each containing domains overlapped with input domains
  @zscore: a vector of 16 terms, containing z-scores
  @pvalue: a vector of 16 terms, containing p-values
  @adjp: a vector of 16 terms, containing adjusted p-values
In summary, a total of 16 terms ('GOMF') are analysed for a group of 22 input domains ('Pfam')
</div>
<div class='input'>### view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue  adjp
GO:0019210 GO:0019210    12     22        3  2.040  0.014 0.059
GO:0004860 GO:0004860    12     22        3  2.040  0.014 0.059
GO:0004857 GO:0004857    13     22        3  1.880  0.019 0.059
GO:0019207 GO:0019207    13     22        3  1.880  0.019 0.059
GO:0019887 GO:0019887    13     22        3  1.880  0.019 0.059
GO:0030234 GO:0030234    19     22        3  1.130  0.070 0.180
GO:0005215 GO:0005215    22     22        3  0.852  0.110 0.180
GO:0022857 GO:0022857    22     22        3  0.852  0.110 0.180
GO:0022892 GO:0022892    22     22        3  0.852  0.110 0.180
GO:0022891 GO:0022891    22     22        3  0.852  0.110 0.180
                                                       term_name
GO:0019210                             kinase inhibitor activity
GO:0004860                     protein kinase inhibitor activity
GO:0004857                             enzyme inhibitor activity
GO:0019207                             kinase regulator activity
GO:0019887                     protein kinase regulator activity
GO:0030234                             enzyme regulator activity
GO:0005215                                  transporter activity
GO:0022857                    transmembrane transporter activity
GO:0022892               substrate-specific transporter activity
GO:0022891 substrate-specific transmembrane transporter activity
               term_namespace term_distance                 members
GO:0019210 molecular_function             3 PF05148,PF04616,PF06452
GO:0004860 molecular_function             4 PF05148,PF04616,PF06452
GO:0004857 molecular_function             2 PF05148,PF04616,PF06452
GO:0019207 molecular_function             2 PF05148,PF04616,PF06452
GO:0019887 molecular_function             3 PF05148,PF04616,PF06452
GO:0030234 molecular_function             1 PF05148,PF04616,PF06452
GO:0005215 molecular_function             1 PF03418,PF15291,PF05388
GO:0022857 molecular_function             2 PF03418,PF15291,PF05388
GO:0022892 molecular_function             2 PF03418,PF15291,PF05388
GO:0022891 molecular_function             3 PF03418,PF15291,PF05388
</div>
<div class='input'>### visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 7 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<div class='input'>
###########################################################
# 3) Enrichment analysis for InterPro domains (InterPro)
## 3a) load InterPro (as 'InfoDataFrame' object)
InterPro <- dcRDataLoader('InterPro')
</div>
<strong class='message'>'InterPro' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<a href="javascript:newWin('dcEnrichment-84.png', 'dcEnrichment-84.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-84.png', width="400" /></a>
<div class='input'>### randomly select 100 domains as a list of domains of interest
data <- sample(rowNames(InterPro), 100)
## 3b) perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, domain="InterPro", ontology="GOMF")
</div>
<strong class='message'>Start at 2015-06-04 16:17:49</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'InterPro', and their associations (2015-06-04 16:17:49) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'InterPro' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'InterPro2GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2015-06-04 16:17:56) ...</strong>
<strong class='message'>	There are 630 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2015-06-04 16:17:57) ...</strong>
<strong class='message'>
End at 2015-06-04 16:17:57</strong>
<strong class='message'>Runtime in total is: 8 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'InterPro'
  @ontology: 'GOMF'
  @term_info: a data.frame of 60 terms X 5 information
  @anno: a list of 60 terms, each storing annotated domains
  @data: a vector containing a group of 71 input domains (annotatable)
  @background: a vector containing a group of 8899 background domains (annotatable)
  @overlap: a list of 60 terms, each containing domains overlapped with input domains
  @zscore: a vector of 60 terms, containing z-scores
  @pvalue: a vector of 60 terms, containing p-values
  @adjp: a vector of 60 terms, containing adjusted p-values
In summary, a total of 60 terms ('GOMF') are analysed for a group of 71 input domains ('InterPro')
</div>
<div class='input'>## 3c) view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore  pvalue  adjp
GO:0016832 GO:0016832    47     71        3   4.32 0.00051 0.031
GO:0008173 GO:0008173    86     71        3   2.82 0.00480 0.140
GO:0004494 GO:0004494   101     71        3   2.47 0.00850 0.170
GO:0032266 GO:0032266   208     71        4   1.85 0.02500 0.260
GO:1901981 GO:1901981   210     71        4   1.82 0.02600 0.260
GO:0016866 GO:0016866   154     71        3   1.62 0.03400 0.260
GO:0035091 GO:0035091   236     71        4   1.57 0.04000 0.260
GO:0016830 GO:0016830   404     71        6   1.59 0.04100 0.260
GO:0005543 GO:0005543   239     71        4   1.54 0.04200 0.260
GO:0008757 GO:0008757   166     71        3   1.48 0.04300 0.260
                                                           term_name
GO:0016832                                   aldehyde-lyase activity
GO:0008173                            RNA methyltransferase activity
GO:0004494                         methylmalonyl-CoA mutase activity
GO:0032266                  phosphatidylinositol-3-phosphate binding
GO:1901981                    phosphatidylinositol phosphate binding
GO:0016866                       intramolecular transferase activity
GO:0035091                              phosphatidylinositol binding
GO:0016830                              carbon-carbon lyase activity
GO:0005543                                      phospholipid binding
GO:0008757 S-adenosylmethionine-dependent methyltransferase activity
               term_namespace term_distance
GO:0016832 molecular_function             4
GO:0008173 molecular_function             5
GO:0004494 molecular_function             4
GO:0032266 molecular_function             7
GO:1901981 molecular_function             6
GO:0016866 molecular_function             3
GO:0035091 molecular_function             5
GO:0016830 molecular_function             3
GO:0005543 molecular_function             3
GO:0008757 molecular_function             5
                                                               members
GO:0016832                               IPR028380,IPR001711,IPR020469
GO:0008173                               IPR005869,IPR012805,IPR020097
GO:0004494                               IPR004764,IPR005292,IPR000463
GO:0032266                     IPR004930,IPR027516,IPR017705,IPR020097
GO:1901981                     IPR004930,IPR027516,IPR017705,IPR020097
GO:0016866                               IPR004764,IPR005292,IPR000463
GO:0035091                     IPR004930,IPR027516,IPR017705,IPR020097
GO:0016830 IPR004708,IPR028380,IPR007871,IPR013705,IPR001711,IPR020469
GO:0005543                     IPR004930,IPR027516,IPR017705,IPR020097
GO:0008757                               IPR005869,IPR012805,IPR020097
</div>
<div class='input'>## 3d) visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 18 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<div class='input'>## 3e) the same as above but using a customised background
### randomly select 1000 domains as background
background <- sample(rowNames(InterPro), 1000)
</div>
<a href="javascript:newWin('dcEnrichment-104.png', 'dcEnrichment-104.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-104.png', width="400" /></a>
<div class='input'>### perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, background=background, domain="InterPro",
ontology="GOMF")
</div>
<strong class='message'>Start at 2015-06-04 16:17:59</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'InterPro', and their associations (2015-06-04 16:17:59) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'InterPro' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'InterPro2GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2015-06-04 16:18:05) ...</strong>
<strong class='message'>	There are 149 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2015-06-04 16:18:05) ...</strong>
<strong class='message'>
End at 2015-06-04 16:18:05</strong>
<strong class='message'>Runtime in total is: 6 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'InterPro'
  @ontology: 'GOMF'
  @term_info: a data.frame of 58 terms X 5 information
  @anno: a list of 58 terms, each storing annotated domains
  @data: a vector containing a group of 71 input domains (annotatable)
  @background: a vector containing a group of 834 background domains (annotatable)
  @overlap: a list of 58 terms, each containing domains overlapped with input domains
  @zscore: a vector of 58 terms, containing z-scores
  @pvalue: a vector of 58 terms, containing p-values
  @adjp: a vector of 58 terms, containing adjusted p-values
In summary, a total of 58 terms ('GOMF') are analysed for a group of 71 input domains ('InterPro')
</div>
<div class='input'>### view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue adjp
GO:0008757 GO:0008757    11     71        3   2.24  0.010 0.26
GO:0008168 GO:0008168    13     71        3   1.90  0.019 0.26
GO:0016741 GO:0016741    14     71        3   1.75  0.025 0.26
GO:0004494 GO:0004494    14     71        3   1.75  0.025 0.26
GO:1901981 GO:1901981    21     71        4   1.75  0.027 0.26
GO:0032266 GO:0032266    21     71        4   1.75  0.027 0.26
GO:0005543 GO:0005543    23     71        4   1.55  0.039 0.28
GO:0035091 GO:0035091    23     71        4   1.55  0.039 0.28
GO:0016830 GO:0016830    40     71        6   1.51  0.046 0.29
GO:0004672 GO:0004672    18     71        3   1.25  0.059 0.34
                                                           term_name
GO:0008757 S-adenosylmethionine-dependent methyltransferase activity
GO:0008168                                methyltransferase activity
GO:0016741      transferase activity, transferring one-carbon groups
GO:0004494                         methylmalonyl-CoA mutase activity
GO:1901981                    phosphatidylinositol phosphate binding
GO:0032266                  phosphatidylinositol-3-phosphate binding
GO:0005543                                      phospholipid binding
GO:0035091                              phosphatidylinositol binding
GO:0016830                              carbon-carbon lyase activity
GO:0004672                                   protein kinase activity
               term_namespace term_distance
GO:0008757 molecular_function             5
GO:0008168 molecular_function             4
GO:0016741 molecular_function             3
GO:0004494 molecular_function             4
GO:1901981 molecular_function             6
GO:0032266 molecular_function             7
GO:0005543 molecular_function             3
GO:0035091 molecular_function             5
GO:0016830 molecular_function             3
GO:0004672 molecular_function             5
                                                               members
GO:0008757                               IPR012805,IPR005869,IPR020097
GO:0008168                               IPR012805,IPR005869,IPR020097
GO:0016741                               IPR012805,IPR005869,IPR020097
GO:0004494                               IPR000463,IPR004764,IPR005292
GO:1901981                     IPR004930,IPR017705,IPR027516,IPR020097
GO:0032266                     IPR004930,IPR017705,IPR027516,IPR020097
GO:0005543                     IPR004930,IPR017705,IPR027516,IPR020097
GO:0035091                     IPR004930,IPR017705,IPR027516,IPR020097
GO:0016830 IPR028380,IPR007871,IPR020469,IPR001711,IPR013705,IPR004708
GO:0004672                               IPR004768,IPR017128,IPR018150
</div>
<div class='input'>### visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 14 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<div class='input'>
###########################################################
# 4) Enrichment analysis for Rfam RNA families (Rfam)
## 4a) load Rfam (as 'InfoDataFrame' object)
Rfam <- dcRDataLoader('Rfam')
</div>
<strong class='message'>'Rfam' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<a href="javascript:newWin('dcEnrichment-125.png', 'dcEnrichment-125.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-125.png', width="400" /></a>
<div class='input'>### randomly select 100 RNAs as a list of RNAs of interest
data <- sample(rowNames(Rfam), 100)
## 4b) perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, domain="Rfam", ontology="GOBP")
</div>
<strong class='message'>Start at 2015-06-04 16:18:07</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOBP', the domain 'Rfam', and their associations (2015-06-04 16:18:07) ...</strong>
<strong class='message'>'onto.GOBP' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Rfam' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Rfam2GOBP' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2015-06-04 16:18:09) ...</strong>
<strong class='message'>	There are 114 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2015-06-04 16:18:09) ...</strong>
<strong class='message'>
End at 2015-06-04 16:18:09</strong>
<strong class='message'>Runtime in total is: 2 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'Rfam'
  @ontology: 'GOBP'
  @term_info: a data.frame of 23 terms X 5 information
  @anno: a list of 23 terms, each storing annotated domains
  @data: a vector containing a group of 51 input domains (annotatable)
  @background: a vector containing a group of 1377 background domains (annotatable)
  @overlap: a list of 23 terms, each containing domains overlapped with input domains
  @zscore: a vector of 23 terms, containing z-scores
  @pvalue: a vector of 23 terms, containing p-values
  @adjp: a vector of 23 terms, containing adjusted p-values
In summary, a total of 23 terms ('GOBP') are analysed for a group of 51 input domains ('Rfam')
</div>
<div class='input'>## 4c) view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=FALSE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap  zscore pvalue adjp
GO:0050789 GO:0050789    81     51        3  0.0000   0.35  0.7
GO:0065007 GO:0065007    85     51        3 -0.0878   0.39  0.7
GO:1901360 GO:1901360  1246     51       46 -0.0720   0.46  0.7
GO:0006725 GO:0006725  1246     51       46 -0.0720   0.46  0.7
GO:0046483 GO:0046483  1246     51       46 -0.0720   0.46  0.7
GO:0034641 GO:0034641  1247     51       46 -0.0903   0.46  0.7
GO:0006807 GO:0006807  1251     51       46 -0.1650   0.49  0.7
GO:0006396 GO:0006396  1225     51       45 -0.1690   0.50  0.7
GO:0008380 GO:0008380   542     51       19 -0.3140   0.56  0.7
GO:0000375 GO:0000375   541     51       19 -0.3030   0.56  0.7
                                                 term_name
GO:0050789                regulation of biological process
GO:0065007                           biological regulation
GO:1901360       organic cyclic compound metabolic process
GO:0006725    cellular aromatic compound metabolic process
GO:0046483                   heterocycle metabolic process
GO:0034641    cellular nitrogen compound metabolic process
GO:0006807             nitrogen compound metabolic process
GO:0006396                                  RNA processing
GO:0008380                                    RNA splicing
GO:0000375 RNA splicing, via transesterification reactions
</div>
<div class='input'>## 4d) visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOBP' containing 10 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<div class='input'>## 4e) the same as above but using a customised background
### randomly select 1000 RNAs as background
background <- sample(rowNames(Rfam), 1000)
</div>
<a href="javascript:newWin('dcEnrichment-145.png', 'dcEnrichment-145.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-145.png', width="400" /></a>
<div class='input'>### perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, background=background, domain="Rfam",
ontology="GOBP")
</div>
<strong class='message'>Start at 2015-06-04 16:18:12</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOBP', the domain 'Rfam', and their associations (2015-06-04 16:18:12) ...</strong>
<strong class='message'>'onto.GOBP' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Rfam' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Rfam2GOBP' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2015-06-04 16:18:13) ...</strong>
<strong class='message'>	There are 70 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2015-06-04 16:18:13) ...</strong>
<strong class='message'>
End at 2015-06-04 16:18:13</strong>
<strong class='message'>Runtime in total is: 1 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'Rfam'
  @ontology: 'GOBP'
  @term_info: a data.frame of 23 terms X 5 information
  @anno: a list of 23 terms, each storing annotated domains
  @data: a vector containing a group of 51 input domains (annotatable)
  @background: a vector containing a group of 644 background domains (annotatable)
  @overlap: a list of 23 terms, each containing domains overlapped with input domains
  @zscore: a vector of 23 terms, containing z-scores
  @pvalue: a vector of 23 terms, containing p-values
  @adjp: a vector of 23 terms, containing adjusted p-values
In summary, a total of 23 terms ('GOBP') are analysed for a group of 51 input domains ('Rfam')
</div>
<div class='input'>### view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=FALSE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap  zscore pvalue adjp
GO:0050789 GO:0050789    37     51        3  0.0438   0.34 0.74
GO:0065007 GO:0065007    40     51        3 -0.1010   0.39 0.74
GO:1901360 GO:1901360   580     51       46  0.0333   0.41 0.74
GO:0006725 GO:0006725   580     51       46  0.0333   0.41 0.74
GO:0046483 GO:0046483   580     51       46  0.0333   0.41 0.74
GO:0034641 GO:0034641   582     51       46 -0.0445   0.44 0.74
GO:0006807 GO:0006807   583     51       46 -0.0843   0.46 0.74
GO:0006396 GO:0006396   574     51       45 -0.2140   0.51 0.74
GO:0016070 GO:0016070   576     51       45 -0.2920   0.54 0.74
GO:0044238 GO:0044238   589     51       46 -0.3360   0.55 0.74
                                              term_name
GO:0050789             regulation of biological process
GO:0065007                        biological regulation
GO:1901360    organic cyclic compound metabolic process
GO:0006725 cellular aromatic compound metabolic process
GO:0046483                heterocycle metabolic process
GO:0034641 cellular nitrogen compound metabolic process
GO:0006807          nitrogen compound metabolic process
GO:0006396                               RNA processing
GO:0016070                        RNA metabolic process
GO:0044238                    primary metabolic process
</div>
<div class='input'>### visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOBP' containing 10 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<div class='input'>
###########################################################
# 5) Advanced usage: customised data for domain, ontology and annotations
# 5a) create domain, ontology and annotations
## for domain
domain <-
dcBuildInfoDataFrame(input.file="http://dcgor.r-forge.r-project.org/data/InterPro/InterPro.txt",
output.file="domain.RData")
</div>
<strong class='message'>An object of S4 class 'InfoDataFrame' has been built and saved into '/Users/hfang/Sites/SUPERFAMILY/dcGOR/domain.RData'.</strong>
<a href="javascript:newWin('dcEnrichment-166.png', 'dcEnrichment-166.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-166.png', width="400" /></a>
<div class='input'>## for ontology
dcBuildOnto(relations.file="http://dcgor.r-forge.r-project.org/data/onto/igraph_GOMF_edges.txt",
nodes.file="http://dcgor.r-forge.r-project.org/data/onto/igraph_GOMF_nodes.txt",
output.file="ontology.RData")
</div>
<strong class='message'>An object of S4 class 'Onto' has been built and saved into '/Users/hfang/Sites/SUPERFAMILY/dcGOR/ontology.RData'.</strong>
<div class='input'>## for annotations
dcBuildAnno(domain_info.file="http://dcgor.r-forge.r-project.org/data/InterPro/InterPro.txt",
term_info.file="http://dcgor.r-forge.r-project.org/data/InterPro/GO.txt",
association.file="http://dcgor.r-forge.r-project.org/data/InterPro/Domain2GOMF.txt",
output.file="annotations.RData")
</div>
<strong class='message'>An object of S4 class 'Anno' has been built and saved into '/Users/hfang/Sites/SUPERFAMILY/dcGOR/annotations.RData'.</strong>
<div class='input'>## 5b) prepare data and background
### randomly select 100 domains as a list of domains of interest
data <- sample(rowNames(domain), 100)
### randomly select 1000 domains as background
background <- sample(rowNames(domain), 1000)
## 5c) perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, background=background,
domain.RData='domain.RData', ontology.RData='ontology.RData',
annotations.RData='annotations.RData')
</div>
<strong class='message'>Start at 2015-06-04 16:18:20</strong>
<strong class='message'></strong>
<strong class='message'>First, load customised ontology 'ontology.RData', the domain 'domain.RData', and their associations 'annotations.RData' (2015-06-04 16:18:20)...</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2015-06-04 16:18:25) ...</strong>
<strong class='message'>	There are 130 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2015-06-04 16:18:25) ...</strong>
<strong class='message'>
End at 2015-06-04 16:18:25</strong>
<strong class='message'>Runtime in total is: 5 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'domain.RData'
  @ontology: 'ontology.RData'
  @term_info: a data.frame of 72 terms X 5 information
  @anno: a list of 72 terms, each storing annotated domains
  @data: a vector containing a group of 81 input domains (annotatable)
  @background: a vector containing a group of 818 background domains (annotatable)
  @overlap: a list of 72 terms, each containing domains overlapped with input domains
  @zscore: a vector of 72 terms, containing z-scores
  @pvalue: a vector of 72 terms, containing p-values
  @adjp: a vector of 72 terms, containing adjusted p-values
In summary, a total of 72 terms ('ontology.RData') are analysed for a group of 81 input domains ('domain.RData')
</div>
<div class='input'>## 5d) view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue adjp
GO:0016829 GO:0016829    66     81       12   2.35 0.0088 0.21
GO:0030515 GO:0030515    10     81        3   2.14 0.0120 0.21
GO:1901363 GO:1901363    28     81        6   2.08 0.0150 0.21
GO:0016835 GO:0016835    16     81        4   2.04 0.0150 0.21
GO:0003723 GO:0003723    16     81        4   2.04 0.0150 0.21
GO:0008092 GO:0008092    11     81        3   1.94 0.0170 0.21
GO:0097159 GO:0097159    30     81        6   1.89 0.0220 0.21
GO:0016651 GO:0016651    12     81        3   1.76 0.0240 0.21
GO:0016830 GO:0016830    45     81        8   1.82 0.0260 0.21
GO:0003676 GO:0003676    19     81        4   1.65 0.0320 0.21
                                            term_name     term_namespace
GO:0016829                             lyase activity molecular_function
GO:0030515                             snoRNA binding molecular_function
GO:1901363              heterocyclic compound binding molecular_function
GO:0016835               carbon-oxygen lyase activity molecular_function
GO:0003723                                RNA binding molecular_function
GO:0008092               cytoskeletal protein binding molecular_function
GO:0097159            organic cyclic compound binding molecular_function
GO:0016651 oxidoreductase activity, acting on NAD(P)H molecular_function
GO:0016830               carbon-carbon lyase activity molecular_function
GO:0003676                       nucleic acid binding molecular_function
           term_distance
GO:0016829             2
GO:0030515             5
GO:1901363             2
GO:0016835             3
GO:0003723             4
GO:0008092             3
GO:0097159             2
GO:0016651             3
GO:0016830             3
GO:0003676             3
                                                                                                                           members
GO:0016829 IPR007072,IPR006324,IPR002401,IPR019614,IPR000807,IPR015555,IPR011808,IPR002002,IPR000878,IPR011888,IPR023530,IPR017726
GO:0030515                                                                                           IPR018028,IPR002401,IPR023530
GO:1901363                                                             IPR018028,IPR002401,IPR008128,IPR008698,IPR019015,IPR023530
GO:0016835                                                                                 IPR006324,IPR000807,IPR002002,IPR011888
GO:0003723                                                                                 IPR018028,IPR002401,IPR008698,IPR023530
GO:0008092                                                                                           IPR006324,IPR000321,IPR014307
GO:0097159                                                             IPR018028,IPR002401,IPR008128,IPR008698,IPR019015,IPR023530
GO:0016651                                                                                           IPR006886,IPR006443,IPR007083
GO:0016830                                         IPR007072,IPR002401,IPR019614,IPR011808,IPR000878,IPR011888,IPR023530,IPR017726
GO:0003676                                                                                 IPR018028,IPR002401,IPR008698,IPR023530
</div>
<div class='input'>### visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'ontology.RData' containing 10 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong></pre>
  </div>
  
  <div class="span4">
    
    <!-- 
    <ul>
      <li>dcEnrichment</li>
    </ul>
    <ul>
      
    </ul> 
    -->
    
    <h2>Source code</h2>
    <a href="javascript:newWin('R/dcEnrichment.r', 'R/dcEnrichment.r', '1200', '1200')" title="Click to view"><B><code>dcEnrichment.r</code></B></a>
    
    <h2>Source man</h2>
    <a href="javascript:newWin('man/dcEnrichment.txt', 'man/dcEnrichment.txt', '1200', '1200')" title="Click to view"><B><code>dcEnrichment.Rd</code></B></a>
    
    <a href="javascript:newWin('man/dcEnrichment.pdf', 'man/dcEnrichment.pdf', '1200', '1200')" title="Click to view"><B><code>dcEnrichment.pdf</code></B></a>
    
    <h2>See also</h2>
    
<code><a href='dcRDataLoader.html'>dcRDataLoader</a></code>, <code><a href='dcDAGannotate.html'>dcDAGannotate</a></code>,
<code><a href='Eoutput-class.html'>Eoutput-class</a></code>, <code><a href='visEnrichment.html'>visEnrichment</a></code>,
<code><a href='dcConverter.html'>dcConverter</a></code>

    <!--

    <h2>Citation</h2>
    <p>Fang H. (2014) <B>dcGOR: an R package for analysing ontologies and protein domain annotations</B>. <I>PLoS Computational Biology</I> 10(10):e1003929. <a href="http://dx.doi.org/10.1371/journal.pcbi.1003929">doi:10.1371/journal.pcbi.1003929</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25356683">25356683</a></p>

<p>You may also cite the package as:</p>

<ul>
<li><code>database</code></li>
</ul>

<p>Fang H, Gough J. (2013) <B>dcGO: database of domain-centric ontologies on functions, phenotypes, diseases and more</B>. <I>Nucleic Acids Research</I> 41(Database issue):D536-44. <a href="http://dx.doi.org/10.1093/nar/gks1080">doi:10.1093/nar/gks1080</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23161684">23161684</a></p>

<ul>
<li><code>web link/URL</code></li>
</ul>

<p>Package &ldquo;dcGOR&rdquo; [<a href="http://cran.r-project.org/package=dcGOR">http://cran.r-project.org/package=dcGOR</a>]</p>

    
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://supfam.org/dcGOR">http://supfam.org/dcGOR</a></li>
      <li><a href=" http://dcgor.r-forge.r-project.org"> http://dcgor.r-forge.r-project.org</a></li>
      <li><a href="
http://cran.r-project.org/package=dcGOR">
http://cran.r-project.org/package=dcGOR</a></li>
      <li><a href="
https://github.com/hfang-bristol/dcGOR">
https://github.com/hfang-bristol/dcGOR</a></li>
    </ul>
    
    
    -->
    
  </div>
  
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@well.ox.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>