<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>dcEnrichment @ dcGOR 1.0.4</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dcGOR 1.0.4</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
       
        <!-- HTML for GOOLE SEARCH BAR 
        <ul class="nav">
            <div id="search-wrapper">
                <form id="search" method="get" action="http://www.google.com/custom">
                    <div class="search-box">
                        <i class="icon-search icon-white"></i>
                        <input type="text" id="tfq" class="textinput" name="q" size="21" maxlength="120" value="Google Search"/>
                    </div>
                </form>
            </div>
        </ul>
        -->
        
      </div>
    </div>
  </div>
</div>

<!-- CSS styles for standard search box with placeholder text-->
<style type="text/css">
    #search-wrapper {
        width:300px;
        height:40px;
        background-color:transparent;
        margin:0px auto 0;
        border:0px solid #ff0000;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

    #search {
        background-color:transparent;
        padding:0px;
        height:40px;
        overflow:hidden;
    }

    .search-box {
        width:250px;
        height:32px;
        background-color:transparent;
        margin:2px 7px;
        border:0px solid #cfcfcf;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

	.textinput{
		margin: 0;
		padding: 5px 15px;
		font-family: Arial, Helvetica, sans-serif;
		font-size:14px;
		color:#999999;
		border:1px solid #f00; border-right:0px;
		border-top-left-radius: 5px 5px;
		border-bottom-left-radius: 5px 5px;
	}
</style>

<!-- JAVASCRIPT to clear search text when the field is clicked -->
<script type="text/javascript">
window.onload = function(){
	//Get submit button
	var submitbutton = document.getElementById("tfq");
	//Add listener to submit button
	if(submitbutton.addEventListener){
		submitbutton.addEventListener("click", function() {
			if (submitbutton.value == 'Google Search'){//Customize this text string to whatever you want
				submitbutton.value = 'dcGOR R ';
			}
		});
	}
}
</script>

<script>
  (function() {
    var cx = '012094530736424854714:k8ieyz4ag4a';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') +
        '//www.google.com/cse/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>

<div><gcse:search></gcse:search></div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Function to conduct ontology enrichment analysis given a group of domains</h2>

<div class="row">
  <div class="span8">
  
    <h3>Description</h3>
    <p><code>dcEnrichment</code> is supposed to conduct enrichment analysis for an
input group of domains using a specified ontology. It returns an object
of S4 class "Eoutput". Enrichment analysis is based on either Fisher's
exact test or Hypergeometric test. The test can respect the hierarchy
of the ontology. The user can customise the background domains;
otherwise, the function will use all annotatable domains as the test
background</p>
  
    <h3>Usage</h3>
    <pre>dcEnrichment(data, background = NULL, domain = c(NA, "SCOP.sf", "SCOP.fa", "Pfam", 
  "InterPro", "Rfam"), ontology = c(NA, "GOBP", "GOMF", "GOCC", "DO", "HPPA", "HPMI", 
      "HPON", "MP", "EC", "KW", "UP"), sizeRange = c(10, 1000), min.overlap = 3, which_distance = NULL, 
      test = c("HypergeoTest", "FisherTest", "BinomialTest"), p.adjust.method = c("BH", 
          "BY", "bonferroni", "holm", "hochberg", "hommel"), ontology.algorithm = c("none", 
          "pc", "elim", "lea"), elim.pvalue = 0.01, lea.depth = 2, verbose = T, domain.RData = NULL, 
      ontology.RData = NULL, annotations.RData = NULL, RData.location = "http://dcgor.r-forge.r-project.org/data")</pre>
    
    <h3>Arguments</h3>
    <dl>
      <dt>data</dt>
      <dd>an input vector. It contains id for a list of domains, for
example, sunids for SCOP domains</dd>
      <dt>background</dt>
      <dd>a background vector. It contains id for a list of
background domains, for example, sunids for SCOP domains. If NULL, by
default all annotatable domains are used as background</dd>
      <dt>domain</dt>
      <dd>the domain identity. It can be one of 'SCOP.sf' for SCOP
superfamilies, 'SCOP.fa' for SCOP families, 'Pfam' for Pfam domains,
'InterPro' for InterPro domains, 'Rfam' for Rfam RNA families</dd>
      <dt>ontology</dt>
      <dd>the ontology identity. It can be "GOBP" for Gene
Ontology Biological Process, "GOMF" for Gene Ontology Molecular
Function, "GOCC" for Gene Ontology Cellular Component, "DO" for Disease
Ontology, "HPPA" for Human Phenotype Phenotypic Abnormality, "HPMI" for
Human Phenotype Mode of Inheritance, "HPON" for Human Phenotype ONset
and clinical course, "MP" for Mammalian Phenotype, "EC" for Enzyme
Commission, "KW" for UniProtKB KeyWords, "UP" for UniProtKB UniPathway.
For details on the eligibility for pairs of input domain and ontology,
please refer to the online Documentations at
<a href = 'http://supfam.org/dcGOR/docs.html'>http://supfam.org/dcGOR/docs.html</a></dd>
      <dt>sizeRange</dt>
      <dd>the minimum and maximum size of members of each term
in consideration. By default, it sets to a minimum of 10 but no more
than 1000</dd>
      <dt>min.overlap</dt>
      <dd>the minimum number of overlaps. Only those terms
that overlap with input data at least min.overlap (3 domains by
default) will be processed</dd>
      <dt>which_distance</dt>
      <dd>which distance of terms in the ontology is used
to restrict terms in consideration. By default, it sets to 'NULL' to
consider all distances</dd>
      <dt>test</dt>
      <dd>the statistic test used. It can be "FisherTest" for using
fisher's exact test, "HypergeoTest" for using hypergeometric test, or
"BinomialTest" for using binomial test. Fisher's exact test is to test
the independence between domain group (domains belonging to a group or
not) and domain annotation (domains annotated by a term or not), and
thus compare sampling to the left part of background (after sampling
without replacement). Hypergeometric test is to sample at random
(without replacement) from the background containing annotated and
non-annotated domains, and thus compare sampling to background. Unlike
hypergeometric test, binomial test is to sample at random (with
replacement) from the background with the constant probability. In
terms of the ease of finding the significance, they are in order:
hypergeometric test > binomial test > fisher's exact test. In other
words, in terms of the calculated p-value, hypergeometric test <
binomial test < fisher's exact test</dd>
      <dt>p.adjust.method</dt>
      <dd>the method used to adjust p-values. It can be
one of "BH", "BY", "bonferroni", "holm", "hochberg" and "hommel". The
first two methods "BH" (widely used) and "BY" control the false
discovery rate (FDR: the expected proportion of false discoveries
amongst the rejected hypotheses); the last four methods "bonferroni",
"holm", "hochberg" and "hommel" are designed to give strong control of
the family-wise error rate (FWER). Notes: FDR is a less stringent
condition than FWER</dd>
      <dt>ontology.algorithm</dt>
      <dd>the algorithm used to account for the
hierarchy of the ontology. It can be one of "none", "pc", "elim" and
"lea". For details, please see 'Note'</dd>
      <dt>elim.pvalue</dt>
      <dd>the parameter only used when "ontology.algorithm" is
"elim". It is used to control how to declare a signficantly enriched
term (and subsequently all domains in this term are eliminated from all
its ancestors)</dd>
      <dt>lea.depth</dt>
      <dd>the parameter only used when "ontology.algorithm" is
"lea". It is used to control how many maximum depth is uded to consider
the children of a term (and subsequently all domains in these children
term are eliminated from the use for the recalculation of the
signifance at this term)</dd>
      <dt>verbose</dt>
      <dd>logical to indicate whether the messages will be
displayed in the screen. By default, it sets to TRUE for display</dd>
      <dt>domain.RData</dt>
      <dd>a file name for RData-formatted file containing an
object of S4 class 'InfoDataFrame' (i.g. domain). By default, it is
NULL. It is only needed when the user wants to customise enrichment
analysis using their own data</dd>
      <dt>ontology.RData</dt>
      <dd>a file name for RData-formatted file containing
an object of S4 class 'Onto' (i.g. ontology). By default, it is NULL.
It is only needed when the user wants to customise enrichment analysis
using their own data</dd>
      <dt>annotations.RData</dt>
      <dd>a file name for RData-formatted file
containing an object of S4 class 'Anno' (i.g. annotations). By default,
it is NULL. It is only needed when the user wants to customise
enrichment analysis using their own data</dd>
      <dt>RData.location</dt>
      <dd>the characters to tell the location of built-in
RData files. By default, it remotely locates at
"http://supfam.org/dcGOR/data" or
"http://dcgor.r-forge.r-project.org/data". For the user equipped with
fast internet connection, this option can be just left as default. But
it is always advisable to download these files locally. Especially when
the user needs to run this function many times, there is no need to ask
the function to remotely download every time (also it will
unnecessarily increase the runtime). For examples, these files (as a
whole or part of them) can be first downloaded into your current
working directory, and then set this option as:
<code class = 'eq'>RData.location="."</code>. If RData to load is already part of package
itself, this parameter can be ignored (since this function will try to
load it via function <code>data</code> first)</dd>
    </dl>
    <!-- 
    -->
    
    <!-- 
    <div class="Value">
    -->
      <h3>Value</h3>
      
      <p>an object of S4 class <code><a href='Eoutput-class.html'>Eoutput</a></code>, with following slots:
<ul>
<li><code>domain</code>: a character specifying the domain identity
</li>
<li><code>ontology</code>: a character specifying the ontology used
</li>
<li><code>term_info</code>: a matrix of nTerm X 5 containing term
information, where nTerm is the number of terms in consideration, and
the 5 columns are "term_id" (i.e. "Term ID"), "term_name" (i.e. "Term
Name"), "namespace" (i.e. "Term Namespace"), "distance" (i.e. "Term
Distance") and "IC" (i.e. "Information Content for the term based on
annotation frequency by it")
</li>
<li><code>anno</code>: a list of terms, each storing annotated domain
members (also within the background domains). Always, terms are
identified by "term_id" and domain members identified by their ids
(e.g. sunids for SCOP domains)
</li>
<li><code>data</code>: a vector containing input data in consideration. It
is not always the same as the input data as only those mappable and
annotatable are retained
</li>
<li><code>background</code>: a vector containing background in
consideration. It is not always the same as the input background as
only those mappable/annotatable are retained
</li>
<li><code>overlap</code>: a list of terms, each storing domains overlapped
between domains annotated by a term and domains in the input data (i.e.
the domains of interest). Always, terms are identified by "term_id" and
domain members identified by their IDs (e.g. sunids for SCOP domains)
</li>
<li><code>zscore</code>: a vector containing z-scores
</li>
<li><code>pvalue</code>: a vector containing p-values
</li>
<li><code>adjp</code>: a vector containing adjusted p-values. It is the p
value but after being adjusted for multiple comparisons
</li>
</ul></p>
      <p></p>
    <!-- 
    </div>
    -->
    <!-- 
    <div class="Note">
    -->
      <h3>Note</h3>
      
      <p>The interpretation of the algorithms used to account for the hierarchy
of the ontology is:
<ul>
<li>"none": does not consider the ontology hierarchy at all.
</li>
<li>"lea": computers the significance of a term in terms of the
significance of its children at the maximum depth (e.g. 2). Precisely,
once domains are already annotated to any children terms with a more
signficance than itself, then all these domains are eliminated from the
use for the recalculation of the signifance at that term. The final
p-values takes the maximum of the original p-value and the recalculated
p-value.
</li>
<li>"elim": computers the significance of a term in terms of the
significance of its all children. Precisely, once domains are already
annotated to a signficantly enriched term under the cutoff of e.g.
pvalue<1e-2, all these domains are eliminated from the ancestors of
that term).
</li>
<li>"pc": requires the significance of a term not only using the
whole domains as background but also using domains annotated to all its
direct parents/ancestors as background. The final p-value takes the
maximum of both p-values in these two calculations.
</li>
<li>"Notes": the order of the number of significant terms is: "none"
> "lea" > "elim" > "pc".
</li>
</ul></p>
      <p></p>
    <!-- 
    </div>
    -->
    
    <h3 id="examples">Examples</h3>
    <pre class="examples"><div class='input'># 1) Enrichment analysis for SCOP domain superfamilies (sf)
## 1a) load SCOP.sf (as 'InfoDataFrame' object)
SCOP.sf <- dcRDataLoader('SCOP.sf')
</div>
<strong class='message'>'SCOP.sf' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<div class='input'>### randomly select 50 domains as a list of domains of interest
data <- sample(rowNames(SCOP.sf), 50)
## 1b) perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, domain="SCOP.sf", ontology="GOMF")
</div>
<strong class='message'>Start at 2014-10-24 19:44:54</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'SCOP.sf', and their associations (2014-10-24 19:44:54) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf2GOMF' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2014-10-24 19:44:59) ...</strong>
<strong class='message'>	There are 811 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-10-24 19:45:00) ...</strong>
<strong class='message'>
End at 2014-10-24 19:45:00</strong>
<strong class='message'>Runtime in total is: 6 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'SCOP.sf'
  @ontology: 'GOMF'
  @term_info: a data.frame of 152 terms X 5 information
  @anno: a list of 152 terms, each storing annotated domains
  @data: a vector containing a group of 25 input domains (annotatable)
  @background: a vector containing a group of 1083 background domains (annotatable)
  @overlap: a list of 152 terms, each containing domains overlapped with input domains
  @zscore: a vector of 152 terms, containing z-scores
  @pvalue: a vector of 152 terms, containing p-values
  @adjp: a vector of 152 terms, containing adjusted p-values
In summary, a total of 152 terms ('GOMF') are analysed for a group of 25 input domains ('SCOP.sf')
</div>
<div class='input'>## 1c) view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue adjp
GO:0019199 GO:0019199    28     25        3   3.00 0.0031 0.47
GO:0004713 GO:0004713    55     25        3   1.59 0.0340 0.78
GO:0046915 GO:0046915   124     25        5   1.36 0.0570 0.78
GO:0019200 GO:0019200   161     25        6   1.30 0.0650 0.78
GO:0003729 GO:0003729   129     25        5   1.26 0.0670 0.78
GO:0008525 GO:0008525    70     25        3   1.14 0.0720 0.78
GO:0004668 GO:0004668    71     25        3   1.11 0.0750 0.78
GO:0015651 GO:0015651    71     25        3   1.11 0.0750 0.78
GO:0035671 GO:0035671    72     25        3   1.09 0.0780 0.78
GO:0004655 GO:0004655   135     25        5   1.15 0.0800 0.78
                                                              term_name
GO:0019199               transmembrane receptor protein kinase activity
GO:0004713                             protein tyrosine kinase activity
GO:0046915      transition metal ion transmembrane transporter activity
GO:0019200                                 carbohydrate kinase activity
GO:0003729                                                 mRNA binding
GO:0008525                     phosphatidylcholine transporter activity
GO:0004668                          protein-arginine deiminase activity
GO:0015651 quaternary ammonium group transmembrane transporter activity
GO:0035671                                     enone reductase activity
GO:0004655                            porphobilinogen synthase activity
               term_namespace term_distance
GO:0019199 molecular_function             5
GO:0004713 molecular_function             6
GO:0046915 molecular_function             7
GO:0019200 molecular_function             5
GO:0003729 molecular_function             5
GO:0008525 molecular_function             4
GO:0004668 molecular_function             5
GO:0015651 molecular_function             7
GO:0035671 molecular_function             5
GO:0004655 molecular_function             5
                                        members
GO:0019199                    63825,52283,81324
GO:0004713                    52283,81324,81340
GO:0046915      111331,103491,52283,81324,81340
GO:0019200 111331,48613,55136,52283,81271,81324
GO:0003729       111331,56214,52283,81271,81324
GO:0008525                   111331,81271,81324
GO:0004668                   110035,63825,81324
GO:0015651                   111331,81271,81324
GO:0035671                   111331,81271,81324
GO:0004655       81383,111331,52283,81271,81324
</div>
<div class='input'>## 1d) visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 27 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-21.png', 'dcEnrichment-21.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-21.png', width="400" /></a>
<div class='input'>## 1e) the same as above but using a customised background
### randomly select 500 domains as background
background <- sample(rowNames(SCOP.sf), 500)
### perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, background=background, domain="SCOP.sf",
ontology="GOMF")
</div>
<strong class='message'>Start at 2014-10-24 19:45:03</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'SCOP.sf', and their associations (2014-10-24 19:45:03) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf2GOMF' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2014-10-24 19:45:08) ...</strong>
<strong class='message'>	There are 403 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-10-24 19:45:08) ...</strong>
<strong class='message'>
End at 2014-10-24 19:45:08</strong>
<strong class='message'>Runtime in total is: 5 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'SCOP.sf'
  @ontology: 'GOMF'
  @term_info: a data.frame of 152 terms X 5 information
  @anno: a list of 152 terms, each storing annotated domains
  @data: a vector containing a group of 25 input domains (annotatable)
  @background: a vector containing a group of 252 background domains (annotatable)
  @overlap: a list of 152 terms, each containing domains overlapped with input domains
  @zscore: a vector of 152 terms, containing z-scores
  @pvalue: a vector of 152 terms, containing p-values
  @adjp: a vector of 152 terms, containing adjusted p-values
In summary, a total of 152 terms ('GOMF') are analysed for a group of 25 input domains ('SCOP.sf')
</div>
<div class='input'>### view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue adjp
GO:0019199 GO:0019199    10     25        3  2.160  0.011  0.8
GO:0004713 GO:0004713    13     25        3  1.630  0.029  0.8
GO:0004641 GO:0004641    16     25        3  1.220  0.060  0.8
GO:0004668 GO:0004668    16     25        3  1.220  0.060  0.8
GO:0016813 GO:0016813    17     25        3  1.100  0.074  0.8
GO:0016882 GO:0016882    17     25        3  1.100  0.074  0.8
GO:0004638 GO:0004638    17     25        3  1.100  0.074  0.8
GO:0046915 GO:0046915    32     25        5  1.150  0.077  0.8
GO:0008525 GO:0008525    18     25        3  0.992  0.088  0.8
GO:0003729 GO:0003729    33     25        5  1.080  0.088  0.8
                                                                                           term_name
GO:0019199                                            transmembrane receptor protein kinase activity
GO:0004713                                                          protein tyrosine kinase activity
GO:0004641                                   phosphoribosylformylglycinamidine cyclo-ligase activity
GO:0004668                                                       protein-arginine deiminase activity
GO:0016813 hydrolase activity, acting on carbon-nitrogen (but not peptide) bonds, in linear amidines
GO:0016882                                                                     cyclo-ligase activity
GO:0004638                                         phosphoribosylaminoimidazole carboxylase activity
GO:0046915                                   transition metal ion transmembrane transporter activity
GO:0008525                                                  phosphatidylcholine transporter activity
GO:0003729                                                                              mRNA binding
               term_namespace term_distance                         members
GO:0019199 molecular_function             5               63825,52283,81324
GO:0004713 molecular_function             6               81340,52283,81324
GO:0004641 molecular_function             5              63825,110035,81324
GO:0004668 molecular_function             5              63825,110035,81324
GO:0016813 molecular_function             4              63825,110035,81324
GO:0016882 molecular_function             4              63825,110035,81324
GO:0004638 molecular_function             5              63825,110035,81324
GO:0046915 molecular_function             7 81340,103491,52283,111331,81324
GO:0008525 molecular_function             4              81271,111331,81324
GO:0003729 molecular_function             5  81271,52283,111331,56214,81324
</div>
<div class='input'>### visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 20 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-40.png', 'dcEnrichment-40.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-40.png', width="400" /></a>
<div class='input'>
###########################################################
# 2) Enrichment analysis for Pfam domains (Pfam)
## 2a) load Pfam (as 'InfoDataFrame' object)
Pfam <- dcRDataLoader('Pfam')
</div>
<strong class='message'>'Pfam' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<div class='input'>### randomly select 100 domains as a list of domains of interest
data <- sample(rowNames(Pfam), 100)
## 2b) perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, domain="Pfam", ontology="GOMF")
</div>
<strong class='message'>Start at 2014-10-24 19:45:11</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'Pfam', and their associations (2014-10-24 19:45:11) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>'Pfam' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>'Pfam2GOMF' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2014-10-24 19:45:14) ...</strong>
<strong class='message'>	There are 334 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-10-24 19:45:14) ...</strong>
<strong class='message'>
End at 2014-10-24 19:45:14</strong>
<strong class='message'>Runtime in total is: 3 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'Pfam'
  @ontology: 'GOMF'
  @term_info: a data.frame of 8 terms X 5 information
  @anno: a list of 8 terms, each storing annotated domains
  @data: a vector containing a group of 19 input domains (annotatable)
  @background: a vector containing a group of 3359 background domains (annotatable)
  @overlap: a list of 8 terms, each containing domains overlapped with input domains
  @zscore: a vector of 8 terms, containing z-scores
  @pvalue: a vector of 8 terms, containing p-values
  @adjp: a vector of 8 terms, containing adjusted p-values
In summary, a total of 8 terms ('GOMF') are analysed for a group of 19 input domains ('Pfam')
</div>
<div class='input'>## 2c) view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue adjp
GO:0016301 GO:0016301   210     19        3  1.720  0.027 0.22
GO:0016772 GO:0016772   332     19        3  0.865  0.110 0.24
GO:0016740 GO:0016740   767     19        6  0.911  0.120 0.24
GO:0016491 GO:0016491   364     19        3  0.696  0.140 0.24
GO:0016830 GO:0016830   372     19        3  0.657  0.150 0.24
GO:0016829 GO:0016829   550     19        4  0.553  0.190 0.25
GO:0003824 GO:0003824  2241     19       13  0.158  0.350 0.40
GO:0016787 GO:0016787   690     19        3 -0.514  0.570 0.57
                                                                 term_name
GO:0016301                                                 kinase activity
GO:0016772 transferase activity, transferring phosphorus-containing groups
GO:0016740                                            transferase activity
GO:0016491                                         oxidoreductase activity
GO:0016830                                    carbon-carbon lyase activity
GO:0016829                                                  lyase activity
GO:0003824                                              catalytic activity
GO:0016787                                              hydrolase activity
               term_namespace term_distance
GO:0016301 molecular_function             4
GO:0016772 molecular_function             3
GO:0016740 molecular_function             2
GO:0016491 molecular_function             2
GO:0016830 molecular_function             3
GO:0016829 molecular_function             2
GO:0003824 molecular_function             1
GO:0016787 molecular_function             2
                                                                                                           members
GO:0016301                                                                                 PF00320,PF03854,PF10508
GO:0016772                                                                                 PF00320,PF03854,PF10508
GO:0016740                                                         PF00320,PF09259,PF06401,PF03854,PF01780,PF10508
GO:0016491                                                                                 PF00320,PF03360,PF02826
GO:0016830                                                                                 PF05793,PF04313,PF08133
GO:0016829                                                                         PF05680,PF05793,PF04313,PF08133
GO:0003824 PF00320,PF05680,PF03360,PF02826,PF09259,PF05793,PF06401,PF01195,PF04313,PF03854,PF08133,PF01780,PF10508
GO:0016787                                                                                 PF01195,PF04313,PF03854
</div>
<div class='input'>## 2d) visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 8 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-61.png', 'dcEnrichment-61.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-61.png', width="400" /></a>
<div class='input'>## 2e) the same as above but using a customised background
### randomly select 1000 domains as background
background <- sample(rowNames(Pfam), 1000)
### perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, background=background, domain="Pfam",
ontology="GOMF")
</div>
<strong class='message'>Start at 2014-10-24 19:45:16</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'Pfam', and their associations (2014-10-24 19:45:16) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>'Pfam' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>'Pfam2GOMF' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2014-10-24 19:45:18) ...</strong>
<strong class='message'>	There are 44 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-10-24 19:45:18) ...</strong>
<strong class='message'>
End at 2014-10-24 19:45:18</strong>
<strong class='message'>Runtime in total is: 2 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'Pfam'
  @ontology: 'GOMF'
  @term_info: a data.frame of 8 terms X 5 information
  @anno: a list of 8 terms, each storing annotated domains
  @data: a vector containing a group of 19 input domains (annotatable)
  @background: a vector containing a group of 259 background domains (annotatable)
  @overlap: a list of 8 terms, each containing domains overlapped with input domains
  @zscore: a vector of 8 terms, containing z-scores
  @pvalue: a vector of 8 terms, containing p-values
  @adjp: a vector of 8 terms, containing adjusted p-values
In summary, a total of 8 terms ('GOMF') are analysed for a group of 19 input domains ('Pfam')
</div>
<div class='input'>### view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue adjp
GO:0016301 GO:0016301    16     19        3  1.800  0.022 0.17
GO:0016772 GO:0016772    23     19        3  1.100  0.074 0.20
GO:0016740 GO:0016740    58     19        6  0.996  0.100 0.20
GO:0016830 GO:0016830    28     19        3  0.725  0.130 0.20
GO:0016491 GO:0016491    29     19        3  0.658  0.150 0.20
GO:0016829 GO:0016829    40     19        4  0.701  0.150 0.20
GO:0003824 GO:0003824   165     19       13  0.443  0.250 0.28
GO:0016787 GO:0016787    54     19        3 -0.563  0.590 0.59
                                                                 term_name
GO:0016301                                                 kinase activity
GO:0016772 transferase activity, transferring phosphorus-containing groups
GO:0016740                                            transferase activity
GO:0016830                                    carbon-carbon lyase activity
GO:0016491                                         oxidoreductase activity
GO:0016829                                                  lyase activity
GO:0003824                                              catalytic activity
GO:0016787                                              hydrolase activity
               term_namespace term_distance
GO:0016301 molecular_function             4
GO:0016772 molecular_function             3
GO:0016740 molecular_function             2
GO:0016830 molecular_function             3
GO:0016491 molecular_function             2
GO:0016829 molecular_function             2
GO:0003824 molecular_function             1
GO:0016787 molecular_function             2
                                                                                                           members
GO:0016301                                                                                 PF03854,PF10508,PF00320
GO:0016772                                                                                 PF03854,PF10508,PF00320
GO:0016740                                                         PF03854,PF10508,PF00320,PF01780,PF09259,PF06401
GO:0016830                                                                                 PF04313,PF05793,PF08133
GO:0016491                                                                                 PF02826,PF00320,PF03360
GO:0016829                                                                         PF04313,PF05793,PF05680,PF08133
GO:0003824 PF02826,PF03854,PF04313,PF10508,PF05793,PF00320,PF05680,PF03360,PF01780,PF01195,PF09259,PF08133,PF06401
GO:0016787                                                                                 PF03854,PF04313,PF01195
</div>
<div class='input'>### visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 8 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-80.png', 'dcEnrichment-80.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-80.png', width="400" /></a>
<div class='input'>
###########################################################
# 3) Enrichment analysis for InterPro domains (InterPro)
## 3a) load InterPro (as 'InfoDataFrame' object)
InterPro <- dcRDataLoader('InterPro')
</div>
<strong class='message'>'InterPro' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<div class='input'>### randomly select 100 domains as a list of domains of interest
data <- sample(rowNames(InterPro), 100)
## 3b) perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, domain="InterPro", ontology="GOMF")
</div>
<strong class='message'>Start at 2014-10-24 19:45:20</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'InterPro', and their associations (2014-10-24 19:45:20) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>'InterPro' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>'InterPro2GOMF' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2014-10-24 19:45:27) ...</strong>
<strong class='message'>	There are 630 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-10-24 19:45:28) ...</strong>
<strong class='message'>
End at 2014-10-24 19:45:28</strong>
<strong class='message'>Runtime in total is: 8 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'InterPro'
  @ontology: 'GOMF'
  @term_info: a data.frame of 53 terms X 5 information
  @anno: a list of 53 terms, each storing annotated domains
  @data: a vector containing a group of 74 input domains (annotatable)
  @background: a vector containing a group of 8899 background domains (annotatable)
  @overlap: a list of 53 terms, each containing domains overlapped with input domains
  @zscore: a vector of 53 terms, containing z-scores
  @pvalue: a vector of 53 terms, containing p-values
  @adjp: a vector of 53 terms, containing adjusted p-values
In summary, a total of 53 terms ('GOMF') are analysed for a group of 74 input domains ('InterPro')
</div>
<div class='input'>## 3c) view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore  pvalue  adjp
GO:0004872 GO:0004872   521     74       12   3.81 0.00032 0.017
GO:0038023 GO:0038023   509     74       11   3.40 0.00091 0.024
GO:0060089 GO:0060089   643     74       12   3.00 0.00220 0.030
GO:0004871 GO:0004871   642     74       12   3.01 0.00220 0.030
GO:0004930 GO:0004930   251     74        6   2.76 0.00470 0.050
GO:0009881 GO:0009881   140     74        4   2.66 0.00610 0.050
GO:0016746 GO:0016746   143     74        4   2.61 0.00660 0.050
GO:0004888 GO:0004888   309     74        6   2.19 0.01400 0.093
GO:0016810 GO:0016810   250     74        5   2.06 0.01700 0.097
GO:0001653 GO:0001653   125     74        3   1.94 0.02000 0.097
                                                                       term_name
GO:0004872                                                     receptor activity
GO:0038023                                           signaling receptor activity
GO:0060089                                         molecular transducer activity
GO:0004871                                            signal transducer activity
GO:0004930                                   G-protein coupled receptor activity
GO:0009881                                                photoreceptor activity
GO:0016746                        transferase activity, transferring acyl groups
GO:0004888                             transmembrane signaling receptor activity
GO:0016810 hydrolase activity, acting on carbon-nitrogen (but not peptide) bonds
GO:0001653                                             peptide receptor activity
               term_namespace term_distance
GO:0004872 molecular_function             1
GO:0038023 molecular_function             2
GO:0060089 molecular_function             1
GO:0004871 molecular_function             2
GO:0004930 molecular_function             5
GO:0009881 molecular_function             4
GO:0016746 molecular_function             3
GO:0004888 molecular_function             4
GO:0016810 molecular_function             3
GO:0001653 molecular_function             4
                                                                                                                           members
GO:0004872 IPR004834,IPR004202,IPR025655,IPR000210,IPR000207,IPR000253,IPR003892,IPR020752,IPR020777,IPR008158,IPR008288,IPR005927
GO:0038023           IPR004834,IPR004202,IPR025655,IPR000210,IPR000253,IPR003892,IPR020752,IPR020777,IPR008158,IPR008288,IPR005927
GO:0060089 IPR004834,IPR023030,IPR004202,IPR025655,IPR000210,IPR000253,IPR003892,IPR020752,IPR020777,IPR008158,IPR008288,IPR005927
GO:0004871 IPR004834,IPR023030,IPR004202,IPR025655,IPR000210,IPR000253,IPR003892,IPR020752,IPR020777,IPR008158,IPR008288,IPR005927
GO:0004930                                                             IPR004834,IPR004202,IPR020777,IPR008158,IPR008288,IPR005927
GO:0009881                                                                                 IPR025655,IPR000210,IPR000253,IPR003892
GO:0016746                                                                                 IPR004589,IPR011547,IPR010051,IPR016004
GO:0004888                                                             IPR004834,IPR004202,IPR020777,IPR008158,IPR008288,IPR005927
GO:0016810                                                                       IPR028373,IPR000579,IPR013471,IPR003610,IPR006103
GO:0001653                                                                                           IPR004202,IPR008158,IPR005927
</div>
<div class='input'>## 3d) visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 7 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-101.png', 'dcEnrichment-101.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-101.png', width="400" /></a>
<div class='input'>## 3e) the same as above but using a customised background
### randomly select 1000 domains as background
background <- sample(rowNames(InterPro), 1000)
### perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, background=background, domain="InterPro",
ontology="GOMF")
</div>
<strong class='message'>Start at 2014-10-24 19:45:29</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'InterPro', and their associations (2014-10-24 19:45:29) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>'InterPro' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>'InterPro2GOMF' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2014-10-24 19:45:35) ...</strong>
<strong class='message'>	There are 139 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-10-24 19:45:35) ...</strong>
<strong class='message'>
End at 2014-10-24 19:45:35</strong>
<strong class='message'>Runtime in total is: 6 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'InterPro'
  @ontology: 'GOMF'
  @term_info: a data.frame of 53 terms X 5 information
  @anno: a list of 53 terms, each storing annotated domains
  @data: a vector containing a group of 74 input domains (annotatable)
  @background: a vector containing a group of 842 background domains (annotatable)
  @overlap: a list of 53 terms, each containing domains overlapped with input domains
  @zscore: a vector of 53 terms, containing z-scores
  @pvalue: a vector of 53 terms, containing p-values
  @adjp: a vector of 53 terms, containing adjusted p-values
In summary, a total of 53 terms ('GOMF') are analysed for a group of 74 input domains ('InterPro')
</div>
<div class='input'>### view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore  pvalue  adjp
GO:0004872 GO:0004872    53     74       12   3.68 0.00032 0.016
GO:0038023 GO:0038023    51     74       11   3.32 0.00085 0.016
GO:0004930 GO:0004930    20     74        6   3.39 0.00093 0.016
GO:0004871 GO:0004871    66     74       12   2.81 0.00300 0.037
GO:0060089 GO:0060089    67     74       12   2.75 0.00350 0.037
GO:0004888 GO:0004888    27     74        6   2.50 0.00640 0.051
GO:0016810 GO:0016810    21     74        5   2.46 0.00700 0.051
GO:0016813 GO:0016813    10     74        3   2.38 0.00770 0.051
GO:0016746 GO:0016746    16     74        4   2.31 0.00930 0.054
GO:0001653 GO:0001653    11     74        3   2.18 0.01100 0.054
                                                                                           term_name
GO:0004872                                                                         receptor activity
GO:0038023                                                               signaling receptor activity
GO:0004930                                                       G-protein coupled receptor activity
GO:0004871                                                                signal transducer activity
GO:0060089                                                             molecular transducer activity
GO:0004888                                                 transmembrane signaling receptor activity
GO:0016810                     hydrolase activity, acting on carbon-nitrogen (but not peptide) bonds
GO:0016813 hydrolase activity, acting on carbon-nitrogen (but not peptide) bonds, in linear amidines
GO:0016746                                            transferase activity, transferring acyl groups
GO:0001653                                                                 peptide receptor activity
               term_namespace term_distance
GO:0004872 molecular_function             1
GO:0038023 molecular_function             2
GO:0004930 molecular_function             5
GO:0004871 molecular_function             2
GO:0060089 molecular_function             1
GO:0004888 molecular_function             4
GO:0016810 molecular_function             3
GO:0016813 molecular_function             4
GO:0016746 molecular_function             3
GO:0001653 molecular_function             4
                                                                                                                           members
GO:0004872 IPR008288,IPR005927,IPR004834,IPR020752,IPR025655,IPR020777,IPR004202,IPR000210,IPR003892,IPR000207,IPR008158,IPR000253
GO:0038023           IPR008288,IPR005927,IPR004834,IPR020752,IPR025655,IPR020777,IPR004202,IPR000210,IPR003892,IPR008158,IPR000253
GO:0004930                                                             IPR008288,IPR005927,IPR004834,IPR020777,IPR004202,IPR008158
GO:0004871 IPR008288,IPR005927,IPR004834,IPR020752,IPR025655,IPR020777,IPR023030,IPR004202,IPR000210,IPR003892,IPR008158,IPR000253
GO:0060089 IPR008288,IPR005927,IPR004834,IPR020752,IPR025655,IPR020777,IPR023030,IPR004202,IPR000210,IPR003892,IPR008158,IPR000253
GO:0004888                                                             IPR008288,IPR005927,IPR004834,IPR020777,IPR004202,IPR008158
GO:0016810                                                                       IPR003610,IPR028373,IPR013471,IPR006103,IPR000579
GO:0016813                                                                                           IPR003610,IPR006103,IPR000579
GO:0016746                                                                                 IPR010051,IPR016004,IPR011547,IPR004589
GO:0001653                                                                                           IPR005927,IPR004202,IPR008158
</div>
<div class='input'>### visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 7 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-120.png', 'dcEnrichment-120.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-120.png', width="400" /></a>
<div class='input'>
###########################################################
# 4) Enrichment analysis for Rfam RNA families (Rfam)
## 4a) load Rfam (as 'InfoDataFrame' object)
Rfam <- dcRDataLoader('Rfam')
</div>
<strong class='message'>'Rfam' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<div class='input'>### randomly select 100 RNAs as a list of RNAs of interest
data <- sample(rowNames(Rfam), 100)
## 4b) perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, domain="Rfam", ontology="GOBP")
</div>
<strong class='message'>Start at 2014-10-24 19:45:37</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOBP', the domain 'Rfam', and their associations (2014-10-24 19:45:37) ...</strong>
<strong class='message'>'onto.GOBP' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>'Rfam' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>'Rfam2GOBP' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2014-10-24 19:45:39) ...</strong>
<strong class='message'>	There are 114 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-10-24 19:45:39) ...</strong>
<strong class='message'>
End at 2014-10-24 19:45:39</strong>
<strong class='message'>Runtime in total is: 2 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'Rfam'
  @ontology: 'GOBP'
  @term_info: a data.frame of 27 terms X 5 information
  @anno: a list of 27 terms, each storing annotated domains
  @data: a vector containing a group of 55 input domains (annotatable)
  @background: a vector containing a group of 1377 background domains (annotatable)
  @overlap: a list of 27 terms, each containing domains overlapped with input domains
  @zscore: a vector of 27 terms, containing z-scores
  @pvalue: a vector of 27 terms, containing p-values
  @adjp: a vector of 27 terms, containing adjusted p-values
In summary, a total of 27 terms ('GOBP') are analysed for a group of 55 input domains ('Rfam')
</div>
<div class='input'>## 4c) view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=FALSE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue adjp
GO:0010468 GO:0010468    41     55        3  1.100  0.077 0.73
GO:0060255 GO:0060255    42     55        3  1.060  0.083 0.73
GO:0019222 GO:0019222    44     55        3  0.972  0.095 0.73
GO:0044699 GO:0044699    46     55        3  0.890  0.110 0.73
GO:0050789 GO:0050789    81     55        4  0.447  0.220 0.80
GO:0065007 GO:0065007    85     55        4  0.346  0.250 0.80
GO:0006139 GO:0006139  1243     55       49 -0.301  0.550 0.80
GO:1901360 GO:1901360  1246     55       49 -0.360  0.570 0.80
GO:0006725 GO:0006725  1246     55       49 -0.360  0.570 0.80
GO:0046483 GO:0046483  1246     55       49 -0.360  0.570 0.80
                                                  term_name
GO:0010468                    regulation of gene expression
GO:0060255    regulation of macromolecule metabolic process
GO:0019222                  regulation of metabolic process
GO:0044699                          single-organism process
GO:0050789                 regulation of biological process
GO:0065007                            biological regulation
GO:0006139 nucleobase-containing compound metabolic process
GO:1901360        organic cyclic compound metabolic process
GO:0006725     cellular aromatic compound metabolic process
GO:0046483                    heterocycle metabolic process
</div>
<div class='input'>## 4d) visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOBP' containing 7 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-141.png', 'dcEnrichment-141.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-141.png', width="400" /></a>
<div class='input'>## 4e) the same as above but using a customised background
### randomly select 1000 RNAs as background
background <- sample(rowNames(Rfam), 1000)
### perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, background=background, domain="Rfam",
ontology="GOBP")
</div>
<strong class='message'>Start at 2014-10-24 19:45:41</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOBP', the domain 'Rfam', and their associations (2014-10-24 19:45:41) ...</strong>
<strong class='message'>'onto.GOBP' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>'Rfam' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>'Rfam2GOBP' (from package 'dcGOR' version 1.0.4) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2014-10-24 19:45:43) ...</strong>
<strong class='message'>	There are 51 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-10-24 19:45:43) ...</strong>
<strong class='message'>
End at 2014-10-24 19:45:43</strong>
<strong class='message'>Runtime in total is: 2 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'Rfam'
  @ontology: 'GOBP'
  @term_info: a data.frame of 27 terms X 5 information
  @anno: a list of 27 terms, each storing annotated domains
  @data: a vector containing a group of 55 input domains (annotatable)
  @background: a vector containing a group of 643 background domains (annotatable)
  @overlap: a list of 27 terms, each containing domains overlapped with input domains
  @zscore: a vector of 27 terms, containing z-scores
  @pvalue: a vector of 27 terms, containing p-values
  @adjp: a vector of 27 terms, containing adjusted p-values
In summary, a total of 27 terms ('GOBP') are analysed for a group of 55 input domains ('Rfam')
</div>
<div class='input'>### view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=FALSE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue adjp
GO:0010468 GO:0010468    15     55        3  1.600  0.032 0.53
GO:0060255 GO:0060255    16     55        3  1.480  0.040 0.53
GO:0019222 GO:0019222    18     55        3  1.250  0.059 0.53
GO:0044699 GO:0044699    22     55        3  0.867  0.110 0.74
GO:0050789 GO:0050789    34     55        4  0.687  0.160 0.81
GO:0065007 GO:0065007    37     55        4  0.505  0.200 0.81
GO:0008380 GO:0008380   236     55       19 -0.347  0.580 0.81
GO:0000375 GO:0000375   236     55       19 -0.347  0.580 0.81
GO:0000376 GO:0000376   225     55       18 -0.368  0.580 0.81
GO:0000372 GO:0000372   225     55       18 -0.368  0.580 0.81
                                                                               term_name
GO:0010468                                                 regulation of gene expression
GO:0060255                                 regulation of macromolecule metabolic process
GO:0019222                                               regulation of metabolic process
GO:0044699                                                       single-organism process
GO:0050789                                              regulation of biological process
GO:0065007                                                         biological regulation
GO:0008380                                                                  RNA splicing
GO:0000375                               RNA splicing, via transesterification reactions
GO:0000376 RNA splicing, via transesterification reactions with guanosine as nucleophile
GO:0000372                                                       Group I intron splicing
</div>
<div class='input'>### visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOBP' containing 7 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-160.png', 'dcEnrichment-160.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-160.png', width="400" /></a>
<div class='input'>
###########################################################
# 5) Advanced usage: customised data for domain, ontology and annotations
# 5a) create domain, ontology and annotations
## for domain
domain <-
dcBuildInfoDataFrame(input.file="http://dcgor.r-forge.r-project.org/data/InterPro/InterPro.txt",
output.file="domain.RData")
</div>
<strong class='message'>An object of S4 class 'InfoDataFrame' has been built and saved into '/Users/hfang/Sites/SUPERFAMILY/dcGOR/domain.RData'.</strong>
<div class='input'>## for ontology
dcBuildOnto(relations.file="http://dcgor.r-forge.r-project.org/data/onto/igraph_GOMF_edges.txt",
nodes.file="http://dcgor.r-forge.r-project.org/data/onto/igraph_GOMF_nodes.txt",
output.file="ontology.RData")
</div>
<strong class='message'>An object of S4 class 'Onto' has been built and saved into '/Users/hfang/Sites/SUPERFAMILY/dcGOR/ontology.RData'.</strong>
<div class='input'>## for annotations
dcBuildAnno(domain_info.file="http://dcgor.r-forge.r-project.org/data/InterPro/InterPro.txt",
term_info.file="http://dcgor.r-forge.r-project.org/data/InterPro/GO.txt",
association.file="http://dcgor.r-forge.r-project.org/data/InterPro/Domain2GOMF.txt",
output.file="annotations.RData")
</div>
<strong class='message'>An object of S4 class 'Anno' has been built and saved into '/Users/hfang/Sites/SUPERFAMILY/dcGOR/annotations.RData'.</strong>
<div class='input'>## 5b) prepare data and background
### randomly select 100 domains as a list of domains of interest
data <- sample(rowNames(domain), 100)
### randomly select 1000 domains as background
background <- sample(rowNames(domain), 1000)
## 5c) perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, background=background,
domain.RData='domain.RData', ontology.RData='ontology.RData',
annotations.RData='annotations.RData')
</div>
<strong class='message'>Start at 2014-10-24 19:45:52</strong>
<strong class='message'></strong>
<strong class='message'>First, load customised ontology 'ontology.RData', the domain 'domain.RData', and their associations 'annotations.RData' (2014-10-24 19:45:52)...</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2014-10-24 19:45:57) ...</strong>
<strong class='message'>	There are 142 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-10-24 19:45:57) ...</strong>
<strong class='message'>
End at 2014-10-24 19:45:57</strong>
<strong class='message'>Runtime in total is: 5 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'domain.RData'
  @ontology: 'ontology.RData'
  @term_info: a data.frame of 62 terms X 5 information
  @anno: a list of 62 terms, each storing annotated domains
  @data: a vector containing a group of 76 input domains (annotatable)
  @background: a vector containing a group of 850 background domains (annotatable)
  @overlap: a list of 62 terms, each containing domains overlapped with input domains
  @zscore: a vector of 62 terms, containing z-scores
  @pvalue: a vector of 62 terms, containing p-values
  @adjp: a vector of 62 terms, containing adjusted p-values
In summary, a total of 62 terms ('ontology.RData') are analysed for a group of 76 input domains ('domain.RData')
</div>
<div class='input'>## 5d) view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue  adjp
GO:0010576 GO:0010576    11     76        4   3.21 0.0015 0.021
GO:0048551 GO:0048551    11     76        4   3.21 0.0015 0.021
GO:0008191 GO:0008191    11     76        4   3.21 0.0015 0.021
GO:0061134 GO:0061134    12     76        4   2.98 0.0024 0.021
GO:0030414 GO:0030414    12     76        4   2.98 0.0024 0.021
GO:0061135 GO:0061135    12     76        4   2.98 0.0024 0.021
GO:0004866 GO:0004866    12     76        4   2.98 0.0024 0.021
GO:0022803 GO:0022803    10     76        3   2.35 0.0082 0.046
GO:0015267 GO:0015267    10     76        3   2.35 0.0082 0.046
GO:0022838 GO:0022838    10     76        3   2.35 0.0082 0.046
                                            term_name     term_namespace
GO:0010576           metalloenzyme regulator activity molecular_function
GO:0048551           metalloenzyme inhibitor activity molecular_function
GO:0008191    metalloendopeptidase inhibitor activity molecular_function
GO:0061134               peptidase regulator activity molecular_function
GO:0030414               peptidase inhibitor activity molecular_function
GO:0061135           endopeptidase regulator activity molecular_function
GO:0004866           endopeptidase inhibitor activity molecular_function
GO:0022803 passive transmembrane transporter activity molecular_function
GO:0015267                           channel activity molecular_function
GO:0022838        substrate-specific channel activity molecular_function
           term_distance                                 members
GO:0010576             2 IPR018033,IPR018669,IPR014944,IPR007312
GO:0048551             3 IPR018033,IPR018669,IPR014944,IPR007312
GO:0008191             4 IPR018033,IPR018669,IPR014944,IPR007312
GO:0061134             2 IPR018033,IPR018669,IPR014944,IPR007312
GO:0030414             3 IPR018033,IPR018669,IPR014944,IPR007312
GO:0061135             3 IPR018033,IPR018669,IPR014944,IPR007312
GO:0004866             4 IPR018033,IPR018669,IPR014944,IPR007312
GO:0022803             3           IPR006258,IPR023668,IPR010603
GO:0015267             4           IPR006258,IPR023668,IPR010603
GO:0022838             4           IPR006258,IPR023668,IPR010603
</div>
<div class='input'>### visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'ontology.RData' containing 8 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-182.png', 'dcEnrichment-182.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-182.png', width="400" /></a></pre>
  </div>
  
  <div class="span4">
    
    <!-- 
    <ul>
      <li>dcEnrichment</li>
    </ul>
    <ul>
      
    </ul> 
    -->
    
    <h2>Source code</h2>
    <a href="javascript:newWin('R/dcEnrichment.r', 'R/dcEnrichment.r', '1200', '1200')" title="Click to view"><B><code>dcEnrichment.r</code></B></a>
    
    <h2>Source man</h2>
    <a href="javascript:newWin('man/dcEnrichment.txt', 'man/dcEnrichment.txt', '1200', '1200')" title="Click to view"><B><code>dcEnrichment.Rd</code></B></a>
    
    <a href="javascript:newWin('man/dcEnrichment.pdf', 'man/dcEnrichment.pdf', '1200', '1200')" title="Click to view"><B><code>dcEnrichment.pdf</code></B></a>
    
    <h2>See also</h2>
    
<code><a href='dcRDataLoader.html'>dcRDataLoader</a></code>, <code><a href='dcDAGannotate.html'>dcDAGannotate</a></code>,
<code><a href='Eoutput-class.html'>Eoutput-class</a></code>, <code><a href='visEnrichment.html'>visEnrichment</a></code>,
<code><a href='dcConverter.html'>dcConverter</a></code>

    <!--

    <h2>Citation</h2>
    <p>Fang H. (2014) <B>dcGOR: an R package for analysing ontologies and protein domain annotations</B>. <I>PLoS Computational Biology</I> (<a href="http://dx.doi.org/10.1371/journal.pcbi.1003929">doi:10.1371/journal.pcbi.1003929</a>)</p>

<p>You may also cite the package as:</p>

<ul>
<li><code>database</code></li>
</ul>

<p>Fang H, Gough J. (2013) dcGO: database of domain-centric ontologies on functions, phenotypes, diseases and more. <I>Nucleic Acids Research</I> 41(Database issue):D536-44 (<a href="http://dx.doi.org/10.1093/nar/gks1080">doi:10.1093/nar/gks1080</a>)</p>

<ul>
<li><code>web link/URL</code></li>
</ul>

<p>Package &ldquo;dcGOR&rdquo; [<a href="http://cran.r-project.org/package=dcGOR">http://cran.r-project.org/package=dcGOR</a>]</p>

    
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://supfam.org/dcGOR">http://supfam.org/dcGOR</a></li>
    </ul>
    
    
    -->
    
  </div>
  
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>