<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>dcEnrichment @ dcGOR 1.0.5</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dcGOR 1.0.5</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
       
        <!-- HTML for GOOLE SEARCH BAR 
        <ul class="nav">
            <div id="search-wrapper">
                <form id="search" method="get" action="http://www.google.com/custom">
                    <div class="search-box">
                        <i class="icon-search icon-white"></i>
                        <input type="text" id="tfq" class="textinput" name="q" size="21" maxlength="120" value="Google Search"/>
                    </div>
                </form>
            </div>
        </ul>
        -->
        
      </div>
    </div>
  </div>
</div>

<!-- CSS styles for standard search box with placeholder text-->
<style type="text/css">
    #search-wrapper {
        width:300px;
        height:40px;
        background-color:transparent;
        margin:0px auto 0;
        border:0px solid #ff0000;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

    #search {
        background-color:transparent;
        padding:0px;
        height:40px;
        overflow:hidden;
    }

    .search-box {
        width:250px;
        height:32px;
        background-color:transparent;
        margin:2px 7px;
        border:0px solid #cfcfcf;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

	.textinput{
		margin: 0;
		padding: 5px 15px;
		font-family: Arial, Helvetica, sans-serif;
		font-size:14px;
		color:#999999;
		border:1px solid #f00; border-right:0px;
		border-top-left-radius: 5px 5px;
		border-bottom-left-radius: 5px 5px;
	}
</style>

<!-- JAVASCRIPT to clear search text when the field is clicked -->
<script type="text/javascript">
window.onload = function(){
	//Get submit button
	var submitbutton = document.getElementById("tfq");
	//Add listener to submit button
	if(submitbutton.addEventListener){
		submitbutton.addEventListener("click", function() {
			if (submitbutton.value == 'Google Search'){//Customize this text string to whatever you want
				submitbutton.value = 'dcGOR R ';
			}
		});
	}
}
</script>

<script>
  (function() {
    var cx = '012094530736424854714:k8ieyz4ag4a';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') +
        '//www.google.com/cse/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>

<div><gcse:search></gcse:search></div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Function to conduct ontology enrichment analysis given a group of domains</h2>

<div class="row">
  <div class="span8">
  
    <h3>Description</h3>
    <p><code>dcEnrichment</code> is supposed to conduct enrichment analysis for an
input group of domains using a specified ontology. It returns an object
of S4 class "Eoutput". Enrichment analysis is based on either Fisher's
exact test or Hypergeometric test. The test can respect the hierarchy
of the ontology. The user can customise the background domains;
otherwise, the function will use all annotatable domains as the test
background</p>
  
    <h3>Usage</h3>
    <pre>dcEnrichment(data, background = NULL, domain = c(NA, "SCOP.sf", "SCOP.fa", "Pfam", 
  "InterPro", "Rfam"), ontology = c(NA, "GOBP", "GOMF", "GOCC", "DO", "HPPA", "HPMI", 
      "HPON", "MP", "EC", "KW", "UP"), sizeRange = c(10, 1000), min.overlap = 3, which_distance = NULL, 
      test = c("HypergeoTest", "FisherTest", "BinomialTest"), p.adjust.method = c("BH", 
          "BY", "bonferroni", "holm", "hochberg", "hommel"), ontology.algorithm = c("none", 
          "pc", "elim", "lea"), elim.pvalue = 0.01, lea.depth = 2, verbose = T, domain.RData = NULL, 
      ontology.RData = NULL, annotations.RData = NULL, RData.location = "http://dcgor.r-forge.r-project.org/data")</pre>
    
    <h3>Arguments</h3>
    <dl>
      <dt>data</dt>
      <dd>an input vector. It contains id for a list of domains, for
example, sunids for SCOP domains</dd>
      <dt>background</dt>
      <dd>a background vector. It contains id for a list of
background domains, for example, sunids for SCOP domains. If NULL, by
default all annotatable domains are used as background</dd>
      <dt>domain</dt>
      <dd>the domain identity. It can be one of 'SCOP.sf' for SCOP
superfamilies, 'SCOP.fa' for SCOP families, 'Pfam' for Pfam domains,
'InterPro' for InterPro domains, 'Rfam' for Rfam RNA families</dd>
      <dt>ontology</dt>
      <dd>the ontology identity. It can be "GOBP" for Gene
Ontology Biological Process, "GOMF" for Gene Ontology Molecular
Function, "GOCC" for Gene Ontology Cellular Component, "DO" for Disease
Ontology, "HPPA" for Human Phenotype Phenotypic Abnormality, "HPMI" for
Human Phenotype Mode of Inheritance, "HPON" for Human Phenotype ONset
and clinical course, "MP" for Mammalian Phenotype, "EC" for Enzyme
Commission, "KW" for UniProtKB KeyWords, "UP" for UniProtKB UniPathway.
For details on the eligibility for pairs of input domain and ontology,
please refer to the online Documentations at
<a href = 'http://supfam.org/dcGOR/docs.html'>http://supfam.org/dcGOR/docs.html</a></dd>
      <dt>sizeRange</dt>
      <dd>the minimum and maximum size of members of each term
in consideration. By default, it sets to a minimum of 10 but no more
than 1000</dd>
      <dt>min.overlap</dt>
      <dd>the minimum number of overlaps. Only those terms
that overlap with input data at least min.overlap (3 domains by
default) will be processed</dd>
      <dt>which_distance</dt>
      <dd>which distance of terms in the ontology is used
to restrict terms in consideration. By default, it sets to 'NULL' to
consider all distances</dd>
      <dt>test</dt>
      <dd>the statistic test used. It can be "FisherTest" for using
fisher's exact test, "HypergeoTest" for using hypergeometric test, or
"BinomialTest" for using binomial test. Fisher's exact test is to test
the independence between domain group (domains belonging to a group or
not) and domain annotation (domains annotated by a term or not), and
thus compare sampling to the left part of background (after sampling
without replacement). Hypergeometric test is to sample at random
(without replacement) from the background containing annotated and
non-annotated domains, and thus compare sampling to background. Unlike
hypergeometric test, binomial test is to sample at random (with
replacement) from the background with the constant probability. In
terms of the ease of finding the significance, they are in order:
hypergeometric test > binomial test > fisher's exact test. In other
words, in terms of the calculated p-value, hypergeometric test <
binomial test < fisher's exact test</dd>
      <dt>p.adjust.method</dt>
      <dd>the method used to adjust p-values. It can be
one of "BH", "BY", "bonferroni", "holm", "hochberg" and "hommel". The
first two methods "BH" (widely used) and "BY" control the false
discovery rate (FDR: the expected proportion of false discoveries
amongst the rejected hypotheses); the last four methods "bonferroni",
"holm", "hochberg" and "hommel" are designed to give strong control of
the family-wise error rate (FWER). Notes: FDR is a less stringent
condition than FWER</dd>
      <dt>ontology.algorithm</dt>
      <dd>the algorithm used to account for the
hierarchy of the ontology. It can be one of "none", "pc", "elim" and
"lea". For details, please see 'Note'</dd>
      <dt>elim.pvalue</dt>
      <dd>the parameter only used when "ontology.algorithm" is
"elim". It is used to control how to declare a signficantly enriched
term (and subsequently all domains in this term are eliminated from all
its ancestors)</dd>
      <dt>lea.depth</dt>
      <dd>the parameter only used when "ontology.algorithm" is
"lea". It is used to control how many maximum depth is uded to consider
the children of a term (and subsequently all domains in these children
term are eliminated from the use for the recalculation of the
signifance at this term)</dd>
      <dt>verbose</dt>
      <dd>logical to indicate whether the messages will be
displayed in the screen. By default, it sets to TRUE for display</dd>
      <dt>domain.RData</dt>
      <dd>a file name for RData-formatted file containing an
object of S4 class 'InfoDataFrame' (i.g. domain). By default, it is
NULL. It is only needed when the user wants to customise enrichment
analysis using their own data. See <code><a href='dcBuildInfoDataFrame.html'>dcBuildInfoDataFrame</a></code>
for how to creat this object</dd>
      <dt>ontology.RData</dt>
      <dd>a file name for RData-formatted file containing
an object of S4 class 'Onto' (i.g. ontology). By default, it is NULL.
It is only needed when the user wants to customise enrichment analysis
using their own data. See <code><a href='dcBuildOnto.html'>dcBuildOnto</a></code> for how to creat
this object</dd>
      <dt>annotations.RData</dt>
      <dd>a file name for RData-formatted file
containing an object of S4 class 'Anno' (i.g. annotations). By default,
it is NULL. It is only needed when the user wants to customise
enrichment analysis using their own data. See <code><a href='dcBuildAnno.html'>dcBuildAnno</a></code>
for how to creat this object</dd>
      <dt>RData.location</dt>
      <dd>the characters to tell the location of built-in
RData files. By default, it remotely locates at
"http://supfam.org/dcGOR/data" or
"http://dcgor.r-forge.r-project.org/data". For the user equipped with
fast internet connection, this option can be just left as default. But
it is always advisable to download these files locally. Especially when
the user needs to run this function many times, there is no need to ask
the function to remotely download every time (also it will
unnecessarily increase the runtime). For examples, these files (as a
whole or part of them) can be first downloaded into your current
working directory, and then set this option as:
<code class = 'eq'>RData.location="."</code>. If RData to load is already part of package
itself, this parameter can be ignored (since this function will try to
load it via function <code>data</code> first)</dd>
    </dl>
    <!-- 
    -->
    
    <!-- 
    <div class="Value">
    -->
      <h3>Value</h3>
      
      <p>an object of S4 class <code><a href='Eoutput-class.html'>Eoutput</a></code>, with following slots:
<ul>
<li><code>domain</code>: a character specifying the domain identity
</li>
<li><code>ontology</code>: a character specifying the ontology used
</li>
<li><code>term_info</code>: a matrix of nTerm X 5 containing term
information, where nTerm is the number of terms in consideration, and
the 5 columns are "term_id" (i.e. "Term ID"), "term_name" (i.e. "Term
Name"), "namespace" (i.e. "Term Namespace"), "distance" (i.e. "Term
Distance") and "IC" (i.e. "Information Content for the term based on
annotation frequency by it")
</li>
<li><code>anno</code>: a list of terms, each storing annotated domain
members (also within the background domains). Always, terms are
identified by "term_id" and domain members identified by their ids
(e.g. sunids for SCOP domains)
</li>
<li><code>data</code>: a vector containing input data in consideration. It
is not always the same as the input data as only those mappable and
annotatable are retained
</li>
<li><code>background</code>: a vector containing background in
consideration. It is not always the same as the input background as
only those mappable/annotatable are retained
</li>
<li><code>overlap</code>: a list of terms, each storing domains overlapped
between domains annotated by a term and domains in the input data (i.e.
the domains of interest). Always, terms are identified by "term_id" and
domain members identified by their IDs (e.g. sunids for SCOP domains)
</li>
<li><code>zscore</code>: a vector containing z-scores
</li>
<li><code>pvalue</code>: a vector containing p-values
</li>
<li><code>adjp</code>: a vector containing adjusted p-values. It is the p
value but after being adjusted for multiple comparisons
</li>
</ul></p>
      <p></p>
    <!-- 
    </div>
    -->
    <!-- 
    <div class="Note">
    -->
      <h3>Note</h3>
      
      <p>The interpretation of the algorithms used to account for the hierarchy
of the ontology is:
<ul>
<li>"none": does not consider the ontology hierarchy at all.
</li>
<li>"lea": computers the significance of a term in terms of the
significance of its children at the maximum depth (e.g. 2). Precisely,
once domains are already annotated to any children terms with a more
signficance than itself, then all these domains are eliminated from the
use for the recalculation of the signifance at that term. The final
p-values takes the maximum of the original p-value and the recalculated
p-value.
</li>
<li>"elim": computers the significance of a term in terms of the
significance of its all children. Precisely, once domains are already
annotated to a signficantly enriched term under the cutoff of e.g.
pvalue<1e-2, all these domains are eliminated from the ancestors of
that term).
</li>
<li>"pc": requires the significance of a term not only using the
whole domains as background but also using domains annotated to all its
direct parents/ancestors as background. The final p-value takes the
maximum of both p-values in these two calculations.
</li>
<li>"Notes": the order of the number of significant terms is: "none"
> "lea" > "elim" > "pc".
</li>
</ul></p>
      <p></p>
    <!-- 
    </div>
    -->
    
    <h3 id="examples">Examples</h3>
    <pre class="examples"><div class='input'># 1) Enrichment analysis for SCOP domain superfamilies (sf)
## 1a) load SCOP.sf (as 'InfoDataFrame' object)
SCOP.sf <- dcRDataLoader('SCOP.sf')
</div>
<strong class='message'>'SCOP.sf' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<div class='input'>### randomly select 50 domains as a list of domains of interest
data <- sample(rowNames(SCOP.sf), 50)
## 1b) perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, domain="SCOP.sf", ontology="GOMF")
</div>
<strong class='message'>Start at 2015-07-21 18:46:02</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'SCOP.sf', and their associations (2015-07-21 18:46:02) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf2GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2015-07-21 18:46:18) ...</strong>
<strong class='message'>	There are 811 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2015-07-21 18:46:19) ...</strong>
<strong class='message'>
End at 2015-07-21 18:46:19</strong>
<strong class='message'>Runtime in total is: 17 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'SCOP.sf'
  @ontology: 'GOMF'
  @term_info: a data.frame of 135 terms X 5 information
  @anno: a list of 135 terms, each storing annotated domains
  @data: a vector containing a group of 19 input domains (annotatable)
  @background: a vector containing a group of 1083 background domains (annotatable)
  @overlap: a list of 135 terms, each containing domains overlapped with input domains
  @zscore: a vector of 135 terms, containing z-scores
  @pvalue: a vector of 135 terms, containing p-values
  @adjp: a vector of 135 terms, containing adjusted p-values
In summary, a total of 135 terms ('GOMF') are analysed for a group of 19 input domains ('SCOP.sf')
</div>
<div class='input'>## 1c) view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue adjp
GO:0032553 GO:0032553    65     19        4   2.79 0.0040 0.29
GO:0008187 GO:0008187    45     19        3   2.56 0.0064 0.29
GO:0008266 GO:0008266    45     19        3   2.56 0.0064 0.29
GO:1900750 GO:1900750    57     19        3   2.07 0.0150 0.40
GO:0032555 GO:0032555    62     19        3   1.90 0.0200 0.40
GO:0000166 GO:0000166   100     19        4   1.79 0.0250 0.40
GO:0017076 GO:0017076    68     19        3   1.72 0.0270 0.40
GO:0042277 GO:0042277   105     19        4   1.69 0.0300 0.40
GO:0004666 GO:0004666    70     19        3   1.67 0.0300 0.40
GO:0004764 GO:0004764    71     19        3   1.64 0.0310 0.40
                                              term_name     term_namespace
GO:0032553                       ribonucleotide binding molecular_function
GO:0008187                poly-pyrimidine tract binding molecular_function
GO:0008266                          poly(U) RNA binding molecular_function
GO:1900750                         oligopeptide binding molecular_function
GO:0032555                purine ribonucleotide binding molecular_function
GO:0000166                           nucleotide binding molecular_function
GO:0017076                    purine nucleotide binding molecular_function
GO:0042277                              peptide binding molecular_function
GO:0004666 prostaglandin-endoperoxide synthase activity molecular_function
GO:0004764   shikimate 3-dehydrogenase (NADP+) activity molecular_function
           term_distance                 members
GO:0032553             5 47216,53901,46689,53300
GO:0008187             6       46689,53300,48092
GO:0008266             7       46689,53300,48092
GO:1900750             4       53901,46689,49309
GO:0032555             6       47216,46689,53300
GO:0000166             3 47216,53901,46689,53300
GO:0017076             5       47216,46689,53300
GO:0042277             3 53901,46689,49309,48092
GO:0004666             4       46689,53300,48092
GO:0004764             5      116846,54585,53300
</div>
<div class='input'>## 1d) visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 17 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-21.png', 'dcEnrichment-21.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-21.png', width="400" /></a>
<div class='input'>## 1e) the same as above but using a customised background
### randomly select 500 domains as background
background <- sample(rowNames(SCOP.sf), 500)
### perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, background=background, domain="SCOP.sf",
ontology="GOMF")
</div>
<strong class='message'>Start at 2015-07-21 18:46:29</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'SCOP.sf', and their associations (2015-07-21 18:46:29) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf2GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2015-07-21 18:46:45) ...</strong>
<strong class='message'>	There are 412 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2015-07-21 18:46:45) ...</strong>
<strong class='message'>
End at 2015-07-21 18:46:45</strong>
<strong class='message'>Runtime in total is: 16 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'SCOP.sf'
  @ontology: 'GOMF'
  @term_info: a data.frame of 135 terms X 5 information
  @anno: a list of 135 terms, each storing annotated domains
  @data: a vector containing a group of 19 input domains (annotatable)
  @background: a vector containing a group of 251 background domains (annotatable)
  @overlap: a list of 135 terms, each containing domains overlapped with input domains
  @zscore: a vector of 135 terms, containing z-scores
  @pvalue: a vector of 135 terms, containing p-values
  @adjp: a vector of 135 terms, containing adjusted p-values
In summary, a total of 135 terms ('GOMF') are analysed for a group of 19 input domains ('SCOP.sf')
</div>
<div class='input'>### view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue adjp
GO:0008187 GO:0008187    12     19        3   2.33  0.008 0.49
GO:0008266 GO:0008266    12     19        3   2.33  0.008 0.49
GO:0032553 GO:0032553    20     19        4   2.19  0.011 0.49
GO:0004666 GO:0004666    17     19        3   1.62  0.030 0.51
GO:0032555 GO:0032555    19     19        3   1.41  0.044 0.51
GO:0016831 GO:0016831    38     19        5   1.41  0.049 0.51
GO:1900750 GO:1900750    20     19        3   1.31  0.052 0.51
GO:0004638 GO:0004638    20     19        3   1.31  0.052 0.51
GO:0015103 GO:0015103    20     19        3   1.31  0.052 0.51
GO:0000166 GO:0000166    31     19        4   1.20  0.068 0.51
                                                    term_name
GO:0008187                      poly-pyrimidine tract binding
GO:0008266                                poly(U) RNA binding
GO:0032553                             ribonucleotide binding
GO:0004666       prostaglandin-endoperoxide synthase activity
GO:0032555                      purine ribonucleotide binding
GO:0016831                             carboxy-lyase activity
GO:1900750                               oligopeptide binding
GO:0004638  phosphoribosylaminoimidazole carboxylase activity
GO:0015103 inorganic anion transmembrane transporter activity
GO:0000166                                 nucleotide binding
               term_namespace term_distance                        members
GO:0008187 molecular_function             6              53300,46689,48092
GO:0008266 molecular_function             7              53300,46689,48092
GO:0032553 molecular_function             5        53300,46689,53901,47216
GO:0004666 molecular_function             4              53300,46689,48092
GO:0032555 molecular_function             6              53300,46689,47216
GO:0016831 molecular_function             4 63380,53300,46689,48092,101912
GO:1900750 molecular_function             4              49309,46689,53901
GO:0004638 molecular_function             5             46689,48092,101912
GO:0015103 molecular_function             6             53300,46689,101912
GO:0000166 molecular_function             3        53300,46689,53901,47216
</div>
<div class='input'>### visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 18 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-40.png', 'dcEnrichment-40.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-40.png', width="400" /></a>
<div class='input'>
###########################################################
# 2) Enrichment analysis for Pfam domains (Pfam)
## 2a) load Pfam (as 'InfoDataFrame' object)
Pfam <- dcRDataLoader('Pfam')
</div>
<strong class='message'>'Pfam' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<div class='input'>### randomly select 100 domains as a list of domains of interest
data <- sample(rowNames(Pfam), 100)
## 2b) perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, domain="Pfam", ontology="GOMF")
</div>
<strong class='message'>Start at 2015-07-21 18:46:59</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'Pfam', and their associations (2015-07-21 18:46:59) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Pfam' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Pfam2GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2015-07-21 18:47:07) ...</strong>
<strong class='message'>	There are 334 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2015-07-21 18:47:08) ...</strong>
<strong class='message'>
End at 2015-07-21 18:47:08</strong>
<strong class='message'>Runtime in total is: 9 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'Pfam'
  @ontology: 'GOMF'
  @term_info: a data.frame of 16 terms X 5 information
  @anno: a list of 16 terms, each storing annotated domains
  @data: a vector containing a group of 24 input domains (annotatable)
  @background: a vector containing a group of 3359 background domains (annotatable)
  @overlap: a list of 16 terms, each containing domains overlapped with input domains
  @zscore: a vector of 16 terms, containing z-scores
  @pvalue: a vector of 16 terms, containing p-values
  @adjp: a vector of 16 terms, containing adjusted p-values
In summary, a total of 16 terms ('GOMF') are analysed for a group of 24 input domains ('Pfam')
</div>
<div class='input'>## 2c) view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue  adjp
GO:0043167 GO:0043167   124     24        3  2.300  0.011 0.057
GO:0016818 GO:0016818   133     24        3  2.150  0.013 0.057
GO:0016462 GO:0016462   132     24        3  2.170  0.013 0.057
GO:0016817 GO:0016817   135     24        3  2.120  0.014 0.057
GO:0031369 GO:0031369   205     24        3  1.310  0.055 0.140
GO:0005488 GO:0005488   730     24        8  1.380  0.057 0.140
GO:0003824 GO:0003824  2241     24       19  1.300  0.059 0.140
GO:0005515 GO:0005515   445     24        5  1.100  0.087 0.170
GO:0016787 GO:0016787   690     24        7  1.050  0.100 0.180
GO:0016740 GO:0016740   767     24        7  0.742  0.160 0.230
                                                                                    term_name
GO:0043167                                                                        ion binding
GO:0016818 hydrolase activity, acting on acid anhydrides, in phosphorus-containing anhydrides
GO:0016462                                                           pyrophosphatase activity
GO:0016817                                      hydrolase activity, acting on acid anhydrides
GO:0031369                                              translation initiation factor binding
GO:0005488                                                                            binding
GO:0003824                                                                 catalytic activity
GO:0005515                                                                    protein binding
GO:0016787                                                                 hydrolase activity
GO:0016740                                                               transferase activity
               term_namespace term_distance
GO:0043167 molecular_function             2
GO:0016818 molecular_function             4
GO:0016462 molecular_function             5
GO:0016817 molecular_function             3
GO:0031369 molecular_function             3
GO:0005488 molecular_function             1
GO:0003824 molecular_function             1
GO:0005515 molecular_function             2
GO:0016787 molecular_function             2
GO:0016740 molecular_function             2
                                                                                                                                                           members
GO:0043167                                                                                                                                 PF01207,PF08015,PF03147
GO:0016818                                                                                                                                 PF01025,PF02776,PF01070
GO:0016462                                                                                                                                 PF01025,PF02776,PF01070
GO:0016817                                                                                                                                 PF01025,PF02776,PF01070
GO:0031369                                                                                                                                 PF08465,PF00288,PF03147
GO:0005488                                                                                         PF08465,PF00288,PF00389,PF08675,PF01207,PF08015,PF08273,PF03147
GO:0003824 PF00178,PF08465,PF14372,PF00389,PF01025,PF08661,PF02776,PF08675,PF06367,PF01070,PF04145,PF09497,PF01245,PF01207,PF10385,PF02302,PF08273,PF03089,PF03147
GO:0005515                                                                                                                 PF08465,PF00288,PF00389,PF08273,PF03147
GO:0016787                                                                                                 PF08465,PF01025,PF02776,PF08675,PF06367,PF01070,PF10385
GO:0016740                                                                                                 PF00178,PF01025,PF09497,PF01245,PF02302,PF08273,PF03147
</div>
<div class='input'>## 2d) visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 10 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-61.png', 'dcEnrichment-61.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-61.png', width="400" /></a>
<div class='input'>## 2e) the same as above but using a customised background
### randomly select 1000 domains as background
background <- sample(rowNames(Pfam), 1000)
### perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, background=background, domain="Pfam",
ontology="GOMF")
</div>
<strong class='message'>Start at 2015-07-21 18:47:20</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'Pfam', and their associations (2015-07-21 18:47:20) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Pfam' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Pfam2GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2015-07-21 18:47:28) ...</strong>
<strong class='message'>	There are 36 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2015-07-21 18:47:28) ...</strong>
<strong class='message'>
End at 2015-07-21 18:47:28</strong>
<strong class='message'>Runtime in total is: 8 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'Pfam'
  @ontology: 'GOMF'
  @term_info: a data.frame of 13 terms X 5 information
  @anno: a list of 13 terms, each storing annotated domains
  @data: a vector containing a group of 24 input domains (annotatable)
  @background: a vector containing a group of 249 background domains (annotatable)
  @overlap: a list of 13 terms, each containing domains overlapped with input domains
  @zscore: a vector of 13 terms, containing z-scores
  @pvalue: a vector of 13 terms, containing p-values
  @adjp: a vector of 13 terms, containing adjusted p-values
In summary, a total of 13 terms ('GOMF') are analysed for a group of 24 input domains ('Pfam')
</div>
<div class='input'>### view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue adjp
GO:0043167 GO:0043167    11     24        3  2.020  0.014 0.18
GO:0003824 GO:0003824   163     24       19  1.480  0.039 0.25
GO:0005488 GO:0005488    57     24        8  1.280  0.067 0.25
GO:0031369 GO:0031369    18     24        3  1.050  0.081 0.25
GO:0016787 GO:0016787    55     24        7  0.878  0.130 0.25
GO:0005515 GO:0005515    38     24        5  0.797  0.140 0.25
GO:0016740 GO:0016740    57     24        7  0.768  0.150 0.25
GO:0016833 GO:0016833    24     24        3  0.499  0.190 0.25
GO:0004451 GO:0004451    24     24        3  0.499  0.190 0.25
GO:0016491 GO:0016491    25     24        3  0.421  0.210 0.25
                                       term_name     term_namespace
GO:0043167                           ion binding molecular_function
GO:0003824                    catalytic activity molecular_function
GO:0005488                               binding molecular_function
GO:0031369 translation initiation factor binding molecular_function
GO:0016787                    hydrolase activity molecular_function
GO:0005515                       protein binding molecular_function
GO:0016740                  transferase activity molecular_function
GO:0016833               oxo-acid-lyase activity molecular_function
GO:0004451             isocitrate lyase activity molecular_function
GO:0016491               oxidoreductase activity molecular_function
           term_distance
GO:0043167             2
GO:0003824             1
GO:0005488             1
GO:0031369             3
GO:0016787             2
GO:0005515             2
GO:0016740             2
GO:0016833             4
GO:0004451             5
GO:0016491             2
                                                                                                                                                           members
GO:0043167                                                                                                                                 PF08015,PF01207,PF03147
GO:0003824 PF02302,PF04145,PF00178,PF08675,PF00389,PF01070,PF01025,PF06367,PF14372,PF01245,PF10385,PF03089,PF01207,PF08465,PF09497,PF08273,PF03147,PF08661,PF02776
GO:0005488                                                                                         PF00288,PF08675,PF00389,PF08015,PF01207,PF08465,PF08273,PF03147
GO:0031369                                                                                                                                 PF00288,PF08465,PF03147
GO:0016787                                                                                                 PF08675,PF01070,PF01025,PF06367,PF10385,PF08465,PF02776
GO:0005515                                                                                                                 PF00288,PF00389,PF08465,PF08273,PF03147
GO:0016740                                                                                                 PF02302,PF00178,PF01025,PF01245,PF09497,PF08273,PF03147
GO:0016833                                                                                                                                 PF14372,PF03089,PF08661
GO:0004451                                                                                                                                 PF14372,PF03089,PF08661
GO:0016491                                                                                                                                 PF00178,PF08675,PF00389
</div>
<div class='input'>### visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 7 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-80.png', 'dcEnrichment-80.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-80.png', width="400" /></a>
<div class='input'>
###########################################################
# 3) Enrichment analysis for InterPro domains (InterPro)
## 3a) load InterPro (as 'InfoDataFrame' object)
InterPro <- dcRDataLoader('InterPro')
</div>
<strong class='message'>'InterPro' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<div class='input'>### randomly select 100 domains as a list of domains of interest
data <- sample(rowNames(InterPro), 100)
## 3b) perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, domain="InterPro", ontology="GOMF")
</div>
<strong class='message'>Start at 2015-07-21 18:47:43</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'InterPro', and their associations (2015-07-21 18:47:43) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'InterPro' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'InterPro2GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2015-07-21 18:48:07) ...</strong>
<strong class='message'>	There are 630 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2015-07-21 18:48:07) ...</strong>
<strong class='message'>
End at 2015-07-21 18:48:07</strong>
<strong class='message'>Runtime in total is: 24 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'InterPro'
  @ontology: 'GOMF'
  @term_info: a data.frame of 57 terms X 5 information
  @anno: a list of 57 terms, each storing annotated domains
  @data: a vector containing a group of 68 input domains (annotatable)
  @background: a vector containing a group of 8899 background domains (annotatable)
  @overlap: a list of 57 terms, each containing domains overlapped with input domains
  @zscore: a vector of 57 terms, containing z-scores
  @pvalue: a vector of 57 terms, containing p-values
  @adjp: a vector of 57 terms, containing adjusted p-values
In summary, a total of 57 terms ('GOMF') are analysed for a group of 68 input domains ('InterPro')
</div>
<div class='input'>## 3c) view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore  pvalue  adjp
GO:0016853 GO:0016853   375     68        9   3.72 0.00051 0.015
GO:0016645 GO:0016645    49     68        3   4.32 0.00051 0.015
GO:0015631 GO:0015631    72     68        3   3.33 0.00220 0.041
GO:0016860 GO:0016860    87     68        3   2.89 0.00430 0.051
GO:0008092 GO:0008092    88     68        3   2.86 0.00450 0.051
GO:0016866 GO:0016866   154     68        4   2.64 0.00630 0.060
GO:0070569 GO:0070569   126     68        3   2.10 0.01600 0.130
GO:0015078 GO:0015078   209     68        4   1.93 0.02100 0.140
GO:0016675 GO:0016675   149     68        3   1.77 0.02700 0.140
GO:0004129 GO:0004129   149     68        3   1.77 0.02700 0.140
                                                              term_name
GO:0016853                                           isomerase activity
GO:0016645 oxidoreductase activity, acting on the CH-NH group of donors
GO:0015631                                              tubulin binding
GO:0016860                       intramolecular oxidoreductase activity
GO:0008092                                 cytoskeletal protein binding
GO:0016866                          intramolecular transferase activity
GO:0070569                                 uridylyltransferase activity
GO:0015078              hydrogen ion transmembrane transporter activity
GO:0016675    oxidoreductase activity, acting on a heme group of donors
GO:0004129                                cytochrome-c oxidase activity
               term_namespace term_distance
GO:0016853 molecular_function             2
GO:0016645 molecular_function             3
GO:0015631 molecular_function             4
GO:0016860 molecular_function             3
GO:0008092 molecular_function             3
GO:0016866 molecular_function             3
GO:0070569 molecular_function             5
GO:0015078 molecular_function             8
GO:0016675 molecular_function             3
GO:0004129 molecular_function             4
                                                                                             members
GO:0016853 IPR002969,IPR009415,IPR010592,IPR028609,IPR005395,IPR002394,IPR004491,IPR024448,IPR012720
GO:0016645                                                             IPR028747,IPR026274,IPR029568
GO:0015631                                                             IPR000026,IPR001308,IPR017679
GO:0016860                                                             IPR009415,IPR028609,IPR012720
GO:0008092                                                             IPR000026,IPR001308,IPR017679
GO:0016866                                                   IPR002969,IPR010592,IPR005395,IPR024448
GO:0070569                                                             IPR008899,IPR001114,IPR021604
GO:0015078                                                   IPR002957,IPR000662,IPR003927,IPR017679
GO:0016675                                                             IPR002957,IPR000662,IPR003927
GO:0004129                                                             IPR002957,IPR000662,IPR003927
</div>
<div class='input'>## 3d) visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 10 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-101.png', 'dcEnrichment-101.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-101.png', width="400" /></a>
<div class='input'>## 3e) the same as above but using a customised background
### randomly select 1000 domains as background
background <- sample(rowNames(InterPro), 1000)
### perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, background=background, domain="InterPro",
ontology="GOMF")
</div>
<strong class='message'>Start at 2015-07-21 18:48:19</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'InterPro', and their associations (2015-07-21 18:48:19) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'InterPro' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'InterPro2GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2015-07-21 18:48:40) ...</strong>
<strong class='message'>	There are 137 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2015-07-21 18:48:40) ...</strong>
<strong class='message'>
End at 2015-07-21 18:48:41</strong>
<strong class='message'>Runtime in total is: 22 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'InterPro'
  @ontology: 'GOMF'
  @term_info: a data.frame of 55 terms X 5 information
  @anno: a list of 55 terms, each storing annotated domains
  @data: a vector containing a group of 68 input domains (annotatable)
  @background: a vector containing a group of 834 background domains (annotatable)
  @overlap: a list of 55 terms, each containing domains overlapped with input domains
  @zscore: a vector of 55 terms, containing z-scores
  @pvalue: a vector of 55 terms, containing p-values
  @adjp: a vector of 55 terms, containing adjusted p-values
In summary, a total of 55 terms ('GOMF') are analysed for a group of 68 input domains ('InterPro')
</div>
<div class='input'>### view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore  pvalue  adjp
GO:0016853 GO:0016853    35     68        9   3.88 0.00024 0.013
GO:0016866 GO:0016866    13     68        4   3.00 0.00240 0.066
GO:0015631 GO:0015631    12     68        3   2.15 0.01200 0.200
GO:0016788 GO:0016788    42     68        7   2.07 0.01600 0.200
GO:0042578 GO:0042578    14     68        3   1.83 0.02200 0.200
GO:0070569 GO:0070569    14     68        3   1.83 0.02200 0.200
GO:0008092 GO:0008092    15     68        3   1.69 0.02800 0.220
GO:0009055 GO:0009055    25     68        4   1.45 0.04600 0.310
GO:0015078 GO:0015078    26     68        4   1.37 0.05300 0.320
GO:0016651 GO:0016651    19     68        3   1.23 0.06200 0.330
                                                 term_name     term_namespace
GO:0016853                              isomerase activity molecular_function
GO:0016866             intramolecular transferase activity molecular_function
GO:0015631                                 tubulin binding molecular_function
GO:0016788       hydrolase activity, acting on ester bonds molecular_function
GO:0042578             phosphoric ester hydrolase activity molecular_function
GO:0070569                    uridylyltransferase activity molecular_function
GO:0008092                    cytoskeletal protein binding molecular_function
GO:0009055                       electron carrier activity molecular_function
GO:0015078 hydrogen ion transmembrane transporter activity molecular_function
GO:0016651      oxidoreductase activity, acting on NAD(P)H molecular_function
           term_distance
GO:0016853             2
GO:0016866             3
GO:0015631             4
GO:0016788             3
GO:0042578             4
GO:0070569             5
GO:0008092             3
GO:0009055             1
GO:0015078             8
GO:0016651             3
                                                                                             members
GO:0016853 IPR010592,IPR005395,IPR009415,IPR028609,IPR002969,IPR024448,IPR012720,IPR004491,IPR002394
GO:0016866                                                   IPR010592,IPR005395,IPR002969,IPR024448
GO:0015631                                                             IPR001308,IPR000026,IPR017679
GO:0016788                     IPR001528,IPR002490,IPR025782,IPR017820,IPR020884,IPR001095,IPR008389
GO:0042578                                                             IPR001528,IPR017820,IPR020884
GO:0070569                                                             IPR001114,IPR021604,IPR008899
GO:0008092                                                             IPR001308,IPR000026,IPR017679
GO:0009055                                                   IPR018130,IPR005729,IPR005716,IPR023518
GO:0015078                                                   IPR000662,IPR003927,IPR017679,IPR002957
GO:0016651                                                             IPR003716,IPR003919,IPR007081
</div>
<div class='input'>### visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 11 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-120.png', 'dcEnrichment-120.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-120.png', width="400" /></a>
<div class='input'>
###########################################################
# 4) Enrichment analysis for Rfam RNA families (Rfam)
## 4a) load Rfam (as 'InfoDataFrame' object)
Rfam <- dcRDataLoader('Rfam')
</div>
<strong class='message'>'Rfam' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<div class='input'>### randomly select 100 RNAs as a list of RNAs of interest
data <- sample(rowNames(Rfam), 100)
## 4b) perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, domain="Rfam", ontology="GOBP")
</div>
<strong class='message'>Start at 2015-07-21 18:48:57</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOBP', the domain 'Rfam', and their associations (2015-07-21 18:48:57) ...</strong>
<strong class='message'>'onto.GOBP' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Rfam' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Rfam2GOBP' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2015-07-21 18:49:17) ...</strong>
<strong class='message'>	There are 114 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2015-07-21 18:49:17) ...</strong>
<strong class='message'>
End at 2015-07-21 18:49:17</strong>
<strong class='message'>Runtime in total is: 20 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'Rfam'
  @ontology: 'GOBP'
  @term_info: a data.frame of 41 terms X 5 information
  @anno: a list of 41 terms, each storing annotated domains
  @data: a vector containing a group of 66 input domains (annotatable)
  @background: a vector containing a group of 1377 background domains (annotatable)
  @overlap: a list of 41 terms, each containing domains overlapped with input domains
  @zscore: a vector of 41 terms, containing z-scores
  @pvalue: a vector of 41 terms, containing p-values
  @adjp: a vector of 41 terms, containing adjusted p-values
In summary, a total of 41 terms ('GOBP') are analysed for a group of 66 input domains ('Rfam')
</div>
<div class='input'>## 4c) view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=FALSE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue  adjp
GO:0048523 GO:0048523    11     66        3   3.50 0.0012 0.031
GO:0000377 GO:0000377    19     66        4   3.34 0.0015 0.031
GO:0000398 GO:0000398    13     66        3   3.10 0.0025 0.034
GO:0048519 GO:0048519    16     66        3   2.63 0.0057 0.039
GO:0016071 GO:0016071    17     66        3   2.50 0.0071 0.039
GO:0006397 GO:0006397    17     66        3   2.50 0.0071 0.039
GO:0031323 GO:0031323    29     66        4   2.29 0.0110 0.039
GO:0009889 GO:0009889    29     66        4   2.29 0.0110 0.039
GO:0031326 GO:0031326    29     66        4   2.29 0.0110 0.039
GO:0010556 GO:0010556    29     66        4   2.29 0.0110 0.039
                                                                                      term_name
GO:0048523                                              negative regulation of cellular process
GO:0000377 RNA splicing, via transesterification reactions with bulged adenosine as nucleophile
GO:0000398                                                       mRNA splicing, via spliceosome
GO:0048519                                            negative regulation of biological process
GO:0016071                                                               mRNA metabolic process
GO:0006397                                                                      mRNA processing
GO:0031323                                             regulation of cellular metabolic process
GO:0009889                                                   regulation of biosynthetic process
GO:0031326                                          regulation of cellular biosynthetic process
GO:0010556                                     regulation of macromolecule biosynthetic process
</div>
<div class='input'>## 4d) visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOBP' containing 21 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-141.png', 'dcEnrichment-141.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-141.png', width="400" /></a>
<div class='input'>## 4e) the same as above but using a customised background
### randomly select 1000 RNAs as background
background <- sample(rowNames(Rfam), 1000)
### perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, background=background, domain="Rfam",
ontology="GOBP")
</div>
<strong class='message'>Start at 2015-07-21 18:50:03</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOBP', the domain 'Rfam', and their associations (2015-07-21 18:50:03) ...</strong>
<strong class='message'>'onto.GOBP' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Rfam' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Rfam2GOBP' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2015-07-21 18:50:11) ...</strong>
<strong class='message'>	There are 50 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2015-07-21 18:50:11) ...</strong>
<strong class='message'>
End at 2015-07-21 18:50:11</strong>
<strong class='message'>Runtime in total is: 8 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'Rfam'
  @ontology: 'GOBP'
  @term_info: a data.frame of 35 terms X 5 information
  @anno: a list of 35 terms, each storing annotated domains
  @data: a vector containing a group of 66 input domains (annotatable)
  @background: a vector containing a group of 663 background domains (annotatable)
  @overlap: a list of 35 terms, each containing domains overlapped with input domains
  @zscore: a vector of 35 terms, containing z-scores
  @pvalue: a vector of 35 terms, containing p-values
  @adjp: a vector of 35 terms, containing adjusted p-values
In summary, a total of 35 terms ('GOBP') are analysed for a group of 66 input domains ('Rfam')
</div>
<div class='input'>### view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=FALSE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue  adjp
GO:0060255 GO:0060255    19     66        5   2.42 0.0074 0.050
GO:0010468 GO:0010468    19     66        5   2.42 0.0074 0.050
GO:0019222 GO:0019222    20     66        5   2.28 0.0098 0.050
GO:0031323 GO:0031323    15     66        4   2.19 0.0110 0.050
GO:0009889 GO:0009889    15     66        4   2.19 0.0110 0.050
GO:0031326 GO:0031326    15     66        4   2.19 0.0110 0.050
GO:0010556 GO:0010556    15     66        4   2.19 0.0110 0.050
GO:2000112 GO:2000112    15     66        4   2.19 0.0110 0.050
GO:0050794 GO:0050794    28     66        6   2.07 0.0150 0.053
GO:0080090 GO:0080090    16     66        4   2.03 0.0150 0.053
                                                           term_name
GO:0060255             regulation of macromolecule metabolic process
GO:0010468                             regulation of gene expression
GO:0019222                           regulation of metabolic process
GO:0031323                  regulation of cellular metabolic process
GO:0009889                        regulation of biosynthetic process
GO:0031326               regulation of cellular biosynthetic process
GO:0010556          regulation of macromolecule biosynthetic process
GO:2000112 regulation of cellular macromolecule biosynthetic process
GO:0050794                            regulation of cellular process
GO:0080090                   regulation of primary metabolic process
</div>
<div class='input'>### visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOBP' containing 9 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-160.png', 'dcEnrichment-160.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-160.png', width="400" /></a>
<div class='input'>
###########################################################
# 5) Advanced usage: customised data for domain, ontology and annotations
# 5a) create domain, ontology and annotations
## for domain
domain <-
dcBuildInfoDataFrame(input.file="http://dcgor.r-forge.r-project.org/data/InterPro/InterPro.txt",
output.file="domain.RData")
</div>
<strong class='message'>An object of S4 class 'InfoDataFrame' has been built and saved into '/Users/hfang/Sites/SUPERFAMILY/dcGOR/domain.RData'.</strong>
<div class='input'>## for ontology
dcBuildOnto(relations.file="http://dcgor.r-forge.r-project.org/data/onto/igraph_GOMF_edges.txt",
nodes.file="http://dcgor.r-forge.r-project.org/data/onto/igraph_GOMF_nodes.txt",
output.file="ontology.RData")
</div>
<strong class='message'>An object of S4 class 'Onto' has been built and saved into '/Users/hfang/Sites/SUPERFAMILY/dcGOR/ontology.RData'.</strong>
<div class='input'>## for annotations
dcBuildAnno(domain_info.file="http://dcgor.r-forge.r-project.org/data/InterPro/InterPro.txt",
term_info.file="http://dcgor.r-forge.r-project.org/data/InterPro/GO.txt",
association.file="http://dcgor.r-forge.r-project.org/data/InterPro/Domain2GOMF.txt",
output.file="annotations.RData")
</div>
<strong class='message'>An object of S4 class 'Anno' has been built and saved into '/Users/hfang/Sites/SUPERFAMILY/dcGOR/annotations.RData'.</strong>
<div class='input'>## 5b) prepare data and background
### randomly select 100 domains as a list of domains of interest
data <- sample(rowNames(domain), 100)
### randomly select 1000 domains as background
background <- sample(rowNames(domain), 1000)
## 5c) perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, background=background,
domain.RData='domain.RData', ontology.RData='ontology.RData',
annotations.RData='annotations.RData')
</div>
<strong class='message'>Start at 2015-07-21 18:51:00</strong>
<strong class='message'></strong>
<strong class='message'>First, load customised ontology 'ontology.RData', the domain 'domain.RData', and their associations 'annotations.RData' (2015-07-21 18:51:00)...</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2015-07-21 18:51:22) ...</strong>
<strong class='message'>	There are 141 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2015-07-21 18:51:22) ...</strong>
<strong class='message'>
End at 2015-07-21 18:51:22</strong>
<strong class='message'>Runtime in total is: 22 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'domain.RData'
  @ontology: 'ontology.RData'
  @term_info: a data.frame of 70 terms X 5 information
  @anno: a list of 70 terms, each storing annotated domains
  @data: a vector containing a group of 78 input domains (annotatable)
  @background: a vector containing a group of 837 background domains (annotatable)
  @overlap: a list of 70 terms, each containing domains overlapped with input domains
  @zscore: a vector of 70 terms, containing z-scores
  @pvalue: a vector of 70 terms, containing p-values
  @adjp: a vector of 70 terms, containing adjusted p-values
In summary, a total of 70 terms ('ontology.RData') are analysed for a group of 78 input domains ('domain.RData')
</div>
<div class='input'>## 5d) view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue adjp
GO:0016740 GO:0016740   145     78       22   2.67 0.0037 0.26
GO:0008168 GO:0008168    12     78        3   1.88 0.0190 0.46
GO:0016741 GO:0016741    13     78        3   1.72 0.0260 0.46
GO:0016772 GO:0016772    64     78       10   1.80 0.0280 0.46
GO:0016866 GO:0016866    14     78        3   1.57 0.0340 0.46
GO:0016746 GO:0016746    15     78        3   1.44 0.0430 0.46
GO:0016616 GO:0016616    38     78        6   1.40 0.0550 0.46
GO:0016782 GO:0016782    17     78        3   1.19 0.0650 0.46
GO:0015078 GO:0015078    17     78        3   1.19 0.0650 0.46
GO:0003824 GO:0003824   545     78       56   1.30 0.0750 0.46
                                                                                       term_name
GO:0016740                                                                  transferase activity
GO:0008168                                                            methyltransferase activity
GO:0016741                                  transferase activity, transferring one-carbon groups
GO:0016772                       transferase activity, transferring phosphorus-containing groups
GO:0016866                                                   intramolecular transferase activity
GO:0016746                                        transferase activity, transferring acyl groups
GO:0016616 oxidoreductase activity, acting on the CH-OH group of donors, NAD or NADP as acceptor
GO:0016782                           transferase activity, transferring sulfur-containing groups
GO:0015078                                       hydrogen ion transmembrane transporter activity
GO:0003824                                                                    catalytic activity
               term_namespace term_distance
GO:0016740 molecular_function             2
GO:0008168 molecular_function             4
GO:0016741 molecular_function             3
GO:0016772 molecular_function             3
GO:0016866 molecular_function             3
GO:0016746 molecular_function             3
GO:0016616 molecular_function             4
GO:0016782 molecular_function             3
GO:0015078 molecular_function             8
GO:0003824 molecular_function             1
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   members
GO:0016740                                                                                                                                                                                                                                                                                                                                                     IPR028885,IPR018019,IPR003457,IPR005923,IPR020629,IPR011791,IPR015262,IPR017130,IPR000941,IPR003929,IPR000832,IPR026232,IPR027252,IPR013257,IPR002795,IPR017679,IPR010615,IPR002001,IPR020879,IPR026603,IPR012142,IPR007087
GO:0008168                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   IPR018019,IPR005923,IPR002795
GO:0016741                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   IPR018019,IPR005923,IPR002795
GO:0016772                                                                                                                                                                                                                                                                                                                                                                                                                                                                             IPR003457,IPR020629,IPR011791,IPR000941,IPR013257,IPR017679,IPR010615,IPR020879,IPR026603,IPR007087
GO:0016866                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   IPR015851,IPR004812,IPR009150
GO:0016746                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   IPR003929,IPR027252,IPR002001
GO:0016616                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     IPR022880,IPR012682,IPR011791,IPR018366,IPR000394,IPR013697
GO:0016782                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   IPR000832,IPR026232,IPR012142
GO:0015078                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   IPR023717,IPR008110,IPR017679
GO:0003824 IPR022880,IPR028885,IPR018019,IPR015851,IPR003457,IPR005923,IPR000602,IPR020629,IPR023717,IPR004812,IPR009150,IPR012682,IPR014280,IPR005774,IPR011791,IPR008723,IPR015262,IPR017130,IPR000941,IPR003929,IPR010426,IPR028725,IPR003935,IPR000832,IPR026232,IPR027252,IPR003819,IPR013257,IPR008110,IPR004976,IPR002115,IPR002795,IPR017679,IPR015556,IPR018366,IPR011179,IPR015631,IPR003088,IPR000394,IPR010615,IPR002001,IPR008909,IPR010122,IPR010600,IPR016067,IPR021117,IPR020879,IPR015827,IPR000505,IPR012745,IPR013697,IPR023532,IPR026128,IPR026603,IPR012142,IPR007087
</div>
<div class='input'>### visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'ontology.RData' containing 8 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-182.png', 'dcEnrichment-182.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-182.png', width="400" /></a></pre>
  </div>
  
  <div class="span4">
    
    <!-- 
    <ul>
      <li>dcEnrichment</li>
    </ul>
    <ul>
      
    </ul> 
    -->
    
    <h2>Source code</h2>
    <a href="javascript:newWin('R/dcEnrichment.r', 'R/dcEnrichment.r', '1200', '1200')" title="Click to view"><B><code>dcEnrichment.r</code></B></a>
    
    <h2>Source man</h2>
    <a href="javascript:newWin('man/dcEnrichment.txt', 'man/dcEnrichment.txt', '1200', '1200')" title="Click to view"><B><code>dcEnrichment.Rd</code></B></a>
    
    <a href="javascript:newWin('man/dcEnrichment.pdf', 'man/dcEnrichment.pdf', '1200', '1200')" title="Click to view"><B><code>dcEnrichment.pdf</code></B></a>
    
    <h2>See also</h2>
    
<code><a href='dcRDataLoader.html'>dcRDataLoader</a></code>, <code><a href='dcDAGannotate.html'>dcDAGannotate</a></code>,
<code><a href='Eoutput-class.html'>Eoutput-class</a></code>, <code><a href='visEnrichment.html'>visEnrichment</a></code>,
<code><a href='dcConverter.html'>dcConverter</a></code>

    <!--

    <h2>Citation</h2>
    <p>Fang H. (2014) <B>dcGOR: an R package for analysing ontologies and protein domain annotations</B>. <I>PLoS Computational Biology</I> 10(10):e1003929. <a href="http://dx.doi.org/10.1371/journal.pcbi.1003929">doi:10.1371/journal.pcbi.1003929</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25356683">25356683</a></p>

<p>You may also cite the package as:</p>

<ul>
<li><code>database</code></li>
</ul>

<p>Fang H, Gough J. (2013) <B>dcGO: database of domain-centric ontologies on functions, phenotypes, diseases and more</B>. <I>Nucleic Acids Research</I> 41(Database issue):D536-44. <a href="http://dx.doi.org/10.1093/nar/gks1080">doi:10.1093/nar/gks1080</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23161684">23161684</a></p>

<ul>
<li><code>web link/URL</code></li>
</ul>

<p>Package &ldquo;dcGOR&rdquo; [<a href="http://cran.r-project.org/package=dcGOR">http://cran.r-project.org/package=dcGOR</a>]</p>

    
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://supfam.org/dcGOR">http://supfam.org/dcGOR</a></li>
      <li><a href=" http://dcgor.r-forge.r-project.org"> http://dcgor.r-forge.r-project.org</a></li>
      <li><a href=" http://cran.r-
project.org/package=dcGOR"> http://cran.r-
project.org/package=dcGOR</a></li>
      <li><a href=" https://github.com/hfang-bristol/dcGOR"> https://github.com/hfang-bristol/dcGOR</a></li>
    </ul>
    
    
    -->
    
  </div>
  
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@well.ox.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>