<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>dcEnrichment @ dcGOR 1.0.5</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dcGOR 1.0.5</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
       
        <!-- HTML for GOOLE SEARCH BAR 
        <ul class="nav">
            <div id="search-wrapper">
                <form id="search" method="get" action="http://www.google.com/custom">
                    <div class="search-box">
                        <i class="icon-search icon-white"></i>
                        <input type="text" id="tfq" class="textinput" name="q" size="21" maxlength="120" value="Google Search"/>
                    </div>
                </form>
            </div>
        </ul>
        -->
        
      </div>
    </div>
  </div>
</div>

<!-- CSS styles for standard search box with placeholder text-->
<style type="text/css">
    #search-wrapper {
        width:300px;
        height:40px;
        background-color:transparent;
        margin:0px auto 0;
        border:0px solid #ff0000;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

    #search {
        background-color:transparent;
        padding:0px;
        height:40px;
        overflow:hidden;
    }

    .search-box {
        width:250px;
        height:32px;
        background-color:transparent;
        margin:2px 7px;
        border:0px solid #cfcfcf;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

	.textinput{
		margin: 0;
		padding: 5px 15px;
		font-family: Arial, Helvetica, sans-serif;
		font-size:14px;
		color:#999999;
		border:1px solid #f00; border-right:0px;
		border-top-left-radius: 5px 5px;
		border-bottom-left-radius: 5px 5px;
	}
</style>

<!-- JAVASCRIPT to clear search text when the field is clicked -->
<script type="text/javascript">
window.onload = function(){
	//Get submit button
	var submitbutton = document.getElementById("tfq");
	//Add listener to submit button
	if(submitbutton.addEventListener){
		submitbutton.addEventListener("click", function() {
			if (submitbutton.value == 'Google Search'){//Customize this text string to whatever you want
				submitbutton.value = 'dcGOR R ';
			}
		});
	}
}
</script>

<script>
  (function() {
    var cx = '012094530736424854714:k8ieyz4ag4a';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') +
        '//www.google.com/cse/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>

<div><gcse:search></gcse:search></div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Function to conduct ontology enrichment analysis given a group of domains</h2>

<div class="row">
  <div class="span8">
  
    <h3>Description</h3>
    <p><code>dcEnrichment</code> is supposed to conduct enrichment analysis for an
input group of domains using a specified ontology. It returns an object
of S4 class "Eoutput". Enrichment analysis is based on either Fisher's
exact test or Hypergeometric test. The test can respect the hierarchy
of the ontology. The user can customise the background domains;
otherwise, the function will use all annotatable domains as the test
background</p>
  
    <h3>Usage</h3>
    <pre>dcEnrichment(data, background = NULL, domain = c(NA, "SCOP.sf", "SCOP.fa", "Pfam", 
  "InterPro", "Rfam"), ontology = c(NA, "GOBP", "GOMF", "GOCC", "DO", "HPPA", "HPMI", 
      "HPON", "MP", "EC", "KW", "UP"), sizeRange = c(10, 1000), min.overlap = 3, which_distance = NULL, 
      test = c("HypergeoTest", "FisherTest", "BinomialTest"), p.adjust.method = c("BH", 
          "BY", "bonferroni", "holm", "hochberg", "hommel"), ontology.algorithm = c("none", 
          "pc", "elim", "lea"), elim.pvalue = 0.01, lea.depth = 2, verbose = T, domain.RData = NULL, 
      ontology.RData = NULL, annotations.RData = NULL, RData.location = "http://dcgor.r-forge.r-project.org/data")</pre>
    
    <h3>Arguments</h3>
    <dl>
      <dt>data</dt>
      <dd>an input vector. It contains id for a list of domains, for
example, sunids for SCOP domains</dd>
      <dt>background</dt>
      <dd>a background vector. It contains id for a list of
background domains, for example, sunids for SCOP domains. If NULL, by
default all annotatable domains are used as background</dd>
      <dt>domain</dt>
      <dd>the domain identity. It can be one of 'SCOP.sf' for SCOP
superfamilies, 'SCOP.fa' for SCOP families, 'Pfam' for Pfam domains,
'InterPro' for InterPro domains, 'Rfam' for Rfam RNA families</dd>
      <dt>ontology</dt>
      <dd>the ontology identity. It can be "GOBP" for Gene
Ontology Biological Process, "GOMF" for Gene Ontology Molecular
Function, "GOCC" for Gene Ontology Cellular Component, "DO" for Disease
Ontology, "HPPA" for Human Phenotype Phenotypic Abnormality, "HPMI" for
Human Phenotype Mode of Inheritance, "HPON" for Human Phenotype ONset
and clinical course, "MP" for Mammalian Phenotype, "EC" for Enzyme
Commission, "KW" for UniProtKB KeyWords, "UP" for UniProtKB UniPathway.
For details on the eligibility for pairs of input domain and ontology,
please refer to the online Documentations at
<a href = 'http://supfam.org/dcGOR/docs.html'>http://supfam.org/dcGOR/docs.html</a></dd>
      <dt>sizeRange</dt>
      <dd>the minimum and maximum size of members of each term
in consideration. By default, it sets to a minimum of 10 but no more
than 1000</dd>
      <dt>min.overlap</dt>
      <dd>the minimum number of overlaps. Only those terms
that overlap with input data at least min.overlap (3 domains by
default) will be processed</dd>
      <dt>which_distance</dt>
      <dd>which distance of terms in the ontology is used
to restrict terms in consideration. By default, it sets to 'NULL' to
consider all distances</dd>
      <dt>test</dt>
      <dd>the statistic test used. It can be "FisherTest" for using
fisher's exact test, "HypergeoTest" for using hypergeometric test, or
"BinomialTest" for using binomial test. Fisher's exact test is to test
the independence between domain group (domains belonging to a group or
not) and domain annotation (domains annotated by a term or not), and
thus compare sampling to the left part of background (after sampling
without replacement). Hypergeometric test is to sample at random
(without replacement) from the background containing annotated and
non-annotated domains, and thus compare sampling to background. Unlike
hypergeometric test, binomial test is to sample at random (with
replacement) from the background with the constant probability. In
terms of the ease of finding the significance, they are in order:
hypergeometric test > binomial test > fisher's exact test. In other
words, in terms of the calculated p-value, hypergeometric test <
binomial test < fisher's exact test</dd>
      <dt>p.adjust.method</dt>
      <dd>the method used to adjust p-values. It can be
one of "BH", "BY", "bonferroni", "holm", "hochberg" and "hommel". The
first two methods "BH" (widely used) and "BY" control the false
discovery rate (FDR: the expected proportion of false discoveries
amongst the rejected hypotheses); the last four methods "bonferroni",
"holm", "hochberg" and "hommel" are designed to give strong control of
the family-wise error rate (FWER). Notes: FDR is a less stringent
condition than FWER</dd>
      <dt>ontology.algorithm</dt>
      <dd>the algorithm used to account for the
hierarchy of the ontology. It can be one of "none", "pc", "elim" and
"lea". For details, please see 'Note'</dd>
      <dt>elim.pvalue</dt>
      <dd>the parameter only used when "ontology.algorithm" is
"elim". It is used to control how to declare a signficantly enriched
term (and subsequently all domains in this term are eliminated from all
its ancestors)</dd>
      <dt>lea.depth</dt>
      <dd>the parameter only used when "ontology.algorithm" is
"lea". It is used to control how many maximum depth is uded to consider
the children of a term (and subsequently all domains in these children
term are eliminated from the use for the recalculation of the
signifance at this term)</dd>
      <dt>verbose</dt>
      <dd>logical to indicate whether the messages will be
displayed in the screen. By default, it sets to TRUE for display</dd>
      <dt>domain.RData</dt>
      <dd>a file name for RData-formatted file containing an
object of S4 class 'InfoDataFrame' (i.g. domain). By default, it is
NULL. It is only needed when the user wants to customise enrichment
analysis using their own data. See <code><a href='dcBuildInfoDataFrame.html'>dcBuildInfoDataFrame</a></code>
for how to creat this object</dd>
      <dt>ontology.RData</dt>
      <dd>a file name for RData-formatted file containing
an object of S4 class 'Onto' (i.g. ontology). By default, it is NULL.
It is only needed when the user wants to customise enrichment analysis
using their own data. See <code><a href='dcBuildOnto.html'>dcBuildOnto</a></code> for how to creat
this object</dd>
      <dt>annotations.RData</dt>
      <dd>a file name for RData-formatted file
containing an object of S4 class 'Anno' (i.g. annotations). By default,
it is NULL. It is only needed when the user wants to customise
enrichment analysis using their own data. See <code><a href='dcBuildAnno.html'>dcBuildAnno</a></code>
for how to creat this object</dd>
      <dt>RData.location</dt>
      <dd>the characters to tell the location of built-in
RData files. By default, it remotely locates at
"http://supfam.org/dcGOR/data" or
"http://dcgor.r-forge.r-project.org/data". For the user equipped with
fast internet connection, this option can be just left as default. But
it is always advisable to download these files locally. Especially when
the user needs to run this function many times, there is no need to ask
the function to remotely download every time (also it will
unnecessarily increase the runtime). For examples, these files (as a
whole or part of them) can be first downloaded into your current
working directory, and then set this option as:
<code class = 'eq'>RData.location="."</code>. If RData to load is already part of package
itself, this parameter can be ignored (since this function will try to
load it via function <code>data</code> first)</dd>
    </dl>
    <!-- 
    -->
    
    <!-- 
    <div class="Value">
    -->
      <h3>Value</h3>
      
      <p>an object of S4 class <code><a href='Eoutput-class.html'>Eoutput</a></code>, with following slots:
<ul>
<li><code>domain</code>: a character specifying the domain identity
</li>
<li><code>ontology</code>: a character specifying the ontology used
</li>
<li><code>term_info</code>: a matrix of nTerm X 5 containing term
information, where nTerm is the number of terms in consideration, and
the 5 columns are "term_id" (i.e. "Term ID"), "term_name" (i.e. "Term
Name"), "namespace" (i.e. "Term Namespace"), "distance" (i.e. "Term
Distance") and "IC" (i.e. "Information Content for the term based on
annotation frequency by it")
</li>
<li><code>anno</code>: a list of terms, each storing annotated domain
members (also within the background domains). Always, terms are
identified by "term_id" and domain members identified by their ids
(e.g. sunids for SCOP domains)
</li>
<li><code>data</code>: a vector containing input data in consideration. It
is not always the same as the input data as only those mappable and
annotatable are retained
</li>
<li><code>background</code>: a vector containing background in
consideration. It is not always the same as the input background as
only those mappable/annotatable are retained
</li>
<li><code>overlap</code>: a list of terms, each storing domains overlapped
between domains annotated by a term and domains in the input data (i.e.
the domains of interest). Always, terms are identified by "term_id" and
domain members identified by their IDs (e.g. sunids for SCOP domains)
</li>
<li><code>zscore</code>: a vector containing z-scores
</li>
<li><code>pvalue</code>: a vector containing p-values
</li>
<li><code>adjp</code>: a vector containing adjusted p-values. It is the p
value but after being adjusted for multiple comparisons
</li>
</ul></p>
      <p></p>
    <!-- 
    </div>
    -->
    <!-- 
    <div class="Note">
    -->
      <h3>Note</h3>
      
      <p>The interpretation of the algorithms used to account for the hierarchy
of the ontology is:
<ul>
<li>"none": does not consider the ontology hierarchy at all.
</li>
<li>"lea": computers the significance of a term in terms of the
significance of its children at the maximum depth (e.g. 2). Precisely,
once domains are already annotated to any children terms with a more
signficance than itself, then all these domains are eliminated from the
use for the recalculation of the signifance at that term. The final
p-values takes the maximum of the original p-value and the recalculated
p-value.
</li>
<li>"elim": computers the significance of a term in terms of the
significance of its all children. Precisely, once domains are already
annotated to a signficantly enriched term under the cutoff of e.g.
pvalue<1e-2, all these domains are eliminated from the ancestors of
that term).
</li>
<li>"pc": requires the significance of a term not only using the
whole domains as background but also using domains annotated to all its
direct parents/ancestors as background. The final p-value takes the
maximum of both p-values in these two calculations.
</li>
<li>"Notes": the order of the number of significant terms is: "none"
> "lea" > "elim" > "pc".
</li>
</ul></p>
      <p></p>
    <!-- 
    </div>
    -->
    
    <h3 id="examples">Examples</h3>
    <pre class="examples"><div class='input'># 1) Enrichment analysis for SCOP domain superfamilies (sf)
## 1a) load SCOP.sf (as 'InfoDataFrame' object)
SCOP.sf <- dcRDataLoader('SCOP.sf')
</div>
<strong class='message'>'SCOP.sf' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<div class='input'>### randomly select 50 domains as a list of domains of interest
data <- sample(rowNames(SCOP.sf), 50)
## 1b) perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, domain="SCOP.sf", ontology="GOMF")
</div>
<strong class='message'>Start at 2014-12-05 17:17:09</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'SCOP.sf', and their associations (2014-12-05 17:17:09) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf2GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2014-12-05 17:17:15) ...</strong>
<strong class='message'>	There are 811 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-12-05 17:17:15) ...</strong>
<strong class='message'>
End at 2014-12-05 17:17:15</strong>
<strong class='message'>Runtime in total is: 6 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'SCOP.sf'
  @ontology: 'GOMF'
  @term_info: a data.frame of 181 terms X 5 information
  @anno: a list of 181 terms, each storing annotated domains
  @data: a vector containing a group of 18 input domains (annotatable)
  @background: a vector containing a group of 1083 background domains (annotatable)
  @overlap: a list of 181 terms, each containing domains overlapped with input domains
  @zscore: a vector of 181 terms, containing z-scores
  @pvalue: a vector of 181 terms, containing p-values
  @adjp: a vector of 181 terms, containing adjusted p-values
In summary, a total of 181 terms ('GOMF') are analysed for a group of 18 input domains ('SCOP.sf')
</div>
<div class='input'>## 1c) view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore  pvalue  adjp
GO:0004725 GO:0004725    21     18        3   4.57 0.00027 0.036
GO:0022892 GO:0022892   704     18       17   2.64 0.00040 0.036
GO:0005215 GO:0005215   758     18       17   2.28 0.00150 0.078
GO:0015187 GO:0015187    36     18        3   3.18 0.00230 0.078
GO:0015175 GO:0015175    37     18        3   3.12 0.00250 0.078
GO:0005342 GO:0005342   249     18        9   2.74 0.00260 0.078
GO:0004721 GO:0004721    70     18        4   2.74 0.00430 0.098
GO:0004643 GO:0004643    71     18        4   2.71 0.00460 0.098
GO:0016799 GO:0016799    46     18        3   2.63 0.00560 0.098
GO:0004730 GO:0004730    46     18        3   2.63 0.00560 0.098
                                                                    term_name
GO:0004725                              protein tyrosine phosphatase activity
GO:0022892                            substrate-specific transporter activity
GO:0005215                                               transporter activity
GO:0015187                         glycine transmembrane transporter activity
GO:0015175              neutral amino acid transmembrane transporter activity
GO:0005342                    organic acid transmembrane transporter activity
GO:0004721                                phosphoprotein phosphatase activity
GO:0004643 phosphoribosylaminoimidazolecarboxamide formyltransferase activity
GO:0016799               hydrolase activity, hydrolyzing N-glycosyl compounds
GO:0004730                                  pseudouridylate synthase activity
               term_namespace term_distance
GO:0004725 molecular_function             7
GO:0022892 molecular_function             2
GO:0005215 molecular_function             1
GO:0015187 molecular_function             6
GO:0015175 molecular_function             9
GO:0005342 molecular_function             4
GO:0004721 molecular_function             6
GO:0004643 molecular_function             5
GO:0016799 molecular_function             4
GO:0004730 molecular_function             5
                                                                                                               members
GO:0004725                                                                                           49562,54495,81324
GO:0022892 51412,52021,50475,54843,49562,54495,111418,109925,54529,116846,49695,140586,140990,57667,54719,81324,159034
GO:0005215 51412,52021,50475,54843,49562,54495,111418,109925,54529,116846,49695,140586,140990,57667,54719,81324,159034
GO:0015187                                                                                          49562,54495,159034
GO:0015175                                                                                          49562,54495,159034
GO:0005342                                                      51412,52021,50475,49562,54495,57667,54719,81324,159034
GO:0004721                                                                                     49562,54495,57667,81324
GO:0004643                                                                                    52021,49562,54495,159034
GO:0016799                                                                                           49562,57667,81324
GO:0004730                                                                                          49562,54495,159034
</div>
<div class='input'>## 1d) visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 19 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-21.png', 'dcEnrichment-21.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-21.png', width="400" /></a>
<div class='input'>## 1e) the same as above but using a customised background
### randomly select 500 domains as background
background <- sample(rowNames(SCOP.sf), 500)
### perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, background=background, domain="SCOP.sf",
ontology="GOMF")
</div>
<strong class='message'>Start at 2014-12-05 17:17:18</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'SCOP.sf', and their associations (2014-12-05 17:17:18) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf2GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2014-12-05 17:17:23) ...</strong>
<strong class='message'>	There are 401 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-12-05 17:17:23) ...</strong>
<strong class='message'>
End at 2014-12-05 17:17:23</strong>
<strong class='message'>Runtime in total is: 5 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'SCOP.sf'
  @ontology: 'GOMF'
  @term_info: a data.frame of 180 terms X 5 information
  @anno: a list of 180 terms, each storing annotated domains
  @data: a vector containing a group of 18 input domains (annotatable)
  @background: a vector containing a group of 255 background domains (annotatable)
  @overlap: a list of 180 terms, each containing domains overlapped with input domains
  @zscore: a vector of 180 terms, containing z-scores
  @pvalue: a vector of 180 terms, containing p-values
  @adjp: a vector of 180 terms, containing adjusted p-values
In summary, a total of 180 terms ('GOMF') are analysed for a group of 18 input domains ('SCOP.sf')
</div>
<div class='input'>### view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore  pvalue  adjp
GO:0022892 GO:0022892   168     18       17   2.65 0.00039 0.071
GO:0005215 GO:0005215   179     18       17   2.33 0.00130 0.120
GO:0016799 GO:0016799    11     18        3   2.67 0.00430 0.190
GO:0015187 GO:0015187    12     18        3   2.48 0.00610 0.190
GO:0015175 GO:0015175    12     18        3   2.48 0.00610 0.190
GO:0005342 GO:0005342    70     18        9   2.22 0.00860 0.190
GO:0004643 GO:0004643    21     18        4   2.23 0.00980 0.190
GO:0004721 GO:0004721    21     18        4   2.23 0.00980 0.190
GO:0004730 GO:0004730    14     18        3   2.16 0.01100 0.190
GO:0015179 GO:0015179    14     18        3   2.16 0.01100 0.190
                                                                    term_name
GO:0022892                            substrate-specific transporter activity
GO:0005215                                               transporter activity
GO:0016799               hydrolase activity, hydrolyzing N-glycosyl compounds
GO:0015187                         glycine transmembrane transporter activity
GO:0015175              neutral amino acid transmembrane transporter activity
GO:0005342                    organic acid transmembrane transporter activity
GO:0004643 phosphoribosylaminoimidazolecarboxamide formyltransferase activity
GO:0004721                                phosphoprotein phosphatase activity
GO:0004730                                  pseudouridylate synthase activity
GO:0015179                    L-amino acid transmembrane transporter activity
               term_namespace term_distance
GO:0022892 molecular_function             2
GO:0005215 molecular_function             1
GO:0016799 molecular_function             4
GO:0015187 molecular_function             6
GO:0015175 molecular_function             9
GO:0005342 molecular_function             4
GO:0004643 molecular_function             5
GO:0004721 molecular_function             6
GO:0004730 molecular_function             5
GO:0015179 molecular_function             9
                                                                                                               members
GO:0022892 54843,54495,50475,140586,116846,109925,140990,49562,57667,54529,159034,111418,54719,52021,51412,49695,81324
GO:0005215 54843,54495,50475,140586,116846,109925,140990,49562,57667,54529,159034,111418,54719,52021,51412,49695,81324
GO:0016799                                                                                           49562,57667,81324
GO:0015187                                                                                          54495,49562,159034
GO:0015175                                                                                          54495,49562,159034
GO:0005342                                                      54495,50475,49562,57667,159034,54719,52021,51412,81324
GO:0004643                                                                                    54495,49562,159034,52021
GO:0004721                                                                                     54495,49562,57667,81324
GO:0004730                                                                                          54495,49562,159034
GO:0015179                                                                                          54495,49562,159034
</div>
<div class='input'>### visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 16 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-40.png', 'dcEnrichment-40.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-40.png', width="400" /></a>
<div class='input'>
###########################################################
# 2) Enrichment analysis for Pfam domains (Pfam)
## 2a) load Pfam (as 'InfoDataFrame' object)
Pfam <- dcRDataLoader('Pfam')
</div>
<strong class='message'>'Pfam' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<div class='input'>### randomly select 100 domains as a list of domains of interest
data <- sample(rowNames(Pfam), 100)
## 2b) perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, domain="Pfam", ontology="GOMF")
</div>
<strong class='message'>Start at 2014-12-05 17:17:25</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'Pfam', and their associations (2014-12-05 17:17:25) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Pfam' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Pfam2GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2014-12-05 17:17:28) ...</strong>
<strong class='message'>	There are 334 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-12-05 17:17:28) ...</strong>
<strong class='message'>
End at 2014-12-05 17:17:28</strong>
<strong class='message'>Runtime in total is: 3 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'Pfam'
  @ontology: 'GOMF'
  @term_info: a data.frame of 9 terms X 5 information
  @anno: a list of 9 terms, each storing annotated domains
  @data: a vector containing a group of 23 input domains (annotatable)
  @background: a vector containing a group of 3359 background domains (annotatable)
  @overlap: a list of 9 terms, each containing domains overlapped with input domains
  @zscore: a vector of 9 terms, containing z-scores
  @pvalue: a vector of 9 terms, containing p-values
  @adjp: a vector of 9 terms, containing adjusted p-values
In summary, a total of 9 terms ('GOMF') are analysed for a group of 23 input domains ('Pfam')
</div>
<div class='input'>## 2c) view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue adjp
GO:0005488 GO:0005488   730     23        7  1.020   0.11 0.40
GO:0004451 GO:0004451   315     23        3  0.605   0.16 0.40
GO:0016833 GO:0016833   317     23        3  0.594   0.17 0.40
GO:0005515 GO:0005515   445     23        4  0.588   0.18 0.40
GO:0016830 GO:0016830   372     23        3  0.302   0.25 0.42
GO:0016829 GO:0016829   550     23        4  0.132   0.32 0.42
GO:0016787 GO:0016787   690     23        5  0.143   0.33 0.42
GO:0016740 GO:0016740   767     23        5 -0.126   0.43 0.49
GO:0003824 GO:0003824  2241     23       14 -0.597   0.65 0.65
                              term_name     term_namespace term_distance
GO:0005488                      binding molecular_function             1
GO:0004451    isocitrate lyase activity molecular_function             5
GO:0016833      oxo-acid-lyase activity molecular_function             4
GO:0005515              protein binding molecular_function             2
GO:0016830 carbon-carbon lyase activity molecular_function             3
GO:0016829               lyase activity molecular_function             2
GO:0016787           hydrolase activity molecular_function             2
GO:0016740         transferase activity molecular_function             2
GO:0003824           catalytic activity molecular_function             1
                                                                                                                   members
GO:0005488                                                         PF01977,PF02807,PF05927,PF09515,PF05088,PF05111,PF07664
GO:0004451                                                                                         PF01131,PF04565,PF07669
GO:0016833                                                                                         PF01131,PF04565,PF07669
GO:0005515                                                                                 PF01977,PF05927,PF05111,PF07664
GO:0016830                                                                                         PF01131,PF04565,PF07669
GO:0016829                                                                                 PF01131,PF01496,PF04565,PF07669
GO:0016787                                                                         PF00445,PF01526,PF00998,PF04565,PF01909
GO:0016740                                                                         PF13741,PF02833,PF08086,PF07544,PF02302
GO:0003824 PF13741,PF00445,PF01047,PF02833,PF01131,PF01496,PF01526,PF00998,PF08086,PF07544,PF04565,PF02302,PF07669,PF01909
</div>
<div class='input'>## 2d) visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 8 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-61.png', 'dcEnrichment-61.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-61.png', width="400" /></a>
<div class='input'>## 2e) the same as above but using a customised background
### randomly select 1000 domains as background
background <- sample(rowNames(Pfam), 1000)
### perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, background=background, domain="Pfam",
ontology="GOMF")
</div>
<strong class='message'>Start at 2014-12-05 17:17:30</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'Pfam', and their associations (2014-12-05 17:17:30) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Pfam' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Pfam2GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2014-12-05 17:17:33) ...</strong>
<strong class='message'>	There are 32 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-12-05 17:17:33) ...</strong>
<strong class='message'>
End at 2014-12-05 17:17:33</strong>
<strong class='message'>Runtime in total is: 3 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'Pfam'
  @ontology: 'GOMF'
  @term_info: a data.frame of 9 terms X 5 information
  @anno: a list of 9 terms, each storing annotated domains
  @data: a vector containing a group of 23 input domains (annotatable)
  @background: a vector containing a group of 227 background domains (annotatable)
  @overlap: a list of 9 terms, each containing domains overlapped with input domains
  @zscore: a vector of 9 terms, containing z-scores
  @pvalue: a vector of 9 terms, containing p-values
  @adjp: a vector of 9 terms, containing adjusted p-values
In summary, a total of 9 terms ('GOMF') are analysed for a group of 23 input domains ('Pfam')
</div>
<div class='input'>### view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap  zscore pvalue adjp
GO:0005488 GO:0005488    45     23        7  1.3400  0.058 0.35
GO:0005515 GO:0005515    28     23        4  0.7760  0.130 0.35
GO:0016833 GO:0016833    23     23        3  0.4870  0.190 0.35
GO:0004451 GO:0004451    23     23        3  0.4870  0.190 0.35
GO:0016830 GO:0016830    26     23        3  0.2520  0.260 0.35
GO:0016787 GO:0016787    44     23        5  0.3010  0.270 0.35
GO:0016829 GO:0016829    35     23        4  0.2760  0.270 0.35
GO:0003824 GO:0003824   139     23       14 -0.0377  0.430 0.46
GO:0016740 GO:0016740    53     23        5 -0.1920  0.460 0.46
                              term_name     term_namespace term_distance
GO:0005488                      binding molecular_function             1
GO:0005515              protein binding molecular_function             2
GO:0016833      oxo-acid-lyase activity molecular_function             4
GO:0004451    isocitrate lyase activity molecular_function             5
GO:0016830 carbon-carbon lyase activity molecular_function             3
GO:0016787           hydrolase activity molecular_function             2
GO:0016829               lyase activity molecular_function             2
GO:0003824           catalytic activity molecular_function             1
GO:0016740         transferase activity molecular_function             2
                                                                                                                   members
GO:0005488                                                         PF02807,PF05088,PF05111,PF01977,PF05927,PF07664,PF09515
GO:0005515                                                                                 PF05111,PF01977,PF05927,PF07664
GO:0016833                                                                                         PF01131,PF04565,PF07669
GO:0004451                                                                                         PF01131,PF04565,PF07669
GO:0016830                                                                                         PF01131,PF04565,PF07669
GO:0016787                                                                         PF01526,PF00445,PF00998,PF01909,PF04565
GO:0016829                                                                                 PF01131,PF01496,PF04565,PF07669
GO:0003824 PF01526,PF01131,PF13741,PF00445,PF00998,PF01496,PF02302,PF01909,PF07544,PF02833,PF04565,PF01047,PF07669,PF08086
GO:0016740                                                                         PF13741,PF02302,PF07544,PF02833,PF08086
</div>
<div class='input'>### visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 8 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-80.png', 'dcEnrichment-80.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-80.png', width="400" /></a>
<div class='input'>
###########################################################
# 3) Enrichment analysis for InterPro domains (InterPro)
## 3a) load InterPro (as 'InfoDataFrame' object)
InterPro <- dcRDataLoader('InterPro')
</div>
<strong class='message'>'InterPro' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<div class='input'>### randomly select 100 domains as a list of domains of interest
data <- sample(rowNames(InterPro), 100)
## 3b) perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, domain="InterPro", ontology="GOMF")
</div>
<strong class='message'>Start at 2014-12-05 17:17:35</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'InterPro', and their associations (2014-12-05 17:17:35) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'InterPro' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'InterPro2GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2014-12-05 17:17:42) ...</strong>
<strong class='message'>	There are 630 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-12-05 17:17:42) ...</strong>
<strong class='message'>
End at 2014-12-05 17:17:42</strong>
<strong class='message'>Runtime in total is: 7 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'InterPro'
  @ontology: 'GOMF'
  @term_info: a data.frame of 66 terms X 5 information
  @anno: a list of 66 terms, each storing annotated domains
  @data: a vector containing a group of 76 input domains (annotatable)
  @background: a vector containing a group of 8899 background domains (annotatable)
  @overlap: a list of 66 terms, each containing domains overlapped with input domains
  @zscore: a vector of 66 terms, containing z-scores
  @pvalue: a vector of 66 terms, containing p-values
  @adjp: a vector of 66 terms, containing adjusted p-values
In summary, a total of 66 terms ('GOMF') are analysed for a group of 76 input domains ('InterPro')
</div>
<div class='input'>## 3c) view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue  adjp
GO:0003995 GO:0003995    56     76        3   3.67 0.0013 0.053
GO:0051748 GO:0051748   114     76        4   3.10 0.0028 0.053
GO:0003983 GO:0003983   114     76        4   3.10 0.0028 0.053
GO:0004534 GO:0004534    78     76        3   2.88 0.0043 0.053
GO:0070569 GO:0070569   126     76        4   2.85 0.0044 0.053
GO:0008409 GO:0008409    81     76        3   2.80 0.0050 0.053
GO:0009881 GO:0009881   140     76        4   2.60 0.0068 0.053
GO:0004532 GO:0004532    90     76        3   2.57 0.0072 0.053
GO:0016896 GO:0016896    90     76        3   2.57 0.0072 0.053
GO:0015002 GO:0015002   434     76        8   2.30 0.0110 0.068
                                                             term_name
GO:0003995                             acyl-CoA dehydrogenase activity
GO:0051748 UTP-monosaccharide-1-phosphate uridylyltransferase activity
GO:0003983        UTP:glucose-1-phosphate uridylyltransferase activity
GO:0004534                              5'-3' exoribonuclease activity
GO:0070569                                uridylyltransferase activity
GO:0008409                                  5'-3' exonuclease activity
GO:0009881                                      photoreceptor activity
GO:0004532                                    exoribonuclease activity
GO:0016896    exoribonuclease activity, producing 5'-phosphomonoesters
GO:0015002                       heme-copper terminal oxidase activity
               term_namespace term_distance
GO:0003995 molecular_function             4
GO:0051748 molecular_function             6
GO:0003983 molecular_function             7
GO:0004534 molecular_function             7
GO:0070569 molecular_function             5
GO:0008409 molecular_function             6
GO:0009881 molecular_function             4
GO:0004532 molecular_function             6
GO:0016896 molecular_function             7
GO:0015002 molecular_function             3
                                                                                   members
GO:0003995                                                   IPR009396,IPR004978,IPR009608
GO:0051748                                         IPR004688,IPR001802,IPR019831,IPR027352
GO:0003983                                         IPR004688,IPR001802,IPR019831,IPR027352
GO:0004534                                                   IPR015944,IPR012795,IPR014758
GO:0070569                                         IPR004688,IPR001802,IPR019831,IPR027352
GO:0008409                                                   IPR015944,IPR012795,IPR014758
GO:0009881                                         IPR000717,IPR013655,IPR001943,IPR024855
GO:0004532                                                   IPR015944,IPR012795,IPR014758
GO:0016896                                                   IPR015944,IPR012795,IPR014758
GO:0015002 IPR004821,IPR028493,IPR000535,IPR013670,IPR016036,IPR016039,IPR008928,IPR006638
</div>
<div class='input'>## 3d) visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 17 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-101.png', 'dcEnrichment-101.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-101.png', width="400" /></a>
<div class='input'>## 3e) the same as above but using a customised background
### randomly select 1000 domains as background
background <- sample(rowNames(InterPro), 1000)
### perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, background=background, domain="InterPro",
ontology="GOMF")
</div>
<strong class='message'>Start at 2014-12-05 17:17:44</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'InterPro', and their associations (2014-12-05 17:17:44) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'InterPro' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'InterPro2GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2014-12-05 17:17:50) ...</strong>
<strong class='message'>	There are 136 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-12-05 17:17:50) ...</strong>
<strong class='message'>
End at 2014-12-05 17:17:50</strong>
<strong class='message'>Runtime in total is: 6 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'InterPro'
  @ontology: 'GOMF'
  @term_info: a data.frame of 62 terms X 5 information
  @anno: a list of 62 terms, each storing annotated domains
  @data: a vector containing a group of 76 input domains (annotatable)
  @background: a vector containing a group of 862 background domains (annotatable)
  @overlap: a list of 62 terms, each containing domains overlapped with input domains
  @zscore: a vector of 62 terms, containing z-scores
  @pvalue: a vector of 62 terms, containing p-values
  @adjp: a vector of 62 terms, containing adjusted p-values
In summary, a total of 62 terms ('GOMF') are analysed for a group of 76 input domains ('InterPro')
</div>
<div class='input'>### view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue adjp
GO:0015002 GO:0015002    41     76        8   2.47 0.0068 0.13
GO:0004540 GO:0004540    10     76        3   2.38 0.0078 0.13
GO:0016796 GO:0016796    11     76        3   2.17 0.0110 0.13
GO:0016772 GO:0016772    61     76       10   2.16 0.0130 0.13
GO:0003995 GO:0003995    12     76        3   1.99 0.0160 0.13
GO:0004527 GO:0004527    12     76        3   1.99 0.0160 0.13
GO:0009881 GO:0009881    19     76        4   1.90 0.0200 0.13
GO:0070569 GO:0070569    19     76        4   1.90 0.0200 0.13
GO:0051748 GO:0051748    19     76        4   1.90 0.0200 0.13
GO:0003983 GO:0003983    19     76        4   1.90 0.0200 0.13
                                                                                                      term_name
GO:0015002                                                                heme-copper terminal oxidase activity
GO:0004540                                                                                ribonuclease activity
GO:0016796 exonuclease activity, active with either ribo- or deoxyribonucleic acids and producing 5'-phosphomon
GO:0016772                                      transferase activity, transferring phosphorus-containing groups
GO:0003995                                                                      acyl-CoA dehydrogenase activity
GO:0004527                                                                                 exonuclease activity
GO:0009881                                                                               photoreceptor activity
GO:0070569                                                                         uridylyltransferase activity
GO:0051748                                          UTP-monosaccharide-1-phosphate uridylyltransferase activity
GO:0003983                                                 UTP:glucose-1-phosphate uridylyltransferase activity
               term_namespace term_distance
GO:0015002 molecular_function             3
GO:0004540 molecular_function             5
GO:0016796 molecular_function             6
GO:0016772 molecular_function             3
GO:0003995 molecular_function             4
GO:0004527 molecular_function             5
GO:0009881 molecular_function             4
GO:0070569 molecular_function             5
GO:0051748 molecular_function             6
GO:0003983 molecular_function             7
                                                                                                       members
GO:0015002                     IPR004821,IPR028493,IPR008928,IPR016036,IPR016039,IPR000535,IPR006638,IPR013670
GO:0004540                                                                       IPR014758,IPR015944,IPR012795
GO:0016796                                                                       IPR014758,IPR015944,IPR012795
GO:0016772 IPR022350,IPR004688,IPR027352,IPR028740,IPR005086,IPR023652,IPR018430,IPR001765,IPR019831,IPR001802
GO:0003995                                                                       IPR009396,IPR004978,IPR009608
GO:0004527                                                                       IPR014758,IPR015944,IPR012795
GO:0009881                                                             IPR024855,IPR001943,IPR013655,IPR000717
GO:0070569                                                             IPR004688,IPR027352,IPR019831,IPR001802
GO:0051748                                                             IPR004688,IPR027352,IPR019831,IPR001802
GO:0003983                                                             IPR004688,IPR027352,IPR019831,IPR001802
</div>
<div class='input'>### visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 14 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-120.png', 'dcEnrichment-120.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-120.png', width="400" /></a>
<div class='input'>
###########################################################
# 4) Enrichment analysis for Rfam RNA families (Rfam)
## 4a) load Rfam (as 'InfoDataFrame' object)
Rfam <- dcRDataLoader('Rfam')
</div>
<strong class='message'>'Rfam' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<div class='input'>### randomly select 100 RNAs as a list of RNAs of interest
data <- sample(rowNames(Rfam), 100)
## 4b) perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, domain="Rfam", ontology="GOBP")
</div>
<strong class='message'>Start at 2014-12-05 17:17:52</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOBP', the domain 'Rfam', and their associations (2014-12-05 17:17:52) ...</strong>
<strong class='message'>'onto.GOBP' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Rfam' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Rfam2GOBP' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2014-12-05 17:17:54) ...</strong>
<strong class='message'>	There are 114 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-12-05 17:17:55) ...</strong>
<strong class='message'>
End at 2014-12-05 17:17:55</strong>
<strong class='message'>Runtime in total is: 3 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'Rfam'
  @ontology: 'GOBP'
  @term_info: a data.frame of 30 terms X 5 information
  @anno: a list of 30 terms, each storing annotated domains
  @data: a vector containing a group of 67 input domains (annotatable)
  @background: a vector containing a group of 1377 background domains (annotatable)
  @overlap: a list of 30 terms, each containing domains overlapped with input domains
  @zscore: a vector of 30 terms, containing z-scores
  @pvalue: a vector of 30 terms, containing p-values
  @adjp: a vector of 30 terms, containing adjusted p-values
In summary, a total of 30 terms ('GOBP') are analysed for a group of 67 input domains ('Rfam')
</div>
<div class='input'>## 4c) view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=FALSE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue adjp
GO:0009987 GO:0009987  1299     67       66  1.510  0.018 0.29
GO:0000376 GO:0000376   522     67       32  1.700  0.035 0.29
GO:0000372 GO:0000372   522     67       32  1.700  0.035 0.29
GO:0010608 GO:0010608    27     67        3  1.520  0.039 0.29
GO:0000375 GO:0000375   541     67       32  1.460  0.058 0.29
GO:0008380 GO:0008380   542     67       32  1.440  0.059 0.29
GO:0044763 GO:0044763    34     67        3  1.090  0.079 0.29
GO:0010467 GO:0010467  1245     67       63  1.030  0.099 0.29
GO:0043170 GO:0043170  1252     67       63  0.907  0.130 0.29
GO:0044260 GO:0044260  1252     67       63  0.907  0.130 0.29
                                                                               term_name
GO:0009987                                                              cellular process
GO:0000376 RNA splicing, via transesterification reactions with guanosine as nucleophile
GO:0000372                                                       Group I intron splicing
GO:0010608                             posttranscriptional regulation of gene expression
GO:0000375                               RNA splicing, via transesterification reactions
GO:0008380                                                                  RNA splicing
GO:0044763                                              single-organism cellular process
GO:0010467                                                               gene expression
GO:0043170                                               macromolecule metabolic process
GO:0044260                                      cellular macromolecule metabolic process
</div>
<div class='input'>## 4d) visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOBP' containing 20 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-141.png', 'dcEnrichment-141.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-141.png', width="400" /></a>
<div class='input'>## 4e) the same as above but using a customised background
### randomly select 1000 RNAs as background
background <- sample(rowNames(Rfam), 1000)
### perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, background=background, domain="Rfam",
ontology="GOBP")
</div>
<strong class='message'>Start at 2014-12-05 17:17:58</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOBP', the domain 'Rfam', and their associations (2014-12-05 17:17:58) ...</strong>
<strong class='message'>'onto.GOBP' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Rfam' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'Rfam2GOBP' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2014-12-05 17:18:00) ...</strong>
<strong class='message'>	There are 72 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-12-05 17:18:00) ...</strong>
<strong class='message'>
End at 2014-12-05 17:18:00</strong>
<strong class='message'>Runtime in total is: 2 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'Rfam'
  @ontology: 'GOBP'
  @term_info: a data.frame of 30 terms X 5 information
  @anno: a list of 30 terms, each storing annotated domains
  @data: a vector containing a group of 67 input domains (annotatable)
  @background: a vector containing a group of 672 background domains (annotatable)
  @overlap: a list of 30 terms, each containing domains overlapped with input domains
  @zscore: a vector of 30 terms, containing z-scores
  @pvalue: a vector of 30 terms, containing p-values
  @adjp: a vector of 30 terms, containing adjusted p-values
In summary, a total of 30 terms ('GOBP') are analysed for a group of 67 input domains ('Rfam')
</div>
<div class='input'>### view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=FALSE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue adjp
GO:0009987 GO:0009987   635     67       66   1.52  0.018 0.23
GO:0000376 GO:0000376   253     67       32   1.80  0.028 0.23
GO:0000372 GO:0000372   253     67       32   1.80  0.028 0.23
GO:0008380 GO:0008380   260     67       32   1.61  0.042 0.23
GO:0000375 GO:0000375   260     67       32   1.61  0.042 0.23
GO:0010467 GO:0010467   599     67       63   1.36  0.049 0.23
GO:0010608 GO:0010608    15     67        3   1.31  0.053 0.23
GO:0043170 GO:0043170   604     67       63   1.19  0.072 0.24
GO:0044260 GO:0044260   604     67       63   1.19  0.072 0.24
GO:0044238 GO:0044238   609     67       63   1.01  0.100 0.27
                                                                               term_name
GO:0009987                                                              cellular process
GO:0000376 RNA splicing, via transesterification reactions with guanosine as nucleophile
GO:0000372                                                       Group I intron splicing
GO:0008380                                                                  RNA splicing
GO:0000375                               RNA splicing, via transesterification reactions
GO:0010467                                                               gene expression
GO:0010608                             posttranscriptional regulation of gene expression
GO:0043170                                               macromolecule metabolic process
GO:0044260                                      cellular macromolecule metabolic process
GO:0044238                                                     primary metabolic process
</div>
<div class='input'>### visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'GOBP' containing 14 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-160.png', 'dcEnrichment-160.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-160.png', width="400" /></a>
<div class='input'>
###########################################################
# 5) Advanced usage: customised data for domain, ontology and annotations
# 5a) create domain, ontology and annotations
## for domain
domain <-
dcBuildInfoDataFrame(input.file="http://dcgor.r-forge.r-project.org/data/InterPro/InterPro.txt",
output.file="domain.RData")
</div>
<strong class='message'>An object of S4 class 'InfoDataFrame' has been built and saved into '/Users/hfang/Sites/SUPERFAMILY/dcGOR/domain.RData'.</strong>
<div class='input'>## for ontology
dcBuildOnto(relations.file="http://dcgor.r-forge.r-project.org/data/onto/igraph_GOMF_edges.txt",
nodes.file="http://dcgor.r-forge.r-project.org/data/onto/igraph_GOMF_nodes.txt",
output.file="ontology.RData")
</div>
<strong class='message'>An object of S4 class 'Onto' has been built and saved into '/Users/hfang/Sites/SUPERFAMILY/dcGOR/ontology.RData'.</strong>
<div class='input'>## for annotations
dcBuildAnno(domain_info.file="http://dcgor.r-forge.r-project.org/data/InterPro/InterPro.txt",
term_info.file="http://dcgor.r-forge.r-project.org/data/InterPro/GO.txt",
association.file="http://dcgor.r-forge.r-project.org/data/InterPro/Domain2GOMF.txt",
output.file="annotations.RData")
</div>
<strong class='message'>An object of S4 class 'Anno' has been built and saved into '/Users/hfang/Sites/SUPERFAMILY/dcGOR/annotations.RData'.</strong>
<div class='input'>## 5b) prepare data and background
### randomly select 100 domains as a list of domains of interest
data <- sample(rowNames(domain), 100)
### randomly select 1000 domains as background
background <- sample(rowNames(domain), 1000)
## 5c) perform enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- dcEnrichment(data, background=background,
domain.RData='domain.RData', ontology.RData='ontology.RData',
annotations.RData='annotations.RData')
</div>
<strong class='message'>Start at 2014-12-05 17:18:07</strong>
<strong class='message'></strong>
<strong class='message'>First, load customised ontology 'ontology.RData', the domain 'domain.RData', and their associations 'annotations.RData' (2014-12-05 17:18:07)...</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2014-12-05 17:18:13) ...</strong>
<strong class='message'>	There are 147 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-12-05 17:18:13) ...</strong>
<strong class='message'>
End at 2014-12-05 17:18:13</strong>
<strong class='message'>Runtime in total is: 6 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'domain.RData'
  @ontology: 'ontology.RData'
  @term_info: a data.frame of 68 terms X 5 information
  @anno: a list of 68 terms, each storing annotated domains
  @data: a vector containing a group of 75 input domains (annotatable)
  @background: a vector containing a group of 861 background domains (annotatable)
  @overlap: a list of 68 terms, each containing domains overlapped with input domains
  @zscore: a vector of 68 terms, containing z-scores
  @pvalue: a vector of 68 terms, containing p-values
  @adjp: a vector of 68 terms, containing adjusted p-values
In summary, a total of 68 terms ('ontology.RData') are analysed for a group of 75 input domains ('domain.RData')
</div>
<div class='input'>## 5d) view the top 10 significance terms
view(eoutput, top_num=10, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue adjp
GO:0008757 GO:0008757    12     75        4   3.04 0.0021 0.15
GO:0008168 GO:0008168    15     75        4   2.49 0.0066 0.17
GO:0015081 GO:0015081    10     75        3   2.40 0.0075 0.17
GO:0016741 GO:0016741    17     75        4   2.19 0.0120 0.20
GO:0046873 GO:0046873    18     75        4   2.05 0.0150 0.20
GO:0043168 GO:0043168    56     75        9   2.02 0.0180 0.20
GO:0043167 GO:0043167    59     75        9   1.85 0.0260 0.20
GO:0016616 GO:0016616    35     75        6   1.81 0.0260 0.20
GO:0008289 GO:0008289    29     75        5   1.66 0.0340 0.20
GO:0032266 GO:0032266    22     75        4   1.59 0.0360 0.20
                                                                                       term_name
GO:0008757                             S-adenosylmethionine-dependent methyltransferase activity
GO:0008168                                                            methyltransferase activity
GO:0015081                                         sodium ion transmembrane transporter activity
GO:0016741                                  transferase activity, transferring one-carbon groups
GO:0046873                                          metal ion transmembrane transporter activity
GO:0043168                                                                         anion binding
GO:0043167                                                                           ion binding
GO:0016616 oxidoreductase activity, acting on the CH-OH group of donors, NAD or NADP as acceptor
GO:0008289                                                                         lipid binding
GO:0032266                                              phosphatidylinositol-3-phosphate binding
               term_namespace term_distance
GO:0008757 molecular_function             5
GO:0008168 molecular_function             4
GO:0015081 molecular_function             7
GO:0016741 molecular_function             3
GO:0046873 molecular_function             6
GO:0043168 molecular_function             3
GO:0043167 molecular_function             2
GO:0016616 molecular_function             4
GO:0008289 molecular_function             2
GO:0032266 molecular_function             7
                                                                                             members
GO:0008757                                                   IPR006225,IPR005775,IPR016231,IPR021824
GO:0008168                                                   IPR006225,IPR005775,IPR016231,IPR021824
GO:0015081                                                             IPR003613,IPR001367,IPR024928
GO:0016741                                                   IPR006225,IPR005775,IPR016231,IPR021824
GO:0046873                                                   IPR003613,IPR001367,IPR000133,IPR024928
GO:0043168 IPR006225,IPR010507,IPR006365,IPR000315,IPR000962,IPR013723,IPR023342,IPR002344,IPR010158
GO:0043167 IPR006225,IPR010507,IPR006365,IPR000315,IPR000962,IPR013723,IPR023342,IPR002344,IPR010158
GO:0016616                               IPR018044,IPR013249,IPR012850,IPR015833,IPR021824,IPR029703
GO:0008289                                         IPR006225,IPR013723,IPR023342,IPR002344,IPR021824
GO:0032266                                                   IPR006225,IPR013723,IPR023342,IPR002344
</div>
<div class='input'>### visualise the top 10 significant terms in the ontology hierarchy
### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
visEnrichment(eoutput)
</div>
<strong class='message'>Ontology 'ontology.RData' containing 14 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('dcEnrichment-182.png', 'dcEnrichment-182.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcEnrichment-182.png', width="400" /></a></pre>
  </div>
  
  <div class="span4">
    
    <!-- 
    <ul>
      <li>dcEnrichment</li>
    </ul>
    <ul>
      
    </ul> 
    -->
    
    <h2>Source code</h2>
    <a href="javascript:newWin('R/dcEnrichment.r', 'R/dcEnrichment.r', '1200', '1200')" title="Click to view"><B><code>dcEnrichment.r</code></B></a>
    
    <h2>Source man</h2>
    <a href="javascript:newWin('man/dcEnrichment.txt', 'man/dcEnrichment.txt', '1200', '1200')" title="Click to view"><B><code>dcEnrichment.Rd</code></B></a>
    
    <a href="javascript:newWin('man/dcEnrichment.pdf', 'man/dcEnrichment.pdf', '1200', '1200')" title="Click to view"><B><code>dcEnrichment.pdf</code></B></a>
    
    <h2>See also</h2>
    
<code><a href='dcRDataLoader.html'>dcRDataLoader</a></code>, <code><a href='dcDAGannotate.html'>dcDAGannotate</a></code>,
<code><a href='Eoutput-class.html'>Eoutput-class</a></code>, <code><a href='visEnrichment.html'>visEnrichment</a></code>,
<code><a href='dcConverter.html'>dcConverter</a></code>

    <!--

    <h2>Citation</h2>
    <p>Fang H. (2014) <B>dcGOR: an R package for analysing ontologies and protein domain annotations</B>. <I>PLoS Computational Biology</I> 10(10):e1003929. <a href="http://dx.doi.org/10.1371/journal.pcbi.1003929">doi:10.1371/journal.pcbi.1003929</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25356683">25356683</a></p>

<p>You may also cite the package as:</p>

<ul>
<li><code>database</code></li>
</ul>

<p>Fang H, Gough J. (2013) <B>dcGO: database of domain-centric ontologies on functions, phenotypes, diseases and more</B>. <I>Nucleic Acids Research</I> 41(Database issue):D536-44. <a href="http://dx.doi.org/10.1093/nar/gks1080">doi:10.1093/nar/gks1080</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23161684">23161684</a></p>

<ul>
<li><code>web link/URL</code></li>
</ul>

<p>Package &ldquo;dcGOR&rdquo; [<a href="http://cran.r-project.org/package=dcGOR">http://cran.r-project.org/package=dcGOR</a>]</p>

    
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://supfam.org/dcGOR">http://supfam.org/dcGOR</a></li>
      <li><a href=" http://dcgor.r-forge.r-project.org"> http://dcgor.r-forge.r-project.org</a></li>
      <li><a href=" http://cran.r-project.org/package=dcGOR"> http://cran.r-project.org/package=dcGOR</a></li>
      <li><a href=" https://github.com/hfang-bristol/dcGOR"> https://github.com/hfang-bristol/dcGOR</a></li>
    </ul>
    
    
    -->
    
  </div>
  
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>