<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>dcNaivePredict @ dcGOR 1.0.5</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dcGOR 1.0.5</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
       
        <!-- HTML for GOOLE SEARCH BAR 
        <ul class="nav">
            <div id="search-wrapper">
                <form id="search" method="get" action="http://www.google.com/custom">
                    <div class="search-box">
                        <i class="icon-search icon-white"></i>
                        <input type="text" id="tfq" class="textinput" name="q" size="21" maxlength="120" value="Google Search"/>
                    </div>
                </form>
            </div>
        </ul>
        -->
        
      </div>
    </div>
  </div>
</div>

<!-- CSS styles for standard search box with placeholder text-->
<style type="text/css">
    #search-wrapper {
        width:300px;
        height:40px;
        background-color:transparent;
        margin:0px auto 0;
        border:0px solid #ff0000;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

    #search {
        background-color:transparent;
        padding:0px;
        height:40px;
        overflow:hidden;
    }

    .search-box {
        width:250px;
        height:32px;
        background-color:transparent;
        margin:2px 7px;
        border:0px solid #cfcfcf;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

	.textinput{
		margin: 0;
		padding: 5px 15px;
		font-family: Arial, Helvetica, sans-serif;
		font-size:14px;
		color:#999999;
		border:1px solid #f00; border-right:0px;
		border-top-left-radius: 5px 5px;
		border-bottom-left-radius: 5px 5px;
	}
</style>

<!-- JAVASCRIPT to clear search text when the field is clicked -->
<script type="text/javascript">
window.onload = function(){
	//Get submit button
	var submitbutton = document.getElementById("tfq");
	//Add listener to submit button
	if(submitbutton.addEventListener){
		submitbutton.addEventListener("click", function() {
			if (submitbutton.value == 'Google Search'){//Customize this text string to whatever you want
				submitbutton.value = 'dcGOR R ';
			}
		});
	}
}
</script>

<script>
  (function() {
    var cx = '012094530736424854714:k8ieyz4ag4a';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') +
        '//www.google.com/cse/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>

<div><gcse:search></gcse:search></div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Function to perform naive prediction from input known annotations</h2>

<div class="row">
  <div class="span8">
  
    <h3>Description</h3>
    <p><code>dcNaivePredict</code> is supposed to perform naive prediction from
input known annotations. For each gene/protein, a term to be predicted
are simply the frequency of that term appearing in the known
annotations.</p>
  
    <h3>Usage</h3>
    <pre>dcNaivePredict(data, GSP.file, output.file = NULL, ontology = c(NA, "GOBP", "GOMF", 
  "GOCC", "DO", "HPPA", "HPMI", "HPON", "MP", "EC", "KW", "UP"), max.num = 1000, 
      verbose = T, RData.ontology.customised = NULL, RData.location = "http://dcgor.r-forge.r-project.org/data")</pre>
    
    <h3>Arguments</h3>
    <dl>
      <dt>data</dt>
      <dd>an input vector containing genes/proteins to be predicted</dd>
      <dt>GSP.file</dt>
      <dd>a Glod Standard Positive (GSP) file containing known
annotations between proteins/genes and ontology terms. For example, a
file containing annotations between human genes and HP terms can be
found in
<a href = 'http://dcgor.r-forge.r-project.org/data/Algo/HP_anno.txt'>http://dcgor.r-forge.r-project.org/data/Algo/HP_anno.txt</a>. As seen
in this example, the input file must contain the header (in the first
row) and two columns: 1st column for 'SeqID' (actually these IDs can be
anything), 2nd column for 'termID' (HP terms). Alternatively, the
GSP.file can be a matrix or data frame, assuming that GSP file has been
read. Note: the file should use the tab delimiter as the field
separator between columns</dd>
      <dt>output.file</dt>
      <dd>an output file containing predicted results. If not
NULL, a tab-delimited text file will be also written out; otherwise,
there is no output file (by default)</dd>
      <dt>ontology</dt>
      <dd>the ontology identity. It can be "GOBP" for Gene
Ontology Biological Process, "GOMF" for Gene Ontology Molecular
Function, "GOCC" for Gene Ontology Cellular Component, "DO" for Disease
Ontology, "HPPA" for Human Phenotype Phenotypic Abnormality, "HPMI" for
Human Phenotype Mode of Inheritance, "HPON" for Human Phenotype ONset
and clinical course, "MP" for Mammalian Phenotype, "EC" for Enzyme
Commission, "KW" for UniProtKB KeyWords, "UP" for UniProtKB UniPathway.
For details on the eligibility for pairs of input domain and ontology,
please refer to the online Documentations at
<a href = 'http://supfam.org/dcGOR/docs.html'>http://supfam.org/dcGOR/docs.html</a>. If NA, then the user has to
input a customised RData-formatted file (see
<code>RData.ontology.customised</code> below)</dd>
      <dt>max.num</dt>
      <dd>an integer to specify how many terms will be predicted
for each gene/protein</dd>
      <dt>verbose</dt>
      <dd>logical to indicate whether the messages will be
displayed in the screen. By default, it sets to TRUE for display</dd>
      <dt>RData.ontology.customised</dt>
      <dd>a file name for RData-formatted file
containing an object of S4 class 'Onto' (i.g. ontology). By default, it
is NULL. It is only needed when the user wants to perform customised
analysis using their own ontology. See <code><a href='dcBuildOnto.html'>dcBuildOnto</a></code> for
how to creat this object</dd>
      <dt>RData.location</dt>
      <dd>the characters to tell the location of built-in
RData files. By default, it remotely locates at
"http://supfam.org/dcGOR/data" or
"http://dcgor.r-forge.r-project.org/data". For the user equipped with
fast internet connection, this option can be just left as default. But
it is always advisable to download these files locally. Especially when
the user needs to run this function many times, there is no need to ask
the function to remotely download every time (also it will
unnecessarily increase the runtime). For examples, these files (as a
whole or part of them) can be first downloaded into your current
working directory, and then set this option as:
<code class = 'eq'>RData.location="."</code>. If RData to load is already part of package
itself, this parameter can be ignored (since this function will try to
load it via function <code>data</code> first)</dd>
    </dl>
    <!-- 
    -->
    
    <!-- 
    <div class="Value">
    -->
      <h3>Value</h3>
      
      <p>a data frame containing three columns: 1st column the same as the input
file (e.g. 'SeqID'), 2nd for 'Term' (predicted ontology terms), 3rd for
'Score' (along with predicted scores)</p>
    <!-- 
    </div>
    -->
    <!-- 
    <div class="Note">
    -->
      <h3>Note</h3>
      
      <p>When 'output.file' is specified, a tab-delimited text file is written
out, with the column names: 1st column the same as the input file (e.g.
'SeqID'), 2nd for 'Term' (predicted ontology terms), 3rd for 'Score'
(along with predicted scores).</p>
    <!-- 
    </div>
    -->
    
    <h3 id="examples">Examples</h3>
    <pre class="examples"><div class='input'># 1) prepare genes to be predicted
input.file <-
"http://dcgor.r-forge.r-project.org/data/Algo/HP_anno.txt"
#input.file <- "http://dcgor.r-forge.r-project.org/data/Algo/SCOP_architecture.txt"
input <- utils::read.delim(input.file, header=TRUE, sep="\t",
colClasses="character")
data <- unique(input[,1])

# 2) do naive prediction
GSP.file <- "http://dcgor.r-forge.r-project.org/data/Algo/HP_anno.txt"
res <- dcNaivePredict(data=data, GSP.file=GSP.file, ontology="HPPA")
</div>
<strong class='message'>Start at 2015-07-21 18:51:46</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'HPPA' (2015-07-21 18:51:46) ...</strong>
<strong class='message'>'onto.HPPA' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, import files for GSP (2015-07-21 18:51:46) ...</strong>
<strong class='message'>Third, propagate GSP annotations (2015-07-21 18:51:47) ...</strong>
<strong class='message'>	At level 16, there are 2 nodes, and 5 incoming neighbors.</strong>
<strong class='message'>	At level 15, there are 7 nodes, and 9 incoming neighbors.</strong>
<strong class='message'>	At level 14, there are 21 nodes, and 42 incoming neighbors.</strong>
<strong class='message'>	At level 13, there are 54 nodes, and 82 incoming neighbors.</strong>
<strong class='message'>	At level 12, there are 105 nodes, and 105 incoming neighbors.</strong>
<strong class='message'>	At level 11, there are 274 nodes, and 188 incoming neighbors.</strong>
<strong class='message'>	At level 10, there are 463 nodes, and 294 incoming neighbors.</strong>
<strong class='message'>	At level 9, there are 782 nodes, and 441 incoming neighbors.</strong>
<strong class='message'>	At level 8, there are 1004 nodes, and 538 incoming neighbors.</strong>
<strong class='message'>	At level 7, there are 1182 nodes, and 581 incoming neighbors.</strong>
<strong class='message'>	At level 6, there are 1295 nodes, and 527 incoming neighbors.</strong>
<strong class='message'>	At level 5, there are 940 nodes, and 290 incoming neighbors.</strong>
<strong class='message'>	At level 4, there are 408 nodes, and 99 incoming neighbors.</strong>
<strong class='message'>	At level 3, there are 114 nodes, and 21 incoming neighbors.</strong>
<strong class='message'>	At level 2, there are 21 nodes, and 1 incoming neighbors.</strong>
<strong class='message'>	At level 1, there are 1 nodes, and 0 incoming neighbors.</strong>
<strong class='message'>	There are 6673 terms in GSP (2015-07-21 18:52:42).</strong>
<strong class='message'>Fourth, do naive predictions for 3085 genes/proteins (2015-07-21 18:52:42) ...</strong>
<strong class='message'>	Focus on top 1000 predicted terms for each gene/protein</strong>
<strong class='message'>
End at 2015-07-21 18:52:44</strong>
<strong class='message'>Runtime in total is: 58 secs
</strong>
<div class='input'>res[1:10,]
</div>
<div class='output'>      SeqID    Term         Score   
 [1,] "10225"  "HP:0000118" "1"     
 [2,] "10806"  "HP:0000707" "0.6591"
 [3,] "11020"  "HP:0012638" "0.6053"
 [4,] "1131"   "HP:0000924" "0.5299"
 [5,] "123016" "HP:0000152" "0.5217"
 [6,] "129880" "HP:0011842" "0.5171"
 [7,] "1312"   "HP:0000234" "0.5138"
 [8,] "1376"   "HP:0000478" "0.5098"
 [9,] "139285" "HP:0012639" "0.4675"
[10,] "145173" "HP:0000271" "0.4583"
</div>
<div class='input'>
# 3) calculate Precision and Recall
res_PR <- dcAlgoPredictPR(GSP.file=GSP.file, prediction.file=res,
ontology="HPPA")
</div>
<strong class='message'>Start at 2015-07-21 18:52:44</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'HPPA' (2015-07-21 18:52:44) ...</strong>
<strong class='message'>'onto.HPPA' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, import files for GSP and predictions (2015-07-21 18:52:44) ...</strong>
<strong class='message'>Third, propagate GSP annotations (2015-07-21 18:52:46) ...</strong>
<strong class='message'>	At level 16, there are 2 nodes, and 5 incoming neighbors.</strong>
<strong class='message'>	At level 15, there are 7 nodes, and 9 incoming neighbors.</strong>
<strong class='message'>	At level 14, there are 21 nodes, and 42 incoming neighbors.</strong>
<strong class='message'>	At level 13, there are 54 nodes, and 82 incoming neighbors.</strong>
<strong class='message'>	At level 12, there are 105 nodes, and 105 incoming neighbors.</strong>
<strong class='message'>	At level 11, there are 274 nodes, and 188 incoming neighbors.</strong>
<strong class='message'>	At level 10, there are 463 nodes, and 294 incoming neighbors.</strong>
<strong class='message'>	At level 9, there are 782 nodes, and 441 incoming neighbors.</strong>
<strong class='message'>	At level 8, there are 1004 nodes, and 538 incoming neighbors.</strong>
<strong class='message'>	At level 7, there are 1182 nodes, and 581 incoming neighbors.</strong>
<strong class='message'>	At level 6, there are 1295 nodes, and 527 incoming neighbors.</strong>
<strong class='message'>	At level 5, there are 940 nodes, and 290 incoming neighbors.</strong>
<strong class='message'>	At level 4, there are 408 nodes, and 99 incoming neighbors.</strong>
<strong class='message'>	At level 3, there are 114 nodes, and 21 incoming neighbors.</strong>
<strong class='message'>	At level 2, there are 21 nodes, and 1 incoming neighbors.</strong>
<strong class='message'>	At level 1, there are 1 nodes, and 0 incoming neighbors.</strong>
<strong class='message'>	There are 3048 genes/proteins in GSP (2015-07-21 18:53:35).</strong>
<strong class='message'>Fourth, process input predictions (2015-07-21 18:53:35) ...</strong>
<strong class='message'>	There are 3085 genes/proteins in predictions (2015-07-21 18:53:42).</strong>
<strong class='message'>Fifth, calculate the precision and recall for each of 3048 predicted and GSP genes/proteins (2015-07-21 18:53:42).</strong>
<strong class='message'>Finally, calculate the averaged precision and recall (2015-07-21 18:53:44).</strong>
<strong class='message'>In summary, Prediction coverage: 1.00 (amongst 3048 targets in GSP), and F-measure: 0.35.</strong>
<strong class='message'>
End at 2015-07-21 18:53:44</strong>
<strong class='message'>Runtime in total is: 60 secs
</strong>
<div class='input'>res_PR
</div>
<div class='output'>          Precision     Recall
1        1.00000000 0.04034783
0.901476 1.00000000 0.04034783
0.802952 1.00000000 0.04034783
0.704428 1.00000000 0.04034783
0.605904 0.82956037 0.05303582
0.50738  0.60707841 0.09848956
0.408856 0.52151001 0.15324301
0.310332 0.44000040 0.23356765
0.211808 0.32728481 0.36977436
0.113284 0.23301021 0.53028895
0.01476  0.07161905 0.84493817
</div>
<div class='input'>
# 4) plot PR-curve
plot(res_PR[,2], res_PR[,1], xlim=c(0,1), ylim=c(0,1), type="b",
xlab="Recall", ylab="Precision")
</div>
<a href="javascript:newWin('dcNaivePredict-65.png', 'dcNaivePredict-65.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcNaivePredict-65.png', width="400" /></a></pre>
  </div>
  
  <div class="span4">
    
    <!-- 
    <ul>
      <li>dcNaivePredict</li>
    </ul>
    <ul>
      
    </ul> 
    -->
    
    <h2>Source code</h2>
    <a href="javascript:newWin('R/dcNaivePredict.r', 'R/dcNaivePredict.r', '1200', '1200')" title="Click to view"><B><code>dcNaivePredict.r</code></B></a>
    
    <h2>Source man</h2>
    <a href="javascript:newWin('man/dcNaivePredict.txt', 'man/dcNaivePredict.txt', '1200', '1200')" title="Click to view"><B><code>dcNaivePredict.Rd</code></B></a>
    
    <a href="javascript:newWin('man/dcNaivePredict.pdf', 'man/dcNaivePredict.pdf', '1200', '1200')" title="Click to view"><B><code>dcNaivePredict.pdf</code></B></a>
    
    <h2>See also</h2>
    
<code><a href='dcRDataLoader.html'>dcRDataLoader</a></code>, <code><a href='dcAlgoPropagate.html'>dcAlgoPropagate</a></code>

    <!--

    <h2>Citation</h2>
    <p>Fang H. (2014) <B>dcGOR: an R package for analysing ontologies and protein domain annotations</B>. <I>PLoS Computational Biology</I> 10(10):e1003929. <a href="http://dx.doi.org/10.1371/journal.pcbi.1003929">doi:10.1371/journal.pcbi.1003929</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25356683">25356683</a></p>

<p>You may also cite the package as:</p>

<ul>
<li><code>database</code></li>
</ul>

<p>Fang H, Gough J. (2013) <B>dcGO: database of domain-centric ontologies on functions, phenotypes, diseases and more</B>. <I>Nucleic Acids Research</I> 41(Database issue):D536-44. <a href="http://dx.doi.org/10.1093/nar/gks1080">doi:10.1093/nar/gks1080</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23161684">23161684</a></p>

<ul>
<li><code>web link/URL</code></li>
</ul>

<p>Package &ldquo;dcGOR&rdquo; [<a href="http://cran.r-project.org/package=dcGOR">http://cran.r-project.org/package=dcGOR</a>]</p>

    
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://supfam.org/dcGOR">http://supfam.org/dcGOR</a></li>
      <li><a href=" http://dcgor.r-forge.r-project.org"> http://dcgor.r-forge.r-project.org</a></li>
      <li><a href=" http://cran.r-
project.org/package=dcGOR"> http://cran.r-
project.org/package=dcGOR</a></li>
      <li><a href=" https://github.com/hfang-bristol/dcGOR"> https://github.com/hfang-bristol/dcGOR</a></li>
    </ul>
    
    
    -->
    
  </div>
  
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@well.ox.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>