<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>dcRWRpredict @ dcGOR 1.0.5</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dcGOR 1.0.5</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
       
        <!-- HTML for GOOLE SEARCH BAR 
        <ul class="nav">
            <div id="search-wrapper">
                <form id="search" method="get" action="http://www.google.com/custom">
                    <div class="search-box">
                        <i class="icon-search icon-white"></i>
                        <input type="text" id="tfq" class="textinput" name="q" size="21" maxlength="120" value="Google Search"/>
                    </div>
                </form>
            </div>
        </ul>
        -->
        
      </div>
    </div>
  </div>
</div>

<!-- CSS styles for standard search box with placeholder text-->
<style type="text/css">
    #search-wrapper {
        width:300px;
        height:40px;
        background-color:transparent;
        margin:0px auto 0;
        border:0px solid #ff0000;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

    #search {
        background-color:transparent;
        padding:0px;
        height:40px;
        overflow:hidden;
    }

    .search-box {
        width:250px;
        height:32px;
        background-color:transparent;
        margin:2px 7px;
        border:0px solid #cfcfcf;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

	.textinput{
		margin: 0;
		padding: 5px 15px;
		font-family: Arial, Helvetica, sans-serif;
		font-size:14px;
		color:#999999;
		border:1px solid #f00; border-right:0px;
		border-top-left-radius: 5px 5px;
		border-bottom-left-radius: 5px 5px;
	}
</style>

<!-- JAVASCRIPT to clear search text when the field is clicked -->
<script type="text/javascript">
window.onload = function(){
	//Get submit button
	var submitbutton = document.getElementById("tfq");
	//Add listener to submit button
	if(submitbutton.addEventListener){
		submitbutton.addEventListener("click", function() {
			if (submitbutton.value == 'Google Search'){//Customize this text string to whatever you want
				submitbutton.value = 'dcGOR R ';
			}
		});
	}
}
</script>

<script>
  (function() {
    var cx = '012094530736424854714:k8ieyz4ag4a';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') +
        '//www.google.com/cse/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>

<div><gcse:search></gcse:search></div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Function to perform RWR-based ontology term predictions from input known annotations and an input graph</h2>

<div class="row">
  <div class="span8">
  
    <h3>Description</h3>
    <p><code>dcRWRpredict</code> is supposed to perform ontology term predictions
based on Random Walk with Restart (RWR) from input known annotations
and an input graph.</p>
  
    <h3>Usage</h3>
    <pre>dcRWRpredict(data, g, output.file = NULL, ontology = c(NA, "GOBP", "GOMF", "GOCC", 
  "DO", "HPPA", "HPMI", "HPON", "MP", "EC", "KW", "UP"), method = c("indirect", 
      "direct"), normalise = c("laplacian", "row", "column", "none"), restart = 0.75, 
      normalise.affinity.matrix = c("none", "quantile"), leave.one.out = T, propagation = c("max", 
          "sum"), scale.method = c("log", "linear", "none"), parallel = TRUE, multicores = NULL, 
      verbose = T, RData.ontology.customised = NULL, RData.location = "http://dcgor.r-forge.r-project.org/data")</pre>
    
    <h3>Arguments</h3>
    <dl>
      <dt>data</dt>
      <dd>an input gene-term data matrix containing known annotations
used for seeds. Each value in input matrix does not necessarily have to
be binary (non-zeros will be used as a weight, but should be
non-negative for easy interpretation). Also, data can be a list, each
containing the known annotated genes</dd>
      <dt>g</dt>
      <dd>an object of class "igraph" or <code><a href='Dnetwork-class.html'>Dnetwork</a></code></dd>
      <dt>output.file</dt>
      <dd>an output file containing predicted results. If not
NULL, a tab-delimited text file will be also written out; otherwise,
there is no output file (by default)</dd>
      <dt>ontology</dt>
      <dd>the ontology identity. It can be "GOBP" for Gene
Ontology Biological Process, "GOMF" for Gene Ontology Molecular
Function, "GOCC" for Gene Ontology Cellular Component, "DO" for Disease
Ontology, "HPPA" for Human Phenotype Phenotypic Abnormality, "HPMI" for
Human Phenotype Mode of Inheritance, "HPON" for Human Phenotype ONset
and clinical course, "MP" for Mammalian Phenotype, "EC" for Enzyme
Commission, "KW" for UniProtKB KeyWords, "UP" for UniProtKB UniPathway.
For details on the eligibility for pairs of input domain and ontology,
please refer to the online Documentations at
<a href = 'http://supfam.org/dcGOR/docs.html'>http://supfam.org/dcGOR/docs.html</a>. If NA, then the user has to
input a customised RData-formatted file (see
<code>RData.ontology.customised</code> below)</dd>
      <dt>method</dt>
      <dd>the method used to calculate RWR. It can be 'direct' for
directly applying RWR, 'indirect' for indirectly applying RWR (first
pre-compute affinity matrix and then derive the affinity score)</dd>
      <dt>normalise</dt>
      <dd>the way to normalise the adjacency matrix of the input
graph. It can be 'laplacian' for laplacian normalisation, 'row' for
row-wise normalisation, 'column' for column-wise normalisation, or
'none'</dd>
      <dt>restart</dt>
      <dd>the restart probability used for RWR. The restart
probability takes the value from 0 to 1, controlling the range from the
starting nodes/seeds that the walker will explore. The higher the
value, the more likely the walker is to visit the nodes centered on the
starting nodes. At the extreme when the restart probability is zero,
the walker moves freely to the neighbors at each step without
restarting from seeds, i.e., following a random walk (RW)</dd>
      <dt>normalise.affinity.matrix</dt>
      <dd>the way to normalise the output
affinity matrix. It can be 'none' for no normalisation, 'quantile' for
quantile normalisation to ensure that columns (if multiple) of the
output affinity matrix have the same quantiles</dd>
      <dt>leave.one.out</dt>
      <dd>logical to indicate whether the leave-one-out test
is used for predictions. By default, it sets to true for doing
leave-one-out test (that is, known seeds are removed)</dd>
      <dt>propagation</dt>
      <dd>how to propagate the score. It can be "max" for
retaining the maximum score (by default), "sum" for additively
accumulating the score</dd>
      <dt>scale.method</dt>
      <dd>the method used to scale the predictive scores. It
can be: "none" for no scaling, "linear" for being linearily scaled into
the range between 0 and 1, "log" for the same as "linear" but being
first log-transformed before being scaled. The scaling between 0 and 1
is done via: <code class = 'eq'>\frac{S - S_{min}}{S_{max} - S_{min}}</code>, where
<code class = 'eq'>S_{min}</code> and <code class = 'eq'>S_{max}</code> are the minimum and maximum values for
<code class = 'eq'>S</code></dd>
      <dt>parallel</dt>
      <dd>logical to indicate whether parallel computation with
multicores is used. By default, it sets to true, but not necessarily
does so. Partly because parallel backends available will be
system-specific (now only Linux or Mac OS). Also, it will depend on
whether these two packages "foreach" and "doMC" have been installed. It
can be installed via:
<code>source("http://bioconductor.org/biocLite.R");
biocLite(c("foreach","doMC"))</code>. If not yet installed, this option will
be disabled</dd>
      <dt>multicores</dt>
      <dd>an integer to specify how many cores will be
registered as the multicore parallel backend to the 'foreach' package.
If NULL, it will use a half of cores available in a user's computer.
This option only works when parallel computation is enabled</dd>
      <dt>verbose</dt>
      <dd>logical to indicate whether the messages will be
displayed in the screen. By default, it sets to true for display</dd>
      <dt>RData.ontology.customised</dt>
      <dd>a file name for RData-formatted file
containing an object of S4 class 'Onto' (i.g. ontology). By default, it
is NULL. It is only needed when the user wants to perform customised
analysis using their own ontology. See <code><a href='dcBuildOnto.html'>dcBuildOnto</a></code> for
how to creat this object</dd>
      <dt>RData.location</dt>
      <dd>the characters to tell the location of built-in
RData files. By default, it remotely locates at
"http://supfam.org/dcGOR/data" or
"http://dcgor.r-forge.r-project.org/data". For the user equipped with
fast internet connection, this option can be just left as default. But
it is always advisable to download these files locally. Especially when
the user needs to run this function many times, there is no need to ask
the function to remotely download every time (also it will
unnecessarily increase the runtime). For examples, these files (as a
whole or part of them) can be first downloaded into your current
working directory, and then set this option as:
<code class = 'eq'>RData.location="."</code>. If RData to load is already part of package
itself, this parameter can be ignored (since this function will try to
load it via function <code>data</code> first)</dd>
    </dl>
    <!-- 
    -->
    
    <!-- 
    <div class="Value">
    -->
      <h3>Value</h3>
      
      <p>a data frame containing three columns: 1st column the same as the input
file (e.g. 'SeqID'), 2nd for 'Term' (predicted ontology terms), 3rd for
'Score' (along with predicted scores)</p>
    <!-- 
    </div>
    -->
    <!-- 
    <div class="Note">
    -->
      <h3>Note</h3>
      
      <p>When 'output.file' is specified, a tab-delimited text file is written
out, with the column names: 1st column the same as the input file (e.g.
'SeqID'), 2nd for 'Term' (predicted ontology terms), 3rd for 'Score'
(along with predicted scores).
The choice of which method to use RWR depends on the number of seed
sets and whether using leave-one-out test. If the total product of both
numbers are huge, it is better to use 'indrect' method (for a single
run). Also, when using leave-one-out test, it has to be use 'indrect'
method.</p>
    <!-- 
    </div>
    -->
    
    <h3 id="examples">Examples</h3>
    <pre class="examples"><div class='input'># 1) define an input network
## 1a) an igraph object that contains a functional protein association network in human.
### The network is extracted from the STRING database (version 9.1).
### Only those associations with medium confidence (score>=400) are retained
org.Hs.string <- dnet::dRDataLoader(RData='org.Hs.string')
</div>
<strong class='message'>'org.Hs.string' (from http://supfam.org/dnet/RData/1.0.7/org.Hs.string.RData) has been loaded into the working environment</strong>
<div class='input'>## 1b) restrict to those edges with confidence score>=999
### keep the largest connected component
network <- igraph::subgraph.edges(org.Hs.string,
eids=E(org.Hs.string)[combined_score>=999])
g <- dnet::dNetInduce(g=network, nodes_query=V(network)$name,
largest.comp=TRUE)
## Notably, in reality, 1b) can be replaced by:
#g <- igraph::subgraph.edges(org.Hs.string, eids=E(org.Hs.string)[combined_score>=400])
## 1c) make sure there is a 'weight' edge attribute
E(g)$weight <- E(g)$combined_score
### use EntrezGene ID as default 'name' node attribute
V(g)$name <- V(g)$geneid
g
</div>
<div class='output'>IGRAPH UNW- 2071 7603 -- 
+ attr: name (v/c), seqid (v/c), geneid (v/n), symbol (v/c),
| description (v/c), neighborhood_score (e/n), fusion_score (e/n),
| cooccurence_score (e/n), coexpression_score (e/n), experimental_score
| (e/n), database_score (e/n), textmining_score (e/n), combined_score
| (e/n), weight (e/n)
+ edges (vertex names):
 [1] 2288 --3320  1080 --3312  1080 --9368  1080 --7311  1407 --6500 
 [6] 1407 --8864  1407 --5187  1407 --8454  1407 --26224 1407 --1454 
[11] 1407 --8863  1407 --8945  1407 --406   5536 --3320  2067 --4436 
[16] 2072 --2067  2067 --7507  9821 --8408  9776 --9821  10533--11345
+ ... omitted several edges
</div>
<div class='input'>
# 2) define the known annotations as seeds
anno.file <- "http://dcgor.r-forge.r-project.org/data/Algo/HP_anno.txt"
data <- dcSparseMatrix(anno.file)
</div>
<strong class='message'>Read the input file 'http://dcgor.r-forge.r-project.org/data/Algo/HP_anno.txt' ...</strong>
<strong class='message'>There are 80781 entries, converted into a sparse matrix of 3085 X 5715.</strong>
<div class='input'>
# 3) perform RWR-based ontology term predictions
res <- dcRWRpredict(data=data, g=g, ontology="HPPA", parallel=FALSE)
</div>
<strong class='message'>Start at 2015-07-21 18:54:41</strong>
<strong class='message'></strong>
<strong class='message'>First, RWR on input graph (2071 nodes and 7603 edges) using input matrix (3085 rows and 5715 columns) as seeds (2015-07-21 18:54:41)...</strong>
<strong class='message'>	using 'indirect' method to do RWR with leave-one-out test (2015-07-21 18:54:42)...</strong>
<strong class='message'>
##############################</strong>
<strong class='message'>'dnet::dRWR' is being called...</strong>
<strong class='message'>##############################
</strong>
<strong class='message'>Start at 2015-07-21 18:54:42</strong>
<strong class='message'></strong>
<strong class='message'>First, get the adjacency matrix of the input graph (2015-07-21 18:54:42) ...</strong>
<strong class='message'>	Notes: using weighted graph!</strong>
<strong class='message'>Then, normalise the adjacency matrix using laplacian normalisation (2015-07-21 18:54:42) ...</strong>
<strong class='message'>Third, RWR of 2071 sets of seeds using 7.5e-01 restart probability (2015-07-21 18:54:43) ...</strong>
<strong class='message'>	1 out of 2071 seed sets (2015-07-21 18:54:43)</strong>
<strong class='message'>	208 out of 2071 seed sets (2015-07-21 18:54:56)</strong>
<strong class='message'>	416 out of 2071 seed sets (2015-07-21 18:55:11)</strong>
<strong class='message'>	624 out of 2071 seed sets (2015-07-21 18:55:27)</strong>
<strong class='message'>	832 out of 2071 seed sets (2015-07-21 18:55:45)</strong>
<strong class='message'>	1040 out of 2071 seed sets (2015-07-21 18:56:04)</strong>
<strong class='message'>	1248 out of 2071 seed sets (2015-07-21 18:56:27)</strong>
<strong class='message'>	1456 out of 2071 seed sets (2015-07-21 18:56:54)</strong>
<strong class='message'>	1664 out of 2071 seed sets (2015-07-21 18:57:20)</strong>
<strong class='message'>	1872 out of 2071 seed sets (2015-07-21 18:57:49)</strong>
<strong class='message'>	2071 out of 2071 seed sets (2015-07-21 18:58:21)</strong>
<strong class='message'>Fourth, rescale steady probability vector (2015-07-21 18:58:21) ...</strong>
<strong class='message'>Finally, output 2071 by 2071 affinity matrix normalised by none (2015-07-21 18:58:21) ...</strong>
<strong class='message'>
Finish at 2015-07-21 18:58:23</strong>
<strong class='message'>Runtime in total is: 221 secs
</strong>
<strong class='message'>##############################</strong>
<strong class='message'>'dnet::dRWR' has been completed!</strong>
<strong class='message'>##############################
</strong>
<strong class='message'>Second, propagate 'HPPA' ontology annotations via 'max' operation (2015-07-21 18:58:26)...Second, propagate 'HPPA' ontology annotations via 'sum' operation (2015-07-21 18:58:26)...</strong>
<strong class='message'>
##############################</strong>
<strong class='message'>'dcAlgoPropagate' is being called...</strong>
<strong class='message'>##############################
</strong>
<strong class='message'>Start at 2015-07-21 18:58:33</strong>
<strong class='message'></strong>
<strong class='message'>Load the input file (2015-07-21 18:58:33) ...</strong>
<strong class='message'>Load the ontology 'HPPA' (2015-07-21 18:58:38) ...</strong>
<strong class='message'>'onto.HPPA' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Do propagation via 'max' operation (2015-07-21 18:58:40) ...</strong>
<strong class='message'>	At level 16, there are 2 nodes, and 5 incoming neighbors (2015-07-21 18:58:42).</strong>
<strong class='message'>	At level 15, there are 5 nodes, and 9 incoming neighbors (2015-07-21 18:58:42).</strong>
<strong class='message'>	At level 14, there are 15 nodes, and 32 incoming neighbors (2015-07-21 18:58:43).</strong>
<strong class='message'>	At level 13, there are 36 nodes, and 60 incoming neighbors (2015-07-21 18:58:44).</strong>
<strong class='message'>	At level 12, there are 63 nodes, and 66 incoming neighbors (2015-07-21 18:58:46).</strong>
<strong class='message'>	At level 11, there are 125 nodes, and 114 incoming neighbors (2015-07-21 18:58:49).</strong>
<strong class='message'>	At level 10, there are 230 nodes, and 182 incoming neighbors (2015-07-21 18:58:54).</strong>
<strong class='message'>	At level 9, there are 396 nodes, and 279 incoming neighbors (2015-07-21 18:59:01).</strong>
<strong class='message'>	At level 8, there are 557 nodes, and 370 incoming neighbors (2015-07-21 18:59:13).</strong>
<strong class='message'>	At level 7, there are 667 nodes, and 381 incoming neighbors (2015-07-21 18:59:26).</strong>
<strong class='message'>	At level 6, there are 731 nodes, and 361 incoming neighbors (2015-07-21 18:59:41).</strong>
<strong class='message'>	At level 5, there are 583 nodes, and 220 incoming neighbors (2015-07-21 18:59:52).</strong>
<strong class='message'>	At level 4, there are 290 nodes, and 89 incoming neighbors (2015-07-21 18:59:56).</strong>
<strong class='message'>	At level 3, there are 100 nodes, and 21 incoming neighbors (2015-07-21 18:59:58).</strong>
<strong class='message'>	At level 2, there are 21 nodes, and 1 incoming neighbors (2015-07-21 18:59:58).</strong>
<strong class='message'>	after propagation, there are 2056 features annotated by 3822 terms.</strong>
<strong class='message'>Determining IC-based slim levels (2015-07-21 19:00:10) ...</strong>
<strong class='message'>	1 level with 647 terms with IC falling around 0.01 (between 0.00 and 0.02).</strong>
<strong class='message'>	2 level with 0 terms with IC falling around 0.03 (between 0.03 and 0.04).</strong>
<strong class='message'>	3 level with 0 terms with IC falling around 0.05 (between 0.05 and 0.06).</strong>
<strong class='message'>	4 level with 0 terms with IC falling around 0.08 (between 0.07 and 0.08).</strong>
<strong class='message'>
End at 2015-07-21 19:11:23</strong>
<strong class='message'>Runtime in total is: 770 secs
</strong>
<strong class='message'>##############################</strong>
<strong class='message'>'dcAlgoPropagate' has been completed!</strong>
<strong class='message'>##############################
</strong>
<strong class='message'>Third, rescale predictive score using 'log' method (2015-07-21 19:11:23)...</strong>
<strong class='message'>The input list has been converted into a matrix of 7868533 X 3.</strong>
<strong class='message'></strong>
<strong class='message'>Finish at 2015-07-21 19:11:34</strong>
<strong class='message'>Runtime in total is: 1013 secs
</strong>
<div class='input'>res[1:10,]
</div>
<div class='output'>      SeqID  Term         Score   
 [1,] "1000" "HP:0000118" "1"     
 [2,] "1000" "HP:0000119" "0.9202"
 [3,] "1000" "HP:0000152" "0.9519"
 [4,] "1000" "HP:0000478" "0.9079"
 [5,] "1000" "HP:0000598" "0.9519"
 [6,] "1000" "HP:0000707" "0.8922"
 [7,] "1000" "HP:0000769" "0.8509"
 [8,] "1000" "HP:0000818" "0.8692"
 [9,] "1000" "HP:0000924" "0.9585"
[10,] "1000" "HP:0001197" "0.8663"
</div>
<div class='input'># 4) calculate Precision and Recall
GSP.file <- anno.file
prediction.file <- res
res_PR <- dcAlgoPredictPR(GSP.file=GSP.file,
prediction.file=prediction.file, ontology="HPPA")
</div>
<strong class='message'>Start at 2015-07-21 19:11:35</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'HPPA' (2015-07-21 19:11:35) ...</strong>
<strong class='message'>'onto.HPPA' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, import files for GSP and predictions (2015-07-21 19:11:35) ...</strong>
<strong class='message'>Third, propagate GSP annotations (2015-07-21 19:11:39) ...</strong>
<strong class='message'>	At level 16, there are 2 nodes, and 5 incoming neighbors.</strong>
<strong class='message'>	At level 15, there are 7 nodes, and 9 incoming neighbors.</strong>
<strong class='message'>	At level 14, there are 21 nodes, and 42 incoming neighbors.</strong>
<strong class='message'>	At level 13, there are 54 nodes, and 82 incoming neighbors.</strong>
<strong class='message'>	At level 12, there are 105 nodes, and 105 incoming neighbors.</strong>
<strong class='message'>	At level 11, there are 274 nodes, and 188 incoming neighbors.</strong>
<strong class='message'>	At level 10, there are 463 nodes, and 294 incoming neighbors.</strong>
<strong class='message'>	At level 9, there are 782 nodes, and 441 incoming neighbors.</strong>
<strong class='message'>	At level 8, there are 1004 nodes, and 538 incoming neighbors.</strong>
<strong class='message'>	At level 7, there are 1182 nodes, and 581 incoming neighbors.</strong>
<strong class='message'>	At level 6, there are 1295 nodes, and 527 incoming neighbors.</strong>
<strong class='message'>	At level 5, there are 940 nodes, and 290 incoming neighbors.</strong>
<strong class='message'>	At level 4, there are 408 nodes, and 99 incoming neighbors.</strong>
<strong class='message'>	At level 3, there are 114 nodes, and 21 incoming neighbors.</strong>
<strong class='message'>	At level 2, there are 21 nodes, and 1 incoming neighbors.</strong>
<strong class='message'>	At level 1, there are 1 nodes, and 0 incoming neighbors.</strong>
<strong class='message'>	There are 3048 genes/proteins in GSP (2015-07-21 19:12:26).</strong>
<strong class='message'>Fourth, process input predictions (2015-07-21 19:12:26) ...</strong>
<strong class='message'>	There are 2056 genes/proteins in predictions (2015-07-21 19:12:39).</strong>
<strong class='message'>Fifth, calculate the precision and recall for each of 488 predicted and GSP genes/proteins (2015-07-21 19:12:39).</strong>
<strong class='message'>Finally, calculate the averaged precision and recall (2015-07-21 19:12:40).</strong>
<strong class='message'>In summary, Prediction coverage: 0.16 (amongst 3048 targets in GSP), and F-measure: 0.12.</strong>
<strong class='message'>
End at 2015-07-21 19:12:40</strong>
<strong class='message'>Runtime in total is: 65 secs
</strong>
<div class='input'>res_PR
</div>
<div class='output'>     Precision     Recall
1   0.34207485 0.01806081
0.9 0.20221296 0.07321846
0.8 0.12302258 0.11590770
0.7 0.07294678 0.13697636
0.6 0.04637731 0.14749228
0.5 0.03358901 0.15280354
0.4 0.02898669 0.15491591
0.3 0.02810094 0.15542853
0.2 0.02776679 0.15556983
0.1 0.02763143 0.15561603
0   0.02758970 0.15562458
</div>
<div class='input'>
# 5) Plot PR-curve
plot(res_PR[,2], res_PR[,1], xlim=c(0,1), ylim=c(0,1), type="b",
xlab="Recall", ylab="Precision")
</div>
<a href="javascript:newWin('dcRWRpredict-117.png', 'dcRWRpredict-117.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dcRWRpredict-117.png', width="400" /></a></pre>
  </div>
  
  <div class="span4">
    
    <!-- 
    <ul>
      <li>dcRWRpredict</li>
    </ul>
    <ul>
      
    </ul> 
    -->
    
    <h2>Source code</h2>
    <a href="javascript:newWin('R/dcRWRpredict.r', 'R/dcRWRpredict.r', '1200', '1200')" title="Click to view"><B><code>dcRWRpredict.r</code></B></a>
    
    <h2>Source man</h2>
    <a href="javascript:newWin('man/dcRWRpredict.txt', 'man/dcRWRpredict.txt', '1200', '1200')" title="Click to view"><B><code>dcRWRpredict.Rd</code></B></a>
    
    <a href="javascript:newWin('man/dcRWRpredict.pdf', 'man/dcRWRpredict.pdf', '1200', '1200')" title="Click to view"><B><code>dcRWRpredict.pdf</code></B></a>
    
    <h2>See also</h2>
    
<code><a href='dcRDataLoader.html'>dcRDataLoader</a></code>, <code><a href='dcAlgoPropagate.html'>dcAlgoPropagate</a></code>,
<code><a href='dcList2Matrix.html'>dcList2Matrix</a></code>

    <!--

    <h2>Citation</h2>
    <p>Fang H. (2014) <B>dcGOR: an R package for analysing ontologies and protein domain annotations</B>. <I>PLoS Computational Biology</I> 10(10):e1003929. <a href="http://dx.doi.org/10.1371/journal.pcbi.1003929">doi:10.1371/journal.pcbi.1003929</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25356683">25356683</a></p>

<p>You may also cite the package as:</p>

<ul>
<li><code>database</code></li>
</ul>

<p>Fang H, Gough J. (2013) <B>dcGO: database of domain-centric ontologies on functions, phenotypes, diseases and more</B>. <I>Nucleic Acids Research</I> 41(Database issue):D536-44. <a href="http://dx.doi.org/10.1093/nar/gks1080">doi:10.1093/nar/gks1080</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23161684">23161684</a></p>

<ul>
<li><code>web link/URL</code></li>
</ul>

<p>Package &ldquo;dcGOR&rdquo; [<a href="http://cran.r-project.org/package=dcGOR">http://cran.r-project.org/package=dcGOR</a>]</p>

    
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://supfam.org/dcGOR">http://supfam.org/dcGOR</a></li>
      <li><a href=" http://dcgor.r-forge.r-project.org"> http://dcgor.r-forge.r-project.org</a></li>
      <li><a href=" http://cran.r-
project.org/package=dcGOR"> http://cran.r-
project.org/package=dcGOR</a></li>
      <li><a href=" https://github.com/hfang-bristol/dcGOR"> https://github.com/hfang-bristol/dcGOR</a></li>
    </ul>
    
    
    -->
    
  </div>
  
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@well.ox.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>