<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>Demo for creating and using dcGO @ dcGOR 1.0.5</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="Hai Fang and Julian Gough">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dcGOR 1.0.5</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
       
        <!-- HTML for GOOLE SEARCH BAR 
        <ul class="nav">
            <div id="search-wrapper">
                <form id="search" method="get" action="http://www.google.com/custom">
                    <div class="search-box">
                        <i class="icon-search icon-white"></i>
                        <input type="text" id="tfq" class="textinput" name="q" size="21" maxlength="120" value="Google Search"/>
                    </div>
                </form>
            </div>
        </ul>
        -->
        
      </div>
    </div>
  </div>
</div>

<!-- CSS styles for standard search box with placeholder text-->
<style type="text/css">
    #search-wrapper {
        width:300px;
        height:40px;
        background-color:transparent;
        margin:0px auto 0;
        border:0px solid #ff0000;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

    #search {
        background-color:transparent;
        padding:0px;
        height:40px;
        overflow:hidden;
    }

    .search-box {
        width:250px;
        height:32px;
        background-color:transparent;
        margin:2px 7px;
        border:0px solid #cfcfcf;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

	.textinput{
		margin: 0;
		padding: 5px 15px;
		font-family: Arial, Helvetica, sans-serif;
		font-size:14px;
		color:#999999;
		border:1px solid #f00; border-right:0px;
		border-top-left-radius: 5px 5px;
		border-bottom-left-radius: 5px 5px;
	}
</style>

<!-- JAVASCRIPT to clear search text when the field is clicked -->
<script type="text/javascript">
window.onload = function(){
	//Get submit button
	var submitbutton = document.getElementById("tfq");
	//Add listener to submit button
	if(submitbutton.addEventListener){
		submitbutton.addEventListener("click", function() {
			if (submitbutton.value == 'Google Search'){//Customize this text string to whatever you want
				submitbutton.value = 'dcGOR R ';
			}
		});
	}
}
</script>

<script>
  (function() {
    var cx = '012094530736424854714:k8ieyz4ag4a';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') +
        '//www.google.com/cse/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>

<div><gcse:search></gcse:search></div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Demo for creating and using dcGO</h2>

Notes:
<li> All results are based on dcGOR (version 1.0.5). </li>
<li> R scripts (i.e. R expressions) plus necessary comments are highlighted in light cyan, and the rest are outputs in the screen. </li>
<li> Images displayed below may be distorted, but should be normal in your screen. </li>
<li> Functions contained in dcGOR 1.0.5 are hyperlinked in-place and also listed on the right side. </li>

<div class="row">
    <div class="span8">
      <pre>
      <div class='input'># This is a demo for creating and using dcGO
# 
# There are two key concepts behind dcGO. The first concept is to label protein domains with ontology, for example, with Gene Ontology. That is why it is called dcGO, domain-centric Gene Ontology. The second concept is to use ontology-labeled protein domains for, for example, protein function prediction. Put it in a simple way, the first concept is about how to create dcGO resource, and the second concept is about how to use dcGO resource.
# Accordingly, this demo contains two parts. The first part is to creat/label SCOP domains/supra-domains with Human Phenotype (HP) Ontology (actually a subontology called Phenotypic Abnormality: HPPA). The second part is to use the built resource for phenotype predictions in human genes. 
# The demo assumes that the user has collected two bits (files) of information:
## 1) anno.file: an annotation file containing annotations between proteins/genes and ontology terms. For example, a file containing annotations between human genes and HP terms can be found in <a href="http://dcgor.r-forge.r-project.org/data/Algo/HP_anno.txt">HP_anno.txt</a>, with two columns (1st for 'SeqID', 2nd for 'termID')
## 2) architecture.file: an architecture file containing domain architectures (including individual domains) for proteins/genes. For example, a file containing human genes and domain architectures can be found in <a href="http://dcgor.r-forge.r-project.org/data/Algo/SCOP_architecture.txt">SCOP_architecture.txt</a>, with two columns (1st for 'SeqID', 2nd for 'Architecture', that is, a string of comma-separated SCOP domains)
#
###############################################################################
library(dcGOR)


#---------------------------------------------------------------------------
#---------------------------------------------------------------------------
# Creat domains/supra-domains with Human Phenotype Phenotypic Abnormality (HPPA) terms
#---------------------------------------------------------------------------
#---------------------------------------------------------------------------

## 1) prepare input and output files
### input files
anno.file <- "http://dcgor.r-forge.r-project.org/data/Algo/HP_anno.txt"
architecture.file <- "http://dcgor.r-forge.r-project.org/data/Algo/SCOP_architecture.txt"
### output files
output.file <- "sf2HPPA.txt"
RData.HIS.customised <- "sf2HPPA.HIS.RData"
## 2) apply dcGO algorithm to infer domain-centric ontology
res <- <a href='dcAlgo.html' target=dcAlgo>dcAlgo</a>(anno.file, architecture.file, output.file, ontology="HPPA", feature.mode="supra", fdr.cutoff=0.05, parallel=FALSE)
</div>
<strong class='message'>Start at 2014-12-23 14:56:07</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'HPPA' (2014-12-23 14:56:07) ...</strong>
<strong class='message'>'onto.HPPA' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, import files for annotations 'http://dcgor.r-forge.r-project.org/data/Algo/HP_anno.txt' and architectures 'http://dcgor.r-forge.r-project.org/data/Algo/SCOP_architecture.txt' (2014-12-23 14:56:07) ...</strong>
<strong class='message'>Third, propagate annotations (2014-12-23 14:56:09) ...</strong>
<strong class='message'>	At level 16, there are 2 nodes, and 5 incoming neighbors.</strong>
<strong class='message'>	At level 15, there are 7 nodes, and 9 incoming neighbors.</strong>
<strong class='message'>	At level 14, there are 21 nodes, and 42 incoming neighbors.</strong>
<strong class='message'>	At level 13, there are 54 nodes, and 82 incoming neighbors.</strong>
<strong class='message'>	At level 12, there are 105 nodes, and 105 incoming neighbors.</strong>
<strong class='message'>	At level 11, there are 274 nodes, and 188 incoming neighbors.</strong>
<strong class='message'>	At level 10, there are 463 nodes, and 294 incoming neighbors.</strong>
<strong class='message'>	At level 9, there are 782 nodes, and 441 incoming neighbors.</strong>
<strong class='message'>	At level 8, there are 1004 nodes, and 538 incoming neighbors.</strong>
<strong class='message'>	At level 7, there are 1182 nodes, and 581 incoming neighbors.</strong>
<strong class='message'>	At level 6, there are 1295 nodes, and 527 incoming neighbors.</strong>
<strong class='message'>	At level 5, there are 940 nodes, and 290 incoming neighbors.</strong>
<strong class='message'>	At level 4, there are 408 nodes, and 99 incoming neighbors.</strong>
<strong class='message'>	At level 3, there are 114 nodes, and 21 incoming neighbors.</strong>
<strong class='message'>	At level 2, there are 21 nodes, and 1 incoming neighbors.</strong>
<strong class='message'>	At level 1, there are 1 nodes, and 0 incoming neighbors.</strong>
<strong class='message'>	There are 6673 terms used (2014-12-23 14:56:32).</strong>
<strong class='message'>Fourth, define groups using feature mode 'supra' (2014-12-23 14:56:32) ...</strong>
<strong class='message'>	i) split into features (2014-12-23 14:56:32) ...</strong>
<strong class='message'>	1 out of 4351 (2014-12-23 14:56:32)</strong>
<strong class='message'>	436 out of 4351 (2014-12-23 14:56:32)</strong>
<strong class='message'>	872 out of 4351 (2014-12-23 14:56:32)</strong>
<strong class='message'>	1308 out of 4351 (2014-12-23 14:56:33)</strong>
<strong class='message'>	1744 out of 4351 (2014-12-23 14:56:33)</strong>
<strong class='message'>	2180 out of 4351 (2014-12-23 14:56:34)</strong>
<strong class='message'>	2616 out of 4351 (2014-12-23 14:56:34)</strong>
<strong class='message'>	3052 out of 4351 (2014-12-23 14:56:34)</strong>
<strong class='message'>	3488 out of 4351 (2014-12-23 14:56:35)</strong>
<strong class='message'>	3924 out of 4351 (2014-12-23 14:56:35)</strong>
<strong class='message'>	4351 out of 4351 (2014-12-23 14:56:36)</strong>
<strong class='message'>	ii) obtain feature-based groups (2014-12-23 14:56:36) ...</strong>
<strong class='message'>	1 out of 4351 (2014-12-23 14:56:36)</strong>
<strong class='message'>	436 out of 4351 (2014-12-23 14:56:36)</strong>
<strong class='message'>	872 out of 4351 (2014-12-23 14:56:36)</strong>
<strong class='message'>	1308 out of 4351 (2014-12-23 14:56:36)</strong>
<strong class='message'>	1744 out of 4351 (2014-12-23 14:56:36)</strong>
<strong class='message'>	2180 out of 4351 (2014-12-23 14:56:36)</strong>
<strong class='message'>	2616 out of 4351 (2014-12-23 14:56:36)</strong>
<strong class='message'>	3052 out of 4351 (2014-12-23 14:56:36)</strong>
<strong class='message'>	3488 out of 4351 (2014-12-23 14:56:36)</strong>
<strong class='message'>	3924 out of 4351 (2014-12-23 14:56:36)</strong>
<strong class='message'>	4351 out of 4351 (2014-12-23 14:56:36)</strong>
<strong class='message'>	There are 2194 features used (2014-12-23 14:56:36).</strong>
<strong class='message'>Finally, estimate associations between 2194 features and 6673 terms, with 3 min overlaps and 5.0e-02 fdr cutoff (2014-12-23 14:56:36) ...</strong>
<strong class='message'>	1 out of 2194 (2014-12-23 14:56:41)</strong>
<strong class='message'>	220 out of 2194 (2014-12-23 14:56:42)</strong>
<strong class='message'>	440 out of 2194 (2014-12-23 14:56:43)</strong>
<strong class='message'>	660 out of 2194 (2014-12-23 14:56:44)</strong>
<strong class='message'>	880 out of 2194 (2014-12-23 14:56:45)</strong>
<strong class='message'>	1100 out of 2194 (2014-12-23 14:56:46)</strong>
<strong class='message'>	1320 out of 2194 (2014-12-23 14:56:47)</strong>
<strong class='message'>	1540 out of 2194 (2014-12-23 14:56:49)</strong>
<strong class='message'>	1760 out of 2194 (2014-12-23 14:56:50)</strong>
<strong class='message'>	1980 out of 2194 (2014-12-23 14:56:51)</strong>
<strong class='message'>	2194 out of 2194 (2014-12-23 14:56:52)</strong>
<strong class='message'>The results have been saved into '/Users/hfang/Sites/SUPERFAMILY/dcGOR/sf2HPPA.txt'.</strong>
<strong class='message'>
End at 2014-12-23 14:56:52</strong>
<strong class='message'>Runtime in total is: 45 secs
</strong>
<div class='input'>res[1:5,]
</div>
<div class='output'>  Feature_id    Term_id Score
1     100895 HP:0004370  3.00
2     100895 HP:0002027  4.40
3     100895 HP:0011123  0.82
4     100895 HP:0004295  5.00
5     100895 HP:0002024  2.20
</div>
<div class='input'>## 3) propagate ontology annotations
res_RData <- <a href='dcAlgoPropagate.html' target=dcAlgoPropagate>dcAlgoPropagate</a>(input.file=output.file, ontology="HPPA", output.file=RData.HIS.customised)
</div>
<strong class='message'>Start at 2014-12-23 14:56:52</strong>
<strong class='message'></strong>
<strong class='message'>Read the input file 'sf2HPPA.txt' (2014-12-23 14:56:52) ...</strong>
<strong class='message'>Load the ontology 'HPPA' (2014-12-23 14:56:52) ...</strong>
<strong class='message'>'onto.HPPA' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Do propagation via 'max' operation (2014-12-23 14:56:53) ...</strong>
<strong class='message'>	At level 14, there are 1 nodes, and 3 incoming neighbors (2014-12-23 14:56:53).</strong>
<strong class='message'>	At level 13, there are 2 nodes, and 7 incoming neighbors (2014-12-23 14:56:53).</strong>
<strong class='message'>	At level 12, there are 10 nodes, and 18 incoming neighbors (2014-12-23 14:56:53).</strong>
<strong class='message'>	At level 11, there are 19 nodes, and 26 incoming neighbors (2014-12-23 14:56:53).</strong>
<strong class='message'>	At level 10, there are 68 nodes, and 74 incoming neighbors (2014-12-23 14:56:53).</strong>
<strong class='message'>	At level 9, there are 118 nodes, and 100 incoming neighbors (2014-12-23 14:56:53).</strong>
<strong class='message'>	At level 8, there are 174 nodes, and 137 incoming neighbors (2014-12-23 14:56:53).</strong>
<strong class='message'>	At level 7, there are 240 nodes, and 181 incoming neighbors (2014-12-23 14:56:54).</strong>
<strong class='message'>	At level 6, there are 290 nodes, and 180 incoming neighbors (2014-12-23 14:56:54).</strong>
<strong class='message'>	At level 5, there are 300 nodes, and 134 incoming neighbors (2014-12-23 14:56:55).</strong>
<strong class='message'>	At level 4, there are 172 nodes, and 65 incoming neighbors (2014-12-23 14:56:55).</strong>
<strong class='message'>	At level 3, there are 76 nodes, and 21 incoming neighbors (2014-12-23 14:56:55).</strong>
<strong class='message'>	At level 2, there are 21 nodes, and 1 incoming neighbors (2014-12-23 14:56:55).</strong>
<strong class='message'>	after propagation, there are 460 features annotated by 1492 terms.</strong>
<strong class='message'>Determining IC-based slim levels (2014-12-23 14:56:55) ...</strong>
<strong class='message'>	1 level with 16 terms with IC falling around 0.33 (between 0.00 and 0.67).</strong>
<strong class='message'>	2 level with 62 terms with IC falling around 1.00 (between 0.83 and 1.16).</strong>
<strong class='message'>	3 level with 149 terms with IC falling around 1.66 (between 1.50 and 1.83).</strong>
<strong class='message'>	4 level with 267 terms with IC falling around 2.33 (between 2.16 and 2.50).</strong>
<strong class='message'>An object of S3 class 'HIS' has been built and saved into '/Users/hfang/Sites/SUPERFAMILY/dcGOR/sf2HPPA.HIS.RData'.</strong>
<strong class='message'>
End at 2014-12-23 14:57:00</strong>
<strong class='message'>Runtime in total is: 8 secs
</strong>
<div class='input'>## In your working directory, you should see a RData-formatted file: "sf2HPPA.HIS.RData". It is an object of S3 class 'HIS' (see <a href="http://supfam.org/dcGOR/<a href='dcAlgoPropagate.html' target=dcAlgoPropagate>dcAlgoPropagate</a>.html"><a href='dcAlgoPropagate.html' target=dcAlgoPropagate>dcAlgoPropagate</a></a> for details). It contains created domain-centric annotations, which will be used subsequently for prediction.
names(res_RData)
</div>
<div class='output'>[1] "hscore" "ic"     "slim"  
</div>
<div class='input'>
#---------------------------------------------------------------------------
#---------------------------------------------------------------------------
# Predict human genes with Human Phenotype Phenotypic Abnormality (HPPA) terms
#---------------------------------------------------------------------------
#---------------------------------------------------------------------------

## 1) prepare input and output files
### input file
architecture.file <- "http://dcgor.r-forge.r-project.org/data/Algo/SCOP_architecture.txt"
### output file
prediction.file <- "SCOP_architecture.HPPA_predicted.txt"
## 2) do predictions
res_Pred <- <a href='dcAlgoPredictMain.html' target=dcAlgoPredictMain>dcAlgoPredictMain</a>(input.file=architecture.file, output.file=prediction.file, feature.mode="supra", parallel=FALSE, RData.HIS.customised=RData.HIS.customised)
</div>
<strong class='message'>Start at 2014-12-23 14:57:00</strong>
<strong class='message'></strong>
<strong class='message'>Read the input file 'http://dcgor.r-forge.r-project.org/data/Algo/SCOP_architecture.txt' ...</strong>
<strong class='message'>Predictions for 17467 sequences (with 4351 distinct architectures) using 'sf2HPPA.HIS.RData' RData, 'sum' merge method, 'log' scale method and 'supra' feature mode (2014-12-23 14:57:01) ...</strong>
<strong class='message'>
##############################</strong>
<strong class='message'>'<a href='dcAlgoPredict.html' target=dcAlgoPredict>dcAlgoPredict</a>' is being called...</strong>
<strong class='message'>##############################
</strong>
<strong class='message'>Start at 2014-12-23 14:57:01</strong>
<strong class='message'></strong>
<strong class='message'>Load the customised HIS object 'sf2HPPA.HIS.RData' (2014-12-23 14:57:01)...</strong>
<strong class='message'>Predictions for 4351 architectures using 'sum' merge method, 'log' scale method and 'supra' feature mode (2014-12-23 14:57:01)...</strong>
<strong class='message'>	1 out of 4351 (2014-12-23 14:57:01)</strong>
<strong class='message'>	436 out of 4351 (2014-12-23 14:57:01)</strong>
<strong class='message'>	872 out of 4351 (2014-12-23 14:57:03)</strong>
<strong class='message'>	1308 out of 4351 (2014-12-23 14:57:04)</strong>
<strong class='message'>	1744 out of 4351 (2014-12-23 14:57:05)</strong>
<strong class='message'>	2180 out of 4351 (2014-12-23 14:57:06)</strong>
<strong class='message'>	2616 out of 4351 (2014-12-23 14:57:07)</strong>
<strong class='message'>	3052 out of 4351 (2014-12-23 14:57:08)</strong>
<strong class='message'>	3488 out of 4351 (2014-12-23 14:57:09)</strong>
<strong class='message'>	3924 out of 4351 (2014-12-23 14:57:10)</strong>
<strong class='message'>	4351 out of 4351 (2014-12-23 14:57:11)</strong>
<strong class='message'>
End at 2014-12-23 14:57:11</strong>
<strong class='message'>Runtime in total is: 10 secs
</strong>
<strong class='message'>##############################</strong>
<strong class='message'>'<a href='dcAlgoPredict.html' target=dcAlgoPredict>dcAlgoPredict</a>' has been completed!</strong>
<strong class='message'>##############################
</strong>
<strong class='message'>Preparations for output (2014-12-23 14:57:11)...</strong>
<strong class='message'>The predictions have been saved into '/Users/hfang/Sites/SUPERFAMILY/dcGOR/SCOP_architecture.HPPA_predicted.txt'.</strong>
<strong class='message'>
End at 2014-12-23 14:57:13</strong>
<strong class='message'>Runtime in total is: 13 secs
</strong>
<div class='input'>res_Pred[1:5,]
</div>
<div class='output'>     SeqID       Term         Score
[1,] "100288966" "HP:0000118" "1"  
[2,] "100288966" "HP:0000924" "1"  
[3,] "100288966" "HP:0001760" "1"  
[4,] "100288966" "HP:0001780" "1"  
[5,] "100288966" "HP:0002813" "1"  
</div>
<div class='input'>## 3) look at the prediction performance via Precision-Recall (PR) analysis
GSP.file <- "http://dcgor.r-forge.r-project.org/data/Algo/HP_anno.txt"
res_PR <- <a href='dcAlgoPredictPR.html' target=dcAlgoPredictPR>dcAlgoPredictPR</a>(GSP.file=GSP.file, prediction.file=prediction.file, ontology="HPPA")
</div>
<strong class='message'>Start at 2014-12-23 14:57:13</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'HPPA' (2014-12-23 14:57:13) ...</strong>
<strong class='message'>'onto.HPPA' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, import files for GSP and predictions (2014-12-23 14:57:13) ...</strong>
<strong class='message'>Third, propagate GSP annotations (2014-12-23 14:57:17) ...</strong>
<strong class='message'>	At level 16, there are 2 nodes, and 5 incoming neighbors.</strong>
<strong class='message'>	At level 15, there are 7 nodes, and 9 incoming neighbors.</strong>
<strong class='message'>	At level 14, there are 21 nodes, and 42 incoming neighbors.</strong>
<strong class='message'>	At level 13, there are 54 nodes, and 82 incoming neighbors.</strong>
<strong class='message'>	At level 12, there are 105 nodes, and 105 incoming neighbors.</strong>
<strong class='message'>	At level 11, there are 274 nodes, and 188 incoming neighbors.</strong>
<strong class='message'>	At level 10, there are 463 nodes, and 294 incoming neighbors.</strong>
<strong class='message'>	At level 9, there are 782 nodes, and 441 incoming neighbors.</strong>
<strong class='message'>	At level 8, there are 1004 nodes, and 538 incoming neighbors.</strong>
<strong class='message'>	At level 7, there are 1182 nodes, and 581 incoming neighbors.</strong>
<strong class='message'>	At level 6, there are 1295 nodes, and 527 incoming neighbors.</strong>
<strong class='message'>	At level 5, there are 940 nodes, and 290 incoming neighbors.</strong>
<strong class='message'>	At level 4, there are 408 nodes, and 99 incoming neighbors.</strong>
<strong class='message'>	At level 3, there are 114 nodes, and 21 incoming neighbors.</strong>
<strong class='message'>	At level 2, there are 21 nodes, and 1 incoming neighbors.</strong>
<strong class='message'>	At level 1, there are 1 nodes, and 0 incoming neighbors.</strong>
<strong class='message'>	There are 3048 genes/proteins in GSP (2014-12-23 14:57:36).</strong>
<strong class='message'>Fourth, process input predictions (2014-12-23 14:57:36) ...</strong>
<strong class='message'>	There are 7426 genes/proteins in predictions (2014-12-23 14:57:37).</strong>
<strong class='message'>Fifth, calculate the precision and recall for each of 1694 predicted and GSP genes/proteins (2014-12-23 14:57:37).</strong>
<strong class='message'>Finally, calculate the averaged precision and recall (2014-12-23 14:57:39).</strong>
<strong class='message'>In summary, Prediction coverage: 0.56 (amongst 3048 targets in GSP), and F-measure: 0.26.</strong>
<strong class='message'>
End at 2014-12-23 14:57:39</strong>
<strong class='message'>Runtime in total is: 26 secs
</strong>
<div class='input'>res_PR
</div>
<div class='output'>        Precision     Recall
1       0.7029713 0.03603475
0.90001 0.5669893 0.05676330
0.80002 0.4779550 0.08826593
0.70003 0.4242904 0.12708065
0.60004 0.3901488 0.15893078
0.50005 0.3667623 0.18361571
0.40006 0.3526200 0.19984090
0.30007 0.3452423 0.20644254
0.20008 0.3425576 0.21048189
0.10009 0.3403761 0.21287068
1e-04   0.3313322 0.21942770
</div>
<div class='input'>### plot PR curve
plot(res_PR[,2], res_PR[,1], xlim=c(0,1), ylim=c(0,1), type="b", xlab="Recall", ylab="Precision")
</div>
<a href="javascript:newWin('AlgoPred-167.png', 'AlgoPred-167.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='AlgoPred-167.png', width="400" /></a>
<div class='input'>
#---------------------------------------------------------------------------
#---------------------------------------------------------------------------
# Compared to RWR-based predictions and naive predictions
#---------------------------------------------------------------------------
#---------------------------------------------------------------------------

## 1) RWR-based predictions, please see: <a href='dcRWRpredict.html' target=dcRWRpredict>dcRWRpredict</a>
## 2) Naive predictions, please see: <a href='dcNaivePredict.html' target=dcNaivePredict>dcNaivePredict</a></div>
      </pre>
    </div>

  <div class="span4">

    <h2>Source demo</h2>
    <a href="javascript:newWin('demo/AlgoPred.r', 'demo/AlgoPred.r', '1200', '1200')" title="Click to view"><B><code>AlgoPred.r</code></B></a>
    
    <h2>Functions used in this demo</h2>
    <ul>
      <li><a href="dcAlgo.html">dcAlgo</a></li>
      <li><a href="dcAlgoPropagate.html">dcAlgoPropagate</a></li>
      <li><a href="dcAlgoPredictMain.html">dcAlgoPredictMain</a></li>
      <li><a href="dcAlgoPredict.html">dcAlgoPredict</a></li>
      <li><a href="dcAlgoPredictPR.html">dcAlgoPredictPR</a></li>
      <li><a href="dcRWRpredict.html">dcRWRpredict</a></li>
      <li><a href="dcNaivePredict.html">dcNaivePredict</a></li>
    </ul>
    
    <!--
    
    <h2>Citation</h2>
    <p>Fang H. (2014) <B>dcGOR: an R package for analysing ontologies and protein domain annotations</B>. <I>PLoS Computational Biology</I> 10(10):e1003929. <a href="http://dx.doi.org/10.1371/journal.pcbi.1003929">doi:10.1371/journal.pcbi.1003929</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25356683">25356683</a></p>

<p>You may also cite the package as:</p>

<ul>
<li><code>database</code></li>
</ul>

<p>Fang H, Gough J. (2013) <B>dcGO: database of domain-centric ontologies on functions, phenotypes, diseases and more</B>. <I>Nucleic Acids Research</I> 41(Database issue):D536-44. <a href="http://dx.doi.org/10.1093/nar/gks1080">doi:10.1093/nar/gks1080</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23161684">23161684</a></p>

<ul>
<li><code>web link/URL</code></li>
</ul>

<p>Package &ldquo;dcGOR&rdquo; [<a href="http://cran.r-project.org/package=dcGOR">http://cran.r-project.org/package=dcGOR</a>]</p>

    

    
    <h2>URL</h2>
    <ul>
      <li><a href="http://supfam.org/dcGOR">http://supfam.org/dcGOR</a></li>
      <li><a href=" http://dcgor.r-forge.r-project.org"> http://dcgor.r-forge.r-project.org</a></li>
      <li><a href=" http://cran.r-project.org/package=dcGOR"> http://cran.r-project.org/package=dcGOR</a></li>
      <li><a href=" https://github.com/hfang-bristol/dcGOR"> https://github.com/hfang-bristol/dcGOR</a></li>
    </ul>
    
    <h2>Author</h2>
    Hai Fang and Julian Gough
    
    -->
    
  </div>

</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>