<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>Demo for analysing ancestral domainome in Eukaryotes @ dcGOR 1.0.5</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="Hai Fang and Julian Gough">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dcGOR 1.0.5</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
       
        <!-- HTML for GOOLE SEARCH BAR 
        <ul class="nav">
            <div id="search-wrapper">
                <form id="search" method="get" action="http://www.google.com/custom">
                    <div class="search-box">
                        <i class="icon-search icon-white"></i>
                        <input type="text" id="tfq" class="textinput" name="q" size="21" maxlength="120" value="Google Search"/>
                    </div>
                </form>
            </div>
        </ul>
        -->
        
      </div>
    </div>
  </div>
</div>

<!-- CSS styles for standard search box with placeholder text-->
<style type="text/css">
    #search-wrapper {
        width:300px;
        height:40px;
        background-color:transparent;
        margin:0px auto 0;
        border:0px solid #ff0000;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

    #search {
        background-color:transparent;
        padding:0px;
        height:40px;
        overflow:hidden;
    }

    .search-box {
        width:250px;
        height:32px;
        background-color:transparent;
        margin:2px 7px;
        border:0px solid #cfcfcf;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

	.textinput{
		margin: 0;
		padding: 5px 15px;
		font-family: Arial, Helvetica, sans-serif;
		font-size:14px;
		color:#999999;
		border:1px solid #f00; border-right:0px;
		border-top-left-radius: 5px 5px;
		border-bottom-left-radius: 5px 5px;
	}
</style>

<!-- JAVASCRIPT to clear search text when the field is clicked -->
<script type="text/javascript">
window.onload = function(){
	//Get submit button
	var submitbutton = document.getElementById("tfq");
	//Add listener to submit button
	if(submitbutton.addEventListener){
		submitbutton.addEventListener("click", function() {
			if (submitbutton.value == 'Google Search'){//Customize this text string to whatever you want
				submitbutton.value = 'dcGOR R ';
			}
		});
	}
}
</script>

<script>
  (function() {
    var cx = '012094530736424854714:k8ieyz4ag4a';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') +
        '//www.google.com/cse/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>

<div><gcse:search></gcse:search></div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Demo for analysing ancestral domainome in Eukaryotes</h2>

Notes:
<li> All results are based on dcGOR (version 1.0.5). </li>
<li> R scripts (i.e. R expressions) plus necessary comments are highlighted in light cyan, and the rest are outputs in the screen. </li>
<li> Images displayed below may be distorted, but should be normal in your screen. </li>
<li> Functions contained in dcGOR 1.0.5 are hyperlinked in-place and also listed on the right side. </li>

<div class="row">
    <div class="span8">
      <pre>
      <div class='input'># This is a demo for analysing ancestral superfamily domainome in Eukaryotes by Fang et al
# 
# This ancestral domain-ome dataset (available from <a href="http://www.ncbi.nlm.nih.gov/pubmed/23778980" target="23778980">http://www.ncbi.nlm.nih.gov/pubmed/23778980</a>) is stored as an 'Anno' object (S4 class). It contains information about domain repertoires (a complete set of domains: domain-ome) in Eukaryotes (including extant and ancestral genomes):
## annoData(Ancestral_domainome): a sparse matrix of 2019 domains X 875 terms/genomes (including 438 extant genomes and 437 ancestral genomes), with each entry telling how many different architectures a domain has in a genome. Note: zero entry also means that this domain is absent in the genome
## termData(Ancestral_domainome): variables describing terms/genomes (i.e. columns in annoData), including extant/ancestral genome information: "left_id" (unique and used as internal id), "right_id" (used in combination with "left_id" to define the post-ordered binary tree structure), "taxon_id" (NCBI taxonomy id, if matched), "genome" (2-letter genome identifiers used in SUPERFAMILY, if being extant), "name" (NCBI taxonomy name, if matched), "rank" (NCBI taxonomy rank, if matched), "branchlength" (branch length in relevance to the parent), and "common_name" (NCBI taxonomy common name, if matched and existed)
## domainData: variables describing domains (i.e. rows in annoData), including information about domains: "sunid" for SCOP id, "level" for SCOP level, "classification" for SCOP classification, "description" for SCOP description
###############################################################################
library(dcGOR)

#----------------------------------------------------------
#----------------------------------------------------------
# Derive domains unique/gained in human compared to Metazoa
#----------------------------------------------------------
#----------------------------------------------------------

# load data as an 'Anno' object
Ancestral_domainome <- <a href='dcRDataLoader.html' target=dcRDataLoader>dcRDataLoader</a>("Ancestral_domainome")
</div>
<strong class='message'>'Ancestral_domainome' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<div class='input'>Ancestral_domainome
</div>
<div class='output'>An object of S4 class 'Anno'
@annoData: 2019 domains, 875 terms 
@termData (InfoDataFrame)
  termNames: 3 4 5 ... 1743 1750 (875 total)
  tvarLabels: left_id right_id taxon_id ... branchlength common_name (8
    total)
@domainData (InfoDataFrame)
  domainNames: 46458 46548 46557 ... 161266 161270 (2019 total)
  dvarLabels: sunid level classification description
</div>
<div class='input'>
# extract a list of domains that are present at Metazoa
flag_genome <- which(tData(Ancestral_domainome)$name=="Metazoa")
flag_domain <- annoData(Ancestral_domainome)[,flag_genome]!=0
domains_metazoa <- domainData(Ancestral_domainome)[flag_domain,]
domains_metazoa
</div>
<div class='output'>An object of S4 class 'InfoDataFrame'
  rowNames: 46458 46548 46561 ... 161245 161270 (1292 total)
  colNames: sunid level classification description
</div>
<div class='input'>
# extract a list of domains that are present at human
flag_genome <- which(tData(Ancestral_domainome)$name=="Homo sapiens")
flag_domain <- annoData(Ancestral_domainome)[,flag_genome]!=0
domains_human <- domainData(Ancestral_domainome)[flag_domain,]
domains_human
</div>
<div class='output'>An object of S4 class 'InfoDataFrame'
  rowNames: 46458 46548 46561 ... 161245 161256 (1112 total)
  colNames: sunid level classification description
</div>
<div class='input'>
# calculate domains unique in human
domains_human_unique <- setdiff(rowNames(domains_human), rowNames(domains_metazoa))
length(domains_human_unique)
</div>
<div class='output'>[1] 58
</div>
<div class='input'>
# write into a local file <a href="Domains_unique_in_human.txt">Domains_unique_in_human.txt</a>
df <- dData(Ancestral_domainome)
ind <- match(domains_human_unique, rownames(df))
out <- df[ind,]
write.table(out, file="Domains_unique_in_human.txt", col.names=T, row.names=F, sep="\t")

#---------------------------------------------------------------------------
#---------------------------------------------------------------------------
# Enrichment analysis for domains unique/gained in human compared to Metazoa
#---------------------------------------------------------------------------
#---------------------------------------------------------------------------

## define the input data (domains gained in human) and the background (all domains in Metazoa)
data <- domains_human_unique
background <- rowNames(domains_metazoa)

## 1a) GOBP enrichment analysis, producing an object of S4 class 'Eoutput'
### By default, using all annotatable domains as the background
eoutput <- <a href='dcEnrichment.html' target=dcEnrichment>dcEnrichment</a>(data, domain="SCOP.sf", ontology="GOBP")
</div>
<strong class='message'>Start at 2014-12-05 17:02:10</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOBP', the domain 'SCOP.sf', and their associations (2014-12-05 17:02:10) ...</strong>
<strong class='message'>'onto.GOBP' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf2GOBP' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2014-12-05 17:03:23) ...</strong>
<strong class='message'>	There are 5471 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-12-05 17:03:27) ...</strong>
<strong class='message'>
End at 2014-12-05 17:03:29</strong>
<strong class='message'>Runtime in total is: 79 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'SCOP.sf'
  @ontology: 'GOBP'
  @term_info: a data.frame of 2207 terms X 5 information
  @anno: a list of 2207 terms, each storing annotated domains
  @data: a vector containing a group of 43 input domains (annotatable)
  @background: a vector containing a group of 1163 background domains (annotatable)
  @overlap: a list of 2207 terms, each containing domains overlapped with input domains
  @zscore: a vector of 2207 terms, containing z-scores
  @pvalue: a vector of 2207 terms, containing p-values
  @adjp: a vector of 2207 terms, containing adjusted p-values
In summary, a total of 2207 terms ('GOBP') are analysed for a group of 43 input domains ('SCOP.sf')
</div>
<div class='input'>### write into a local file <a href="GOBP_enrichments.txt">GOBP_enrichments.txt</a>
write(eoutput, file='GOBP_enrichments.txt')
</div>
<strong class='message'>A file ('GOBP_enrichments.txt') has been written into your local directory ('/Users/hfang/Sites/SUPERFAMILY/dcGOR')</strong>
<div class='input'>### view the top 5 significant terms
view(eoutput, top_num=5, sortBy="pvalue", details=FALSE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore  pvalue    adjp
GO:0007275 GO:0007275  1066     43       43   2.01 0.0e+00 0.0e+00
GO:0048513 GO:0048513   972     43       43   2.96 0.0e+00 0.0e+00
GO:0048731 GO:0048731  1039     43       43   2.31 0.0e+00 0.0e+00
GO:0034138 GO:0034138   558     43       36   4.78 1.4e-07 7.7e-05
GO:0002224 GO:0002224   564     43       36   4.71 2.0e-07 7.9e-05
                                        term_name
GO:0007275   multicellular organismal development
GO:0048513                      organ development
GO:0048731                     system development
GO:0034138 toll-like receptor 3 signaling pathway
GO:0002224   toll-like receptor signaling pathway
</div>
<div class='input'>### visualise the top 5 significant terms in GOMF DAG
#### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
<a href='visEnrichment.html' target=visEnrichment>visEnrichment</a>(eoutput)
</div>
<strong class='message'>Ontology 'GOBP' containing 14 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('Fang-31.png', 'Fang-31.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='Fang-31.png', width="400" /></a>
<div class='input'>
## 1b) GOBP enrichment analysis, producing an object of S4 class 'Eoutput'
### Alternatively, using all domains in Metazoa as the background (customised)
eoutput <- <a href='dcEnrichment.html' target=dcEnrichment>dcEnrichment</a>(data, background, domain="SCOP.sf", ontology="GOBP")
</div>
<strong class='message'>Start at 2014-12-05 17:03:35</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOBP', the domain 'SCOP.sf', and their associations (2014-12-05 17:03:35) ...</strong>
<strong class='message'>'onto.GOBP' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf2GOBP' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2014-12-05 17:04:48) ...</strong>
<strong class='message'>	There are 5453 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-12-05 17:04:51) ...</strong>
<strong class='message'>
End at 2014-12-05 17:04:53</strong>
<strong class='message'>Runtime in total is: 78 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'SCOP.sf'
  @ontology: 'GOBP'
  @term_info: a data.frame of 2206 terms X 5 information
  @anno: a list of 2206 terms, each storing annotated domains
  @data: a vector containing a group of 43 input domains (annotatable)
  @background: a vector containing a group of 1083 background domains (annotatable)
  @overlap: a list of 2206 terms, each containing domains overlapped with input domains
  @zscore: a vector of 2206 terms, containing z-scores
  @pvalue: a vector of 2206 terms, containing p-values
  @adjp: a vector of 2206 terms, containing adjusted p-values
In summary, a total of 2206 terms ('GOBP') are analysed for a group of 43 input domains ('SCOP.sf')
</div>
<div class='input'>### write into a local file <a href="GOBP_enrichments.txt">GOBP_enrichments.txt</a>
write(eoutput, file='GOBP_enrichments_customised.txt')
</div>
<strong class='message'>A file ('GOBP_enrichments_customised.txt') has been written into your local directory ('/Users/hfang/Sites/SUPERFAMILY/dcGOR')</strong>
<div class='input'>### view the top 5 significant terms
view(eoutput, top_num=5, sortBy="pvalue", details=FALSE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore  pvalue    adjp
GO:0007275 GO:0007275  1029     43       43   1.53 0.0e+00 0.00000
GO:0048513 GO:0048513   942     43       43   2.59 0.0e+00 0.00000
GO:0048731 GO:0048731  1003     43       43   1.89 0.0e+00 0.00000
GO:0034138 GO:0034138   541     43       36   4.52 5.1e-07 0.00027
GO:0002224 GO:0002224   546     43       36   4.46 6.8e-07 0.00027
                                        term_name
GO:0007275   multicellular organismal development
GO:0048513                      organ development
GO:0048731                     system development
GO:0034138 toll-like receptor 3 signaling pathway
GO:0002224   toll-like receptor signaling pathway
</div>
<div class='input'>### visualise the top 5 significant terms in GOMF DAG
#### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
<a href='visEnrichment.html' target=visEnrichment>visEnrichment</a>(eoutput)
</div>
<strong class='message'>Ontology 'GOBP' containing 14 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('Fang-52.png', 'Fang-52.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='Fang-52.png', width="400" /></a>
<div class='input'>
<a href='visEnrichment.html' target=visEnrichment>visEnrichment</a>(eoutput,height=10, width=10,graph.node.attrs=list(fontsize=20),node.attrs=list(fontsize=28), node.info = c("both", "none","term_id", "term_name", "full_term_name")[3])
</div>
<strong class='message'>Ontology 'GOBP' containing 14 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('Fang-55.png', 'Fang-55.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='Fang-55.png', width="400" /></a>
<div class='input'>
## 2) GOMF enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- <a href='dcEnrichment.html' target=dcEnrichment>dcEnrichment</a>(data, background, domain="SCOP.sf", ontology="GOMF")
</div>
<strong class='message'>Start at 2014-12-05 17:05:02</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'GOMF', the domain 'SCOP.sf', and their associations (2014-12-05 17:05:02) ...</strong>
<strong class='message'>'onto.GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf2GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2014-12-05 17:05:07) ...</strong>
<strong class='message'>	There are 810 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-12-05 17:05:07) ...</strong>
<strong class='message'>
End at 2014-12-05 17:05:07</strong>
<strong class='message'>Runtime in total is: 5 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'SCOP.sf'
  @ontology: 'GOMF'
  @term_info: a data.frame of 175 terms X 5 information
  @anno: a list of 175 terms, each storing annotated domains
  @data: a vector containing a group of 34 input domains (annotatable)
  @background: a vector containing a group of 1007 background domains (annotatable)
  @overlap: a list of 175 terms, each containing domains overlapped with input domains
  @zscore: a vector of 175 terms, containing z-scores
  @pvalue: a vector of 175 terms, containing p-values
  @adjp: a vector of 175 terms, containing adjusted p-values
In summary, a total of 175 terms ('GOMF') are analysed for a group of 34 input domains ('SCOP.sf')
</div>
<div class='input'>### write into a local file <a href="GOMF_enrichments.txt">GOMF_enrichments.txt</a>
write(eoutput, file='GOMF_enrichments.txt')
</div>
<strong class='message'>A file ('GOMF_enrichments.txt') has been written into your local directory ('/Users/hfang/Sites/SUPERFAMILY/dcGOR')</strong>
<div class='input'>### view the top 5 significant terms
view(eoutput, top_num=5, sortBy="pvalue", details=FALSE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore  pvalue   adjp
GO:0008935 GO:0008935    14     34        4   5.25 5.2e-05 0.0073
GO:0004824 GO:0004824    48     34        7   4.40 1.0e-04 0.0073
GO:0070569 GO:0070569   633     34       30   3.11 1.7e-04 0.0073
GO:0003977 GO:0003977   633     34       30   3.11 1.7e-04 0.0073
GO:0016880 GO:0016880    55     34        7   3.95 2.8e-04 0.0090
                                                  term_name
GO:0008935  1,4-dihydroxy-2-naphthoyl-CoA synthase activity
GO:0004824                      lysine-tRNA ligase activity
GO:0070569                     uridylyltransferase activity
GO:0003977 UDP-N-acetylglucosamine diphosphorylase activity
GO:0016880          acid-ammonia (or amide) ligase activity
</div>
<div class='input'>### visualise the top 5 significant terms in GOMF DAG
#### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
<a href='visEnrichment.html' target=visEnrichment>visEnrichment</a>(eoutput)
</div>
<strong class='message'>Ontology 'GOMF' containing 18 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('Fang-76.png', 'Fang-76.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='Fang-76.png', width="400" /></a>
<div class='input'>
## 3) HPPA enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- <a href='dcEnrichment.html' target=dcEnrichment>dcEnrichment</a>(data, domain="SCOP.sf", ontology="HPPA")
</div>
<strong class='message'>Start at 2014-12-05 17:05:10</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'HPPA', the domain 'SCOP.sf', and their associations (2014-12-05 17:05:10) ...</strong>
<strong class='message'>'onto.HPPA' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf2HPPA' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2014-12-05 17:05:12) ...</strong>
<strong class='message'>	There are 259 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-12-05 17:05:12) ...</strong>
<strong class='message'>
End at 2014-12-05 17:05:12</strong>
<strong class='message'>Runtime in total is: 2 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'SCOP.sf'
  @ontology: 'HPPA'
  @term_info: a data.frame of 37 terms X 5 information
  @anno: a list of 37 terms, each storing annotated domains
  @data: a vector containing a group of 4 input domains (annotatable)
  @background: a vector containing a group of 166 background domains (annotatable)
  @overlap: a list of 37 terms, each containing domains overlapped with input domains
  @zscore: a vector of 37 terms, containing z-scores
  @pvalue: a vector of 37 terms, containing p-values
  @adjp: a vector of 37 terms, containing adjusted p-values
In summary, a total of 37 terms ('HPPA') are analysed for a group of 4 input domains ('SCOP.sf')
</div>
<div class='input'>### write into a local file <a href="HPPA_enrichments.txt">HPPA_enrichments.txt</a>
write(eoutput, file='HPPA_enrichments.txt')
</div>
<strong class='message'>A file ('HPPA_enrichments.txt') has been written into your local directory ('/Users/hfang/Sites/SUPERFAMILY/dcGOR')</strong>
<div class='input'>### view the top 5 significant terms
view(eoutput, top_num=5, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore pvalue adjp
HP:0000152 HP:0000152   115      4        4  1.340      0    0
HP:0000707 HP:0000707   132      4        4  1.020      0    0
HP:0000924 HP:0000924   146      4        4  0.747      0    0
HP:0001574 HP:0001574    88      4        4  1.900      0    0
HP:0002664 HP:0002664   106      4        4  1.520      0    0
                                    term_name         term_namespace
HP:0000152       Abnormality of head and neck Phenotypic_abnormality
HP:0000707  Abnormality of the nervous system Phenotypic_abnormality
HP:0000924 Abnormality of the skeletal system Phenotypic_abnormality
HP:0001574      Abnormality of the integument Phenotypic_abnormality
HP:0002664                           Neoplasm Phenotypic_abnormality
           term_distance                 members
HP:0000152             1 47686,75399,57581,57630
HP:0000707             1 47686,75399,57581,57630
HP:0000924             1 47686,75399,57581,57630
HP:0001574             1 47686,75399,57581,57630
HP:0002664             1 47686,75399,57581,57630
</div>
<div class='input'>### visualise the top 5 significant terms in GOMF DAG
#### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
<a href='visEnrichment.html' target=visEnrichment>visEnrichment</a>(eoutput, path.mode="all_paths", node.info="full_term_name")
</div>
<strong class='message'>Ontology 'HPPA' containing 6 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('Fang-97.png', 'Fang-97.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='Fang-97.png', width="400" /></a>
<div class='input'>
## 4) DO enrichment analysis, producing an object of S4 class 'Eoutput'
eoutput <- <a href='dcEnrichment.html' target=dcEnrichment>dcEnrichment</a>(data, domain="SCOP.sf", ontology="DO")
</div>
<strong class='message'>Start at 2014-12-05 17:05:14</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology 'DO', the domain 'SCOP.sf', and their associations (2014-12-05 17:05:14) ...</strong>
<strong class='message'>'onto.DO' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>'SCOP.sf2DO' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<strong class='message'>Second, perform enrichment analysis using HypergeoTest (2014-12-05 17:05:15) ...</strong>
<strong class='message'>	There are 219 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-12-05 17:05:15) ...</strong>
<strong class='message'>
End at 2014-12-05 17:05:15</strong>
<strong class='message'>Runtime in total is: 1 secs
</strong>
<div class='input'>eoutput
</div>
<div class='output'>An object of S4 class 'Eoutput', containing following slots:
  @domain: 'SCOP.sf'
  @ontology: 'DO'
  @term_info: a data.frame of 109 terms X 5 information
  @anno: a list of 109 terms, each storing annotated domains
  @data: a vector containing a group of 11 input domains (annotatable)
  @background: a vector containing a group of 219 background domains (annotatable)
  @overlap: a list of 109 terms, each containing domains overlapped with input domains
  @zscore: a vector of 109 terms, containing z-scores
  @pvalue: a vector of 109 terms, containing p-values
  @adjp: a vector of 109 terms, containing adjusted p-values
In summary, a total of 109 terms ('DO') are analysed for a group of 11 input domains ('SCOP.sf')
</div>
<div class='input'>### write into a local file <a href="DO_enrichments.txt">DO_enrichments.txt</a>
write(eoutput, file='DO_enrichments.txt')
</div>
<strong class='message'>A file ('DO_enrichments.txt') has been written into your local directory ('/Users/hfang/Sites/SUPERFAMILY/dcGOR')</strong>
<div class='input'>### view the top 5 significant terms
view(eoutput, top_num=5, sortBy="pvalue", details=TRUE)
</div>
<div class='output'>              term_id nAnno nGroup nOverlap zscore  pvalue   adjp
DOID:14566 DOID:14566   196     11       11   1.16 0.0e+00 0.0000
DOID:2583   DOID:2583    11     11        4   4.87 4.6e-05 0.0017
DOID:620     DOID:620    12     11        4   4.61 7.7e-05 0.0017
DOID:2115   DOID:2115    12     11        4   4.61 7.7e-05 0.0017
DOID:7004   DOID:7004    12     11        4   4.61 7.7e-05 0.0017
                                   term_name   term_namespace term_distance
DOID:14566 disease of cellular proliferation Disease_Ontology             1
DOID:2583                 agammaglobulinemia Disease_Ontology             5
DOID:620               blood protein disease Disease_Ontology             4
DOID:2115                  B cell deficiency Disease_Ontology             4
DOID:7004   ACTH-secreting pituitary adenoma Disease_Ontology             7
                                                                      members
DOID:14566 57944,47353,57630,110035,69125,47686,48552,75399,57392,54117,47266
DOID:2583                                             47686,57392,54117,47266
DOID:620                                              47686,57392,54117,47266
DOID:2115                                             47686,57392,54117,47266
DOID:7004                                             47686,57392,54117,47266
</div>
<div class='input'>### visualise the top 5 significant terms in GOMF DAG
#### color-coded according to 10-based negative logarithm of adjusted p-values (adjp)
<a href='visEnrichment.html' target=visEnrichment>visEnrichment</a>(eoutput, path.mode="all_paths")
</div>
<strong class='message'>Ontology 'DO' containing 15 nodes/terms (including 5 in query; also highlighted in frame) has been shown in your screen, with colorbar indicating -1*log10(adjusted p-values)</strong>
<a href="javascript:newWin('Fang-118.png', 'Fang-118.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='Fang-118.png', width="400" /></a>
<div class='input'>
#---------------------------------------------------------------------
#---------------------------------------------------------------------
# Pair-wise semantic similarity between domains unique/gained in human
#---------------------------------------------------------------------
#---------------------------------------------------------------------

## 1) load onto.DO (as 'Onto' object)
g <- <a href='dcRDataLoader.html' target=dcRDataLoader>dcRDataLoader</a>('onto.DO')
</div>
<strong class='message'>'onto.DO' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<div class='input'>g
</div>
<div class='output'>An object of S4 class 'Onto'
@adjMatrix: a direct matrix of 6352 terms (parents/from) X 6352 terms (children/to) 
@nodeInfo (InfoDataFrame)
  nodeNames: DOID:4 DOID:630 DOID:7 ... DOID:5746 DOID:7438 (6352
    total)
  nodeAttr: term_id term_name term_namespace term_distance
</div>
<div class='input'>## 2) load SCOP superfamilies annotated by DO (as 'Anno' object)
Anno <- <a href='dcRDataLoader.html' target=dcRDataLoader>dcRDataLoader</a>('SCOP.sf2DO')
</div>
<strong class='message'>'SCOP.sf2DO' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<div class='input'>Anno
</div>
<div class='output'>An object of S4 class 'Anno'
@annoData: 219 domains, 548 terms 
@termData (InfoDataFrame)
  termNames: DOID:0014667 DOID:0050117 DOID:0080015 ... DOID:8541
    DOID:9849 (548 total)
  tvarLabels: ID Name Namespace Distance
@domainData (InfoDataFrame)
  domainNames: 46458 46579 46689 ... 161070 161084 (219 total)
  dvarLabels: id level description
</div>
<div class='input'>
## 3) prepare for ontology appended with annotation information
dag <- <a href='dcDAGannotate.html' target=dcDAGannotate>dcDAGannotate</a>(g, annotations=Anno, path.mode="shortest_paths",verbose=FALSE)
dag
</div>
<div class='output'>An object of S4 class 'Onto'
@adjMatrix: a direct matrix of 549 terms (parents/from) X 549 terms (children/to) 
@nodeInfo (InfoDataFrame)
  nodeNames: DOID:4 DOID:630 DOID:7 ... DOID:9849 DOID:1070 (549 total)
  nodeAttr: term_id term_name term_namespace term_distance annotations
    IC
</div>
<div class='input'>
## 4) calculate pair-wise semantic similarity between domains unique/gained in human
domains <- domains_human_unique
dnetwork <- <a href='dcDAGdomainSim.html' target=dcDAGdomainSim>dcDAGdomainSim</a>(g=dag, domains=domains, method.domain="BM.average", method.term="Resnik", parallel=FALSE, verbose=TRUE)
</div>
<strong class='message'>Start at 2014-12-05 17:05:18</strong>
<strong class='message'></strong>
<strong class='message'>First, extract all annotatable domains (2014-12-05 17:05:18)...</strong>
<strong class='message'>	there are 11 input domains amongst 219 annotatable domains</strong>
<strong class='message'>Second, pre-compute semantic similarity between 129 terms (forced to be the most specific for each domain) using Resnik method (2014-12-05 17:05:19)...</strong>
<strong class='message'>Last, calculate pair-wise semantic similarity between 11 domains using BM.average method (2014-12-05 17:05:19)...</strong>
<strong class='message'>	1 out of 11 (2014-12-05 17:05:19)</strong>
<strong class='message'>	2 out of 11 (2014-12-05 17:05:19)</strong>
<strong class='message'>	4 out of 11 (2014-12-05 17:05:19)</strong>
<strong class='message'>	6 out of 11 (2014-12-05 17:05:19)</strong>
<strong class='message'>	8 out of 11 (2014-12-05 17:05:20)</strong>
<strong class='message'>	10 out of 11 (2014-12-05 17:05:20)</strong>
<strong class='message'></strong>
<strong class='message'>Finish at 2014-12-05 17:05:20</strong>
<strong class='message'>Runtime in total is: 2 secs
</strong>
<div class='input'>dnetwork
</div>
<div class='output'>An object of S4 class 'Dnetwork'
@adjMatrix: a weighted symmetric matrix of 11 domains X 11 domains 
@nodeInfo (InfoDataFrame)
  nodeNames: 47266 47353 47686 ... 75399 110035 (11 total)
  nodeAttr: id
</div>
<div class='input'>
## 5) heatmap the adjacency matrix of the domain network
D <- as.matrix(adjMatrix(dnetwork))
supraHex::visHeatmapAdv(D, Rowv=F, Colv=F, dendrogram="none", colormap="white-lightpink-darkred", zlim=c(0,1.2), KeyValueName="DO semantic similarity")
</div>
<a href="javascript:newWin('Fang-148.png', 'Fang-148.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='Fang-148.png', width="400" /></a>
<div class='input'>
## 6) visualise the domain network as a graph
### convert it to an object of class 'igraph' (for subsequent visualisation)
ig <- <a href='dcConverter.html' target=dcConverter>dcConverter</a>(dnetwork, from='Dnetwork', to='igraph')
</div>
<strong class='message'>Your input object 'dnetwork' of class 'Dnetwork' has been converted into an object of class 'igraph'.</strong>
<div class='input'>ig
</div>
<div class='output'>IGRAPH UNW- 11 55 -- 
+ attr: name (v/c), id (v/c), weight (e/n)
</div>
<div class='input'>### extract edge weight (with 2-digit precision)
x <- signif(E(ig)$weight, digits=2)
### rescale into an interval [1,4] as edge width
edge.width <- 1 + (x-min(x))/(max(x)-min(x))*3
### prepare the node labels (including domain id and description)
ind <- match(V(ig)$name,domainNames(Anno))
vertex.label <- paste(V(ig)$name, '\n', as.character(dData(Anno)[ind,]$description), sep='')
### do visualisation
dnet::visNet(g=ig, vertex.label=vertex.label, vertex.label.color="black", vertex.label.cex=0.7, vertex.shape="sphere", edge.width=edge.width, edge.label=x, edge.label.cex=0.7)
</div>
<a href="javascript:newWin('Fang-154.png', 'Fang-154.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='Fang-154.png', width="400" /></a>
<div class='input'>
#------------------------------------------------------------
#------------------------------------------------------------
# Random Walk with Restart (RWR)-based contacts between terms
#------------------------------------------------------------
#------------------------------------------------------------

## 1) define sets of seeds: each seed with equal weight (i.e. all non-zero entries are '1')
### seeds are terms from GOMF
Anno <- <a href='dcRDataLoader.html' target=dcRDataLoader>dcRDataLoader</a>('SCOP.sf2GOMF')
</div>
<strong class='message'>'SCOP.sf2GOMF' (from package 'dcGOR' version 1.0.5) has been loaded into the working environment</strong>
<div class='input'>flag <- match(domainNames(Anno), nodeNames(dnetwork))
ind <- which(!is.na(flag))
data <- as.matrix(annoData(Anno)[ind,])
### GOMF terms having at least 3 annotatable domains that are also in human-unique domains above
ind <- apply(data,2,sum)>=3
data <- data[,ind]

## 2) assess their contact significance and strength based on RWR
coutput <- <a href='dcRWRpipeline.html' target=dcRWRpipeline>dcRWRpipeline</a>(data=data, g=dnetwork, permutation="degree", num.permutation=2000, adjp.cutoff=0.1, parallel=FALSE)
</div>
<strong class='message'>Start at 2014-12-05 17:05:22</strong>
<strong class='message'></strong>
<strong class='message'>First, RWR on input graph (11 nodes and 55 edges) using input matrix (10 rows and 12 columns) as seeds (2014-12-05 17:05:22)...</strong>
<strong class='message'>	using indirect method to do RWR (2014-12-05 17:05:22)...</strong>
<strong class='message'>Second, calculate contact strength (2014-12-05 17:05:22)...</strong>
<strong class='message'>Third, generate the distribution of contact strength based on 2000 permutations on nodes respecting degree (2014-12-05 17:05:22)...</strong>
<strong class='message'>	1 out of 2000 (2014-12-05 17:05:22)</strong>
<strong class='message'>	200 out of 2000 (2014-12-05 17:05:24)</strong>
<strong class='message'>	400 out of 2000 (2014-12-05 17:05:25)</strong>
<strong class='message'>	600 out of 2000 (2014-12-05 17:05:27)</strong>
<strong class='message'>	800 out of 2000 (2014-12-05 17:05:28)</strong>
<strong class='message'>	1000 out of 2000 (2014-12-05 17:05:30)</strong>
<strong class='message'>	1200 out of 2000 (2014-12-05 17:05:32)</strong>
<strong class='message'>	1400 out of 2000 (2014-12-05 17:05:33)</strong>
<strong class='message'>	1600 out of 2000 (2014-12-05 17:05:35)</strong>
<strong class='message'>	1800 out of 2000 (2014-12-05 17:05:36)</strong>
<strong class='message'>	2000 out of 2000 (2014-12-05 17:05:38)</strong>
<strong class='message'>Last, estimate the significance of contact strength: zscore, pvalue, and BH adjusted-pvalue (2014-12-05 17:05:38)...</strong>
<strong class='message'>Also, construct the contact graph under the cutoff 1.0e-01 of adjusted-pvalue (2014-12-05 17:05:38)...</strong>
<strong class='message'>Your input object 'icontact' of class 'igraph' has been converted into an object of class 'Cnetwork'.</strong>
<strong class='message'></strong>
<strong class='message'>Finish at 2014-12-05 17:05:38</strong>
<strong class='message'>Runtime in total is: 16 secs
</strong>
<div class='input'>coutput
</div>
<div class='output'>An object of S4 class 'Coutput', containing following slots:
  @ratio: a matrix of 12 X 12, containing ratio
  @zscore: a matrix of 12 X 12, containing z-scores
  @pvalue: a matrix of 12 X 12, containing p-values
  @adjp: a matrix of 12 X 12, containing adjusted p-values
  @cnetwork: an object of S4 class 'Cnetwork', containing 5 interacting nodes
</div>
<div class='input'>
# 3) write into the files in your local directory
## <a href="Coutput_adjp.txt">Coutput_adjp.txt</a> for contact significant adjusted p-values
write(coutput, file='Coutput_adjp.txt', saveBy="adjp")
</div>
<strong class='message'>A file ('Coutput_adjp.txt') has been written into your local directory ('/Users/hfang/Sites/SUPERFAMILY/dcGOR')</strong>
<div class='input'>## <a href="Coutput_zscore.txt">Coutput_zscore.txt</a> for contact strength z-scores
write(coutput, file='Coutput_zscore.txt', saveBy="zscore")
</div>
<strong class='message'>A file ('Coutput_zscore.txt') has been written into your local directory ('/Users/hfang/Sites/SUPERFAMILY/dcGOR')</strong>
<div class='input'>## <a href="Coutput_node_info.txt">Coutput_node_info.txt</a> for node info
df <- tData(Anno)
ind <- match(rownames(zscore(coutput)), rownames(df))
out <- df[ind,]
write.table(out, file="Coutput_node_info.txt", col.names=T, row.names=F, sep="\t")

## 4) extract contact network
cnet <- cnetwork(coutput)
cnet
</div>
<div class='output'>An object of S4 class 'Cnetwork'
@adjMatrix: a weighted symmetric matrix of 5 samples/terms X 5 samples/terms 
@nodeInfo (InfoDataFrame)
  nodeNames: GO:0097159 GO:1901363 GO:0005488 GO:0003676 GO:0003677
  nodeAttr: name
</div>
<div class='input'>
## 5) visualise the contact network as a graph
### convert it to an object of class 'igraph' (for subsequent visualisation)
ig <- <a href='dcConverter.html' target=dcConverter>dcConverter</a>(cnet, from='Cnetwork', to='igraph')
</div>
<strong class='message'>Your input object 'cnet' of class 'Cnetwork' has been converted into an object of class 'igraph'.</strong>
<div class='input'>ig
</div>
<div class='output'>IGRAPH UNW- 5 10 -- 
+ attr: name (v/c), name.1 (v/c), weight (e/n)
</div>
<div class='input'>### extract edge weight (with 2-digit precision)
x <- signif(E(ig)$weight, digits=2)
### rescale into an interval [1,4] as edge width
edge.width <- 1 + (x-min(x))/(max(x)-min(x))*3
### prepare the node labels (including term id and name)
ind <- match(V(ig)$name,termNames(Anno))
vertex.label <- paste(V(ig)$name, '\n', tData(Anno)[ind,]$Name, sep='')
### do visualisation
dnet::visNet(g=ig, vertex.label=vertex.label, vertex.label.color="blue", vertex.label.cex=0.7, vertex.shape="sphere", vertex.color="blue", edge.width=edge.width, edge.label=x, edge.label.cex=0.7)</div>
<a href="javascript:newWin('Fang-194.png', 'Fang-194.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='Fang-194.png', width="400" /></a>
      </pre>
    </div>

  <div class="span4">

    <h2>Source demo</h2>
    <a href="javascript:newWin('demo/Fang.r', 'demo/Fang.r', '1200', '1200')" title="Click to view"><B><code>Fang.r</code></B></a>
    
    <h2>Functions used in this demo</h2>
    <ul>
      <li><a href="dcRDataLoader.html">dcRDataLoader</a></li>
      <li><a href="dcEnrichment.html">dcEnrichment</a></li>
      <li><a href="visEnrichment.html">visEnrichment</a></li>
      <li><a href="dcDAGannotate.html">dcDAGannotate</a></li>
      <li><a href="dcDAGdomainSim.html">dcDAGdomainSim</a></li>
      <li><a href="dcConverter.html">dcConverter</a></li>
      <li><a href="dcRWRpipeline.html">dcRWRpipeline</a></li>
    </ul>
    
    <!--
    
    <h2>Citation</h2>
    <p>Fang H. (2014) <B>dcGOR: an R package for analysing ontologies and protein domain annotations</B>. <I>PLoS Computational Biology</I> 10(10):e1003929. <a href="http://dx.doi.org/10.1371/journal.pcbi.1003929">doi:10.1371/journal.pcbi.1003929</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25356683">25356683</a></p>

<p>You may also cite the package as:</p>

<ul>
<li><code>database</code></li>
</ul>

<p>Fang H, Gough J. (2013) <B>dcGO: database of domain-centric ontologies on functions, phenotypes, diseases and more</B>. <I>Nucleic Acids Research</I> 41(Database issue):D536-44. <a href="http://dx.doi.org/10.1093/nar/gks1080">doi:10.1093/nar/gks1080</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23161684">23161684</a></p>

<ul>
<li><code>web link/URL</code></li>
</ul>

<p>Package &ldquo;dcGOR&rdquo; [<a href="http://cran.r-project.org/package=dcGOR">http://cran.r-project.org/package=dcGOR</a>]</p>

    

    
    <h2>URL</h2>
    <ul>
      <li><a href="http://supfam.org/dcGOR">http://supfam.org/dcGOR</a></li>
      <li><a href=" http://dcgor.r-forge.r-project.org"> http://dcgor.r-forge.r-project.org</a></li>
      <li><a href=" http://cran.r-project.org/package=dcGOR"> http://cran.r-project.org/package=dcGOR</a></li>
      <li><a href=" https://github.com/hfang-bristol/dcGOR"> https://github.com/hfang-bristol/dcGOR</a></li>
    </ul>
    
    <h2>Author</h2>
    Hai Fang and Julian Gough
    
    -->
    
  </div>

</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>