<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>Docs @ dcGOR 1.0.5</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="Hai Fang and Julian Gough">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dcGOR 1.0.5</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
       
        <!-- HTML for GOOLE SEARCH BAR 
        <ul class="nav">
            <div id="search-wrapper">
                <form id="search" method="get" action="http://www.google.com/custom">
                    <div class="search-box">
                        <i class="icon-search icon-white"></i>
                        <input type="text" id="tfq" class="textinput" name="q" size="21" maxlength="120" value="Google Search"/>
                    </div>
                </form>
            </div>
        </ul>
        -->
        
      </div>
    </div>
  </div>
</div>

<!-- CSS styles for standard search box with placeholder text-->
<style type="text/css">
    #search-wrapper {
        width:300px;
        height:40px;
        background-color:transparent;
        margin:0px auto 0;
        border:0px solid #ff0000;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

    #search {
        background-color:transparent;
        padding:0px;
        height:40px;
        overflow:hidden;
    }

    .search-box {
        width:250px;
        height:32px;
        background-color:transparent;
        margin:2px 7px;
        border:0px solid #cfcfcf;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

	.textinput{
		margin: 0;
		padding: 5px 15px;
		font-family: Arial, Helvetica, sans-serif;
		font-size:14px;
		color:#999999;
		border:1px solid #f00; border-right:0px;
		border-top-left-radius: 5px 5px;
		border-bottom-left-radius: 5px 5px;
	}
</style>

<!-- JAVASCRIPT to clear search text when the field is clicked -->
<script type="text/javascript">
window.onload = function(){
	//Get submit button
	var submitbutton = document.getElementById("tfq");
	//Add listener to submit button
	if(submitbutton.addEventListener){
		submitbutton.addEventListener("click", function() {
			if (submitbutton.value == 'Google Search'){//Customize this text string to whatever you want
				submitbutton.value = 'dcGOR R ';
			}
		});
	}
}
</script>

<script>
  (function() {
    var cx = '012094530736424854714:k8ieyz4ag4a';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') +
        '//www.google.com/cse/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>

<div><gcse:search></gcse:search></div>

    <div class="container">
      <header>
        
      </header>
      
      <div class="row">
  <div class="span8">

    <h1>Documentations</h1><p>

    <h3>Functions for analysis and visualisations</h3>
    <p>These analysis and visualisation functions are used to process ontologies (and annotations), to do enrichment analysis, to calculate semantic similarity between annotated domains based on ontology term semantic similarity, and to perform random walk with restart upon domain-domain (semantic) networks. Most of analyses are supported by high-performance parallel computing.</p>

    
    <ul class="index">
        
      <li>
        <code><a href="dcDAGannotate.html">dcDAGannotate</a></code><br />Function to generate a subgraph of a direct acyclic graph (DAG) induced by the input annotation data</li>
            
      <li>
        <code><a href="dcRDataLoader.html">dcRDataLoader</a></code><br />Function to load dcGOR built-in RData</li>
            
      <li>
        <code><a href="dcConverter.html">dcConverter</a></code><br />Function to convert an object between graph classes</li>
            
      <li>
        <code><a href="dcEnrichment.html">dcEnrichment</a></code><br />Function to conduct ontology enrichment analysis given a group of domains</li>
            
      <li>
        <code><a href="visEnrichment.html">visEnrichment</a></code><br />Function to visualise enrichment analysis outputs in the context of the ontology hierarchy</li>
            
      <li>
        <code><a href="dcDAGdomainSim.html">dcDAGdomainSim</a></code><br />Function to calculate pair-wise semantic similarity between domains based on a direct acyclic graph (DAG) with annotated data</li>
            
      <li>
        <code><a href="dcRWRpipeline.html">dcRWRpipeline</a></code><br />Function to setup a pipeine to estimate RWR-based contact strength between samples from an input domain-sample data matrix and an input graph</li>
    
    </ul>
    <h3>Functions for built-in (customised) data building</h3>
    <p>These functions are used to build objects of S4 classes &#39;InfoDataFrame&#39;, &#39;Anno&#39; and &#39;Onto&#39; from user-customised input data (domains, ontologies and annotations).</p>

    
    <ul class="index">
        
      <li>
        <code><a href="dcBuildInfoDataFrame.html">dcBuildInfoDataFrame</a></code><br />Function to build an object of the S4 class InfoDataframe from an input file</li>
            
      <li>
        <code><a href="dcBuildAnno.html">dcBuildAnno</a></code><br />Function to build an object of the S4 class Anno from input files</li>
            
      <li>
        <code><a href="dcBuildOnto.html">dcBuildOnto</a></code><br />Function to build an object of the S4 class Onto from input files</li>
    
    </ul>
    <h3>Definitions for S4 classes and methods</h3>
    <p>These documentations are to help understand S4 classes and methods defined in the package.</p>

    
    <ul class="index">
        
      <li>
        <code><a href="InfoDataFrame-class.html">InfoDataFrame-class</a></code>(InfoDataFrame)<br />Definition for S4 class InfoDataFrame</li>
            
      <li>
        <code><a href="InfoDataFrame-method.html">InfoDataFrame-method</a></code>(Data, Data,InfoDataFrame-method, [,InfoDataFrame,ANY,ANY,ANY-method, [,InfoDataFrame-method, colNames, colNames,InfoDataFrame-method, data.frame2InfoDataFrame, dim,InfoDataFrame-method, dimLabels, dimLabels,InfoDataFrame-method, ncol, ncol,InfoDataFrame-method, nrow, nrow,InfoDataFrame-method, rowNames, rowNames,InfoDataFrame-method, show,InfoDataFrame-method)<br />Methods defined for S4 class InfoDataFrame</li>
            
      <li>
        <code><a href="AnnoData-class.html">AnnoData-class</a></code>(AnnoData)<br />Definition for VIRTUAL S4 class AnnoData</li>
            
      <li>
        <code><a href="Anno-class.html">Anno-class</a></code>(Anno)<br />Definition for S4 class Anno</li>
            
      <li>
        <code><a href="Anno-method.html">Anno-method</a></code>([,Anno,ANY,ANY,ANY-method, [,Anno-method, annoData, annoData,Anno-method, dData, dData,Anno-method, dgCMatrix2Anno, dim,Anno-method, domainData, domainData,Anno-method, domainNames, domainNames,Anno-method, matrix2Anno, show,Anno-method, tData, tData,Anno-method, termData, termData,Anno-method, termNames, termNames,Anno-method)<br />Methods defined for S4 class Anno</li>
            
      <li>
        <code><a href="AdjData-class.html">AdjData-class</a></code>(AdjData)<br />Definition for VIRTUAL S4 class AdjData</li>
            
      <li>
        <code><a href="Onto-class.html">Onto-class</a></code>(Onto)<br />Definition for S4 class Onto</li>
            
      <li>
        <code><a href="Onto-method.html">Onto-method</a></code>([,Onto,ANY,ANY,ANY-method, [,Onto-method, adjMatrix, adjMatrix,Onto-method, dgCMatrix2Onto, dim,Onto-method, matrix2Onto, nInfo, nInfo,Onto-method, nodeInfo, nodeInfo,Onto-method, nodeNames, nodeNames,Onto-method, show,Onto-method, term_distance, term_distance,Onto-method, term_id, term_id,Onto-method, term_name, term_name,Onto-method, term_namespace, term_namespace,Onto-method)<br />Methods defined for S4 class Onto</li>
            
      <li>
        <code><a href="Eoutput-class.html">Eoutput-class</a></code>(Eoutput)<br />Definition for S4 class Eoutput</li>
            
      <li>
        <code><a href="Eoutput-method.html">Eoutput-method</a></code>(adjp, adjp,Eoutput-method, pvalue, pvalue,Eoutput-method, show,Eoutput-method, view, view,Eoutput-method, write, write,Eoutput-method, zscore, zscore,Eoutput-method)<br />Methods defined for S4 class Eoutput</li>
            
      <li>
        <code><a href="Dnetwork-class.html">Dnetwork-class</a></code>(Dnetwork)<br />Definition for S4 class Dnetwork</li>
            
      <li>
        <code><a href="Dnetwork-method.html">Dnetwork-method</a></code>([,Dnetwork,ANY,ANY,ANY-method, [,Dnetwork-method, adjMatrix,Dnetwork-method, description, description,Dnetwork-method, dgCMatrix2Dnetwork, dim,Dnetwork-method, id, id,Dnetwork-method, level, level,Dnetwork-method, matrix2Dnetwork, nInfo,Dnetwork-method, nodeInfo,Dnetwork-method, nodeNames,Dnetwork-method, show,Dnetwork-method)<br />Methods defined for S4 class Dnetwork</li>
            
      <li>
        <code><a href="Cnetwork-class.html">Cnetwork-class</a></code>(Cnetwork)<br />Definition for S4 class Cnetwork</li>
            
      <li>
        <code><a href="Cnetwork-method.html">Cnetwork-method</a></code>([,Cnetwork,ANY,ANY,ANY-method, [,Cnetwork-method, adjMatrix,Cnetwork-method, dgCMatrix2Cnetwork, dim,Cnetwork-method, matrix2Cnetwork, nInfo,Cnetwork-method, nodeInfo,Cnetwork-method, nodeNames,Cnetwork-method, show,Cnetwork-method)<br />Methods defined for S4 class Cnetwork</li>
            
      <li>
        <code><a href="Coutput-class.html">Coutput-class</a></code>(Coutput)<br />Definition for S4 class Coutput</li>
            
      <li>
        <code><a href="Coutput-method.html">Coutput-method</a></code>(adjp,Coutput-method, cnetwork, cnetwork,Coutput-method, pvalue,Coutput-method, ratio, ratio,Coutput-method, show,Coutput-method, write,Coutput-method, zscore,Coutput-method)<br />Methods defined for S4 class Coutput</li>
    
    </ul>
    <h3>Ontologies mainly including open biomedical ontology (obo)</h3>
    <p>These ontologies each are represented as a direct acyclic graph (DAG). DAG is stored as an object of class &#39;Onto&#39;.</p>

    
    <ul class="index">
        
      <li>
        <code><a href="onto.GOBP.html">onto.GOBP</a></code><br />Gene Ontology Biological Process (GOBP).</li>
            
      <li>
        <code><a href="onto.GOMF.html">onto.GOMF</a></code><br />Gene Ontology Molecular Function (GOMF).</li>
            
      <li>
        <code><a href="onto.GOCC.html">onto.GOCC</a></code><br />Gene Ontology Cellular Component (GOCC).</li>
            
      <li>
        <code><a href="onto.DO.html">onto.DO</a></code><br />Disease Ontology (DO).</li>
            
      <li>
        <code><a href="onto.HPPA.html">onto.HPPA</a></code><br />Human Phenotype Phenotypic Abnormality (HPPA).</li>
            
      <li>
        <code><a href="onto.HPMI.html">onto.HPMI</a></code><br />Human Phenotype Mode of Inheritance (HPMI).</li>
            
      <li>
        <code><a href="onto.HPON.html">onto.HPON</a></code><br />Human Phenotype ONset and clinical course (HPON).</li>
            
      <li>
        <code><a href="onto.MP.html">onto.MP</a></code><br />Mammalian Phenotype (MP).</li>
            
      <li>
        <code><a href="onto.EC.html">onto.EC</a></code><br />Enzyme Commission (EC).</li>
            
      <li>
        <code><a href="onto.KW.html">onto.KW</a></code><br />UniProtKB KeyWords (KW).</li>
            
      <li>
        <code><a href="onto.UP.html">onto.UP</a></code><br />UniProtKB UniPathway (UP).</li>
    
    </ul>
    <h3>SCOP domain superfamilies and their annotations by ontologies</h3>
    <p>These R objects are about SCOP domain superfamilies (sf) and their annotations by various ontologies, derived from the dcGO database.</p>

    
    <ul class="index">
        
      <li>
        <code><a href="SCOP.sf.html">SCOP.sf</a></code><br />SCOP domain superfamilies (sf).</li>
            
      <li>
        <code><a href="SCOP.sf2GOBP.html">SCOP.sf2GOBP</a></code><br />Annotations of SCOP domain superfamilies (sf) by Gene Ontology Biological Process (GOBP).</li>
            
      <li>
        <code><a href="SCOP.sf2GOMF.html">SCOP.sf2GOMF</a></code><br />Annotations of SCOP domain superfamilies (sf) by Gene Ontology Molecular Function (GOMF).</li>
            
      <li>
        <code><a href="SCOP.sf2GOCC.html">SCOP.sf2GOCC</a></code><br />Annotations of SCOP domain superfamilies (sf) by Gene Ontology Cellular Component (GOCC).</li>
            
      <li>
        <code><a href="SCOP.sf2DO.html">SCOP.sf2DO</a></code><br />Annotations of SCOP domain superfamilies (sf) by Disease Ontology (DO).</li>
            
      <li>
        <code><a href="SCOP.sf2HPPA.html">SCOP.sf2HPPA</a></code><br />Annotations of SCOP domain superfamilies (sf) by Human Phenotype Phenotypic Abnormality (HPPA).</li>
            
      <li>
        <code><a href="SCOP.sf2HPMI.html">SCOP.sf2HPMI</a></code><br />Annotations of SCOP domain superfamilies (sf) by Human Phenotype Mode of Inheritance (HPMI).</li>
            
      <li>
        <code><a href="SCOP.sf2HPON.html">SCOP.sf2HPON</a></code><br />Annotations of SCOP domain superfamilies (sf) by Human Phenotype ONset and clinical course (HPON).</li>
            
      <li>
        <code><a href="SCOP.sf2MP.html">SCOP.sf2MP</a></code><br />Annotations of SCOP domain superfamilies (sf) by Mammalian Phenotype (MP).</li>
            
      <li>
        <code><a href="SCOP.sf2EC.html">SCOP.sf2EC</a></code><br />Annotations of SCOP domain superfamilies (sf) by Enzyme Commission (EC).</li>
            
      <li>
        <code><a href="SCOP.sf2KW.html">SCOP.sf2KW</a></code><br />Annotations of SCOP domain superfamilies (sf) by UniProtKB KeyWords (KW).</li>
            
      <li>
        <code><a href="SCOP.sf2UP.html">SCOP.sf2UP</a></code><br />Annotations of SCOP domain superfamilies (sf) by UniProtKB UniPathway (UP).</li>
    
    </ul>
    <h3>SCOP domain families and their annotations by ontologies</h3>
    <p>These R objects are about SCOP domain families (fa) and their annotations by various ontologies, derived from the dcGO database.</p>

    
    <ul class="index">
        
      <li>
        <code><a href="SCOP.fa.html">SCOP.fa</a></code><br />SCOP domain families (fa).</li>
            
      <li>
        <code><a href="SCOP.fa2GOBP.html">SCOP.fa2GOBP</a></code><br />Annotations of SCOP domain families (fa) by Gene Ontology Biological Process (GOBP).</li>
            
      <li>
        <code><a href="SCOP.fa2GOMF.html">SCOP.fa2GOMF</a></code><br />Annotations of SCOP domain families (fa) by Gene Ontology Molecular Function (GOMF).</li>
            
      <li>
        <code><a href="SCOP.fa2GOCC.html">SCOP.fa2GOCC</a></code><br />Annotations of SCOP domain families (fa) by Gene Ontology Cellular Component (GOCC).</li>
            
      <li>
        <code><a href="SCOP.fa2DO.html">SCOP.fa2DO</a></code><br />Annotations of SCOP domain families (fa) by Disease Ontology (DO).</li>
            
      <li>
        <code><a href="SCOP.fa2HPPA.html">SCOP.fa2HPPA</a></code><br />Annotations of SCOP domain families (fa) by Human Phenotype Phenotypic Abnormality (HPPA).</li>
            
      <li>
        <code><a href="SCOP.fa2HPMI.html">SCOP.fa2HPMI</a></code><br />Annotations of SCOP domain families (fa) by Human Phenotype Mode of Inheritance (HPMI).</li>
            
      <li>
        <code><a href="SCOP.fa2HPON.html">SCOP.fa2HPON</a></code><br />Annotations of SCOP domain families (fa) by Human Phenotype ONset and clinical course (HPON).</li>
            
      <li>
        <code><a href="SCOP.fa2MP.html">SCOP.fa2MP</a></code><br />Annotations of SCOP domain families (fa) by Mammalian Phenotype (MP).</li>
            
      <li>
        <code><a href="SCOP.fa2EC.html">SCOP.fa2EC</a></code><br />Annotations of SCOP domain families (fa) by Enzyme Commission (EC).</li>
            
      <li>
        <code><a href="SCOP.fa2KW.html">SCOP.fa2KW</a></code><br />Annotations of SCOP domain families (fa) by UniProtKB KeyWords (KW).</li>
            
      <li>
        <code><a href="SCOP.fa2UP.html">SCOP.fa2UP</a></code><br />Annotations of SCOP domain families (fa) by UniProtKB UniPathway (UP).</li>
    
    </ul>
    <h3>Pfam domains and their annotations by ontologies</h3>
    <p>These R objects are about Pfam domains (Pfam) and their annotations by Gene Ontology (GO).</p>

    
    <ul class="index">
        
      <li>
        <code><a href="Pfam.html">Pfam</a></code><br />Pfam domains (Pfam).</li>
            
      <li>
        <code><a href="Pfam2GOBP.html">Pfam2GOBP</a></code><br />Annotations of Pfam domains by Gene Ontology Biological Process (GOBP).</li>
            
      <li>
        <code><a href="Pfam2GOMF.html">Pfam2GOMF</a></code><br />Annotations of Pfam domains by Gene Ontology Molecular Function (GOMF).</li>
            
      <li>
        <code><a href="Pfam2GOCC.html">Pfam2GOCC</a></code><br />Annotations of Pfam domains by Gene Ontology Cellular Component (GOCC).</li>
    
    </ul>
    <h3>InterPro domains and their annotations by ontologies</h3>
    <p>These R objects are about InterPro domains (InterPro) and their annotations by Gene Ontology (GO).</p>

    
    <ul class="index">
        
      <li>
        <code><a href="InterPro.html">InterPro</a></code><br />InterPro domains (InterPro).</li>
            
      <li>
        <code><a href="InterPro2GOBP.html">InterPro2GOBP</a></code><br />Annotations of InterPro domains by Gene Ontology Biological Process (GOBP).</li>
            
      <li>
        <code><a href="InterPro2GOMF.html">InterPro2GOMF</a></code><br />Annotations of InterPro domains by Gene Ontology Molecular Function (GOMF).</li>
            
      <li>
        <code><a href="InterPro2GOCC.html">InterPro2GOCC</a></code><br />Annotations of InterPro domains by Gene Ontology Cellular Component (GOCC).</li>
    
    </ul>
    <h3>Rfam RNA families and their annotations by ontologies</h3>
    <p>These R objects are about Rfam RNA families (Rfam) and their annotations by Gene Ontology (GO).</p>

    
    <ul class="index">
        
      <li>
        <code><a href="Rfam.html">Rfam</a></code><br />RNA families (Rfam).</li>
            
      <li>
        <code><a href="Rfam2GOBP.html">Rfam2GOBP</a></code><br />Annotations of Rfam RNA families by Gene Ontology Biological Process (GOBP).</li>
            
      <li>
        <code><a href="Rfam2GOMF.html">Rfam2GOMF</a></code><br />Annotations of Rfam RNA families by Gene Ontology Molecular Function (GOMF).</li>
            
      <li>
        <code><a href="Rfam2GOCC.html">Rfam2GOCC</a></code><br />Annotations of Rfam RNA families by Gene Ontology Cellular Component (GOCC).</li>
    
    </ul>
    <h3>Complete domains (domain-ome) in Eukaryotic tree of life (eTOL)</h3>
    <p>These databases and functions are used for domain-centric genome analysis in Eukaryotes. Note: these domains are defined as SCOP domain superfamilies.</p>

    
    <ul class="index">
        
      <li>
        <code><a href="Ancestral_domainome.html">Ancestral_domainome</a></code><br />Ancestral superfamily domain repertoires in Eukaryotes</li>
            
      <li>
        <code><a href="eTOL.html">eTOL</a></code><br />eukaryotic Tree Of Life (eTOL)</li>
            
      <li>
        <code><a href="dcAncestralML.html">dcAncestralML</a></code><br />Function to reconstruct ancestral discrete states using fast maximum likelihood algorithm</li>
            
      <li>
        <code><a href="dcAncestralMP.html">dcAncestralMP</a></code><br />Function to reconstruct ancestral discrete states using maximum parsimony algorithm</li>
            
      <li>
        <code><a href="dcSubtreeClade.html">dcSubtreeClade</a></code><br />Function to extract a subtree under a given clade from a phylo-formatted phylogenetic tree</li>
            
      <li>
        <code><a href="dcSubtreeTips.html">dcSubtreeTips</a></code><br />Function to extract a tip-induced subtree from a phylo-formatted phylogenetic tree</li>
            
      <li>
        <code><a href="dcTreeConnectivity.html">dcTreeConnectivity</a></code><br />Function to calculate the sparse connectivity matrix between parents and children from a phylo-formatted phylogenetic tree</li>
            
      <li>
        <code><a href="dcDuplicated.html">dcDuplicated</a></code><br />Function to determine the duplicated patterns from input data matrix</li>
    
    </ul>
    <h3>Functions for domain-centric ontology creation and ontology term predictions</h3>
    <p>These functions are used for creating domain-centric ontology, which in turn is used for predicting domain-centric ontology terms from input protein domain architectures.</p>

    
    <ul class="index">
        
      <li>
        <code><a href="dcSplitArch.html">dcSplitArch</a></code><br />Function to obtain a list of features via splitting an input architecture</li>
            
      <li>
        <code><a href="dcFunArgs.html">dcFunArgs</a></code><br />Function to assign (and evaluate) arguments with default values for an input function</li>
            
      <li>
        <code><a href="dcSparseMatrix.html">dcSparseMatrix</a></code><br />Function to create a sparse matrix for an input file with three columns</li>
            
      <li>
        <code><a href="dcList2Matrix.html">dcList2Matrix</a></code><br />Function to convert a list into a matrix containing three columns</li>
            
      <li>
        <code><a href="dcSupraBetter.html">dcSupraBetter</a></code><br />Function to find supra-domains with better scores than their individual domains</li>
            
      <li>
        <code><a href="dcAlgo.html">dcAlgo</a></code><br />Function to apply dcGO algorithm to infer domain-centric ontology</li>
            
      <li>
        <code><a href="dcAlgoPropagate.html">dcAlgoPropagate</a></code><br />Function to propagate ontology annotations according to an input file</li>
            
      <li>
        <code><a href="dcAlgoPredict.html">dcAlgoPredict</a></code><br />Function to predict ontology terms given domain architectures (including individual domains)</li>
            
      <li>
        <code><a href="dcAlgoPredictMain.html">dcAlgoPredictMain</a></code><br />Function to predict ontology terms given an input file containing domain architectures (including individual domains)</li>
            
      <li>
        <code><a href="dcAlgoPredictGenome.html">dcAlgoPredictGenome</a></code><br />Function to predict ontology terms for genomes with domain architectures (including individual domains)</li>
            
      <li>
        <code><a href="dcAlgoPredictPR.html">dcAlgoPredictPR</a></code><br />Function to assess the prediction performance via Precision-Recall (PR) analysis</li>
            
      <li>
        <code><a href="dcRWRpredict.html">dcRWRpredict</a></code><br />Function to perform RWR-based ontology term predictions from input known annotations and an input graph</li>
            
      <li>
        <code><a href="dcNaivePredict.html">dcNaivePredict</a></code><br />Function to perform naive prediction from input known annotations</li>
    
    </ul>
    <h3>Databases used for ontology term predictions</h3>
    <p>These databases contain ontology annotations (along with hypergeometric scores) for domains and domain combinations, which are then used for domain-centric ontology term predictions.</p>

    
    <ul class="index">
        
      <li>
        <code><a href="Feature2GOBP.sf.html">Feature2GOBP.sf</a></code><br />Annotations and scores between SCOP domains/combinations (at the superfamily level; sf) and Gene Ontology Biological Process (GOBP).</li>
            
      <li>
        <code><a href="Feature2GOMF.sf.html">Feature2GOMF.sf</a></code><br />Annotations and scores between SCOP domains/combinations (at the superfamily level; sf) and Gene Ontology Molecular Function (GOMF).</li>
            
      <li>
        <code><a href="Feature2GOCC.sf.html">Feature2GOCC.sf</a></code><br />Annotations and scores between SCOP domains/combinations (at the superfamily level; sf) and Gene Ontology Cellular Component (GOCC).</li>
            
      <li>
        <code><a href="Feature2HPPA.sf.html">Feature2HPPA.sf</a></code><br />Annotations and scores between SCOP domains/combinations (at the superfamily level; sf) and Human Phenotype Phenotypic Abnormality (HPPA).</li>
            
      <li>
        <code><a href="Feature2GOBP.pfam.html">Feature2GOBP.pfam</a></code><br />Annotations and scores between Pfam domains/combinations and Gene Ontology Biological Process (GOBP).</li>
            
      <li>
        <code><a href="Feature2GOMF.pfam.html">Feature2GOMF.pfam</a></code><br />Annotations and scores between Pfam domains/combinations and Gene Ontology Molecular Function (GOMF).</li>
            
      <li>
        <code><a href="Feature2GOCC.pfam.html">Feature2GOCC.pfam</a></code><br />Annotations and scores between Pfam domains/combinations and Gene Ontology Cellular Component (GOCC).</li>
            
      <li>
        <code><a href="Feature2HPPA.pfam.html">Feature2HPPA.pfam</a></code><br />Annotations and scores between Pfam domains/combinations and Human Phenotype Phenotypic Abnormality (HPPA).</li>
            
      <li>
        <code><a href="Feature2GOBP.interpro.html">Feature2GOBP.interpro</a></code><br />Annotations and scores between InterPro domains and Gene Ontology Biological Process (GOBP).</li>
            
      <li>
        <code><a href="Feature2GOMF.interpro.html">Feature2GOMF.interpro</a></code><br />Annotations and scores between InterPro domains and Gene Ontology Molecular Function (GOMF).</li>
            
      <li>
        <code><a href="Feature2GOCC.interpro.html">Feature2GOCC.interpro</a></code><br />Annotations and scores between InterPro domains and Gene Ontology Cellular Component (GOCC).</li>
            
      <li>
        <code><a href="Feature2HPPA.interpro.html">Feature2HPPA.interpro</a></code><br />Annotations and scores between InterPro domains and Human Phenotype Phenotypic Abnormality (HPPA).</li>
    
    </ul>
  </div>
  
  
  <div class="span4">
    
    <!--
    
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://supfam.org/dcGOR">http://supfam.org/dcGOR</a></li>
      <li><a href=" http://dcgor.r-forge.r-project.org"> http://dcgor.r-forge.r-project.org</a></li>
      <li><a href="
http://cran.r-project.org/package=dcGOR">
http://cran.r-project.org/package=dcGOR</a></li>
      <li><a href="
https://github.com/hfang-bristol/dcGOR">
https://github.com/hfang-bristol/dcGOR</a></li>
    </ul>
    
    <h2>Citation</h2>
    <p>Fang H. (2014) <B>dcGOR: an R package for analysing ontologies and protein domain annotations</B>. <I>PLoS Computational Biology</I> 10(10):e1003929. <a href="http://dx.doi.org/10.1371/journal.pcbi.1003929">doi:10.1371/journal.pcbi.1003929</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=25356683">25356683</a></p>

<p>You may also cite the package as:</p>

<ul>
<li><code>database</code></li>
</ul>

<p>Fang H, Gough J. (2013) <B>dcGO: database of domain-centric ontologies on functions, phenotypes, diseases and more</B>. <I>Nucleic Acids Research</I> 41(Database issue):D536-44. <a href="http://dx.doi.org/10.1093/nar/gks1080">doi:10.1093/nar/gks1080</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=23161684">23161684</a></p>

<ul>
<li><code>web link/URL</code></li>
</ul>

<p>Package &ldquo;dcGOR&rdquo; [<a href="http://cran.r-project.org/package=dcGOR">http://cran.r-project.org/package=dcGOR</a>]</p>

    
    <h2>Author</h2>
    Hai Fang and Julian Gough
    
    -->
    
  </div>
  
  
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>